<!DOCTYPE html>



  


<html class="theme-next mist use-motion" lang="zh-Hans">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">








<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css">




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css">


  <meta name="keywords" content="Hexo, NexT">





  <link rel="alternate" href="/atom.xml" title="诸子百家" type="application/atom+xml">




  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.2">






<meta name="description" content="本教程目的是对上一步生成的BAM文件进行质控，并检测变异，以及对变异进行注释，参考教程：Single Nucleotide Variant Calling and Annotation，Data pre-processing for variant discovery，以及Somatic short variant discovery (SNVs + Indels) 一、查看BAM信息我们先来具体">
<meta name="keywords" content="诸子,百家,诸子百家">
<meta property="og:type" content="article">
<meta property="og:title" content="肿瘤基因组分析教程：四、BAM文件预处理">
<meta property="og:url" content="http://uteric.github.io/cancer_seq4/index.html">
<meta property="og:site_name" content="诸子百家">
<meta property="og:description" content="本教程目的是对上一步生成的BAM文件进行质控，并检测变异，以及对变异进行注释，参考教程：Single Nucleotide Variant Calling and Annotation，Data pre-processing for variant discovery，以及Somatic short variant discovery (SNVs + Indels) 一、查看BAM信息我们先来具体">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="http://uteric.github.io/images/WES/cancer16.png">
<meta property="og:image" content="http://uteric.github.io/images/WES/cancer17.png">
<meta property="og:image" content="http://uteric.github.io/images/WES/cancer18.png">
<meta property="og:updated_time" content="2020-11-22T19:08:22.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="肿瘤基因组分析教程：四、BAM文件预处理">
<meta name="twitter:description" content="本教程目的是对上一步生成的BAM文件进行质控，并检测变异，以及对变异进行注释，参考教程：Single Nucleotide Variant Calling and Annotation，Data pre-processing for variant discovery，以及Somatic short variant discovery (SNVs + Indels) 一、查看BAM信息我们先来具体">
<meta name="twitter:image" content="http://uteric.github.io/images/WES/cancer16.png">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '9e233f3fa2f200ca831529de7ba10253',
      indexName: 'uteric',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://uteric.github.io/cancer_seq4/">





  <title>肿瘤基因组分析教程：四、BAM文件预处理 | 诸子百家</title>
  




<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-104081724-1', 'auto');
ga('send', 'pageview');

</script>











</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <meta name="baidu-site-verification" content="bfQpbnNmEL">

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">诸子百家</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br>
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br>
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br>
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br>
            
            关于
          </a>
        </li>
      

      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br>
            
            搜索
          </a>
        </li>
      
    </ul>
  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
<div class="local-search-header clearfix">
<span class="search-icon">
<i class="fa fa-search"></i>
</span>
<span class="popup-btn-close">
<i class="fa fa-times-circle"></i>
</span>
<div class="local-search-input-wrapper">
<input autocomplete="off" placeholder="搜索..." spellcheck="false" type="text" id="local-search-input">
</div>
</div>
<div id="local-search-result"></div>
</div>


    </div>
  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://uteric.github.io/cancer_seq4/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="括囊无誉">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="诸子百家">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">肿瘤基因组分析教程：四、BAM文件预处理</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-11-22T13:08:22-06:00">
                2020-11-22
              </time>
            

            

            
          </span>

          


          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/cancer_seq4/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count" data-disqus-identifier="cancer_seq4/" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          

          
            <span class="post-meta-divider">|</span>
            <span class="page-pv"><i class="fa fa-file-o"></i> 热度
            <span class="busuanzi-value" id="busuanzi_value_page_pv"></span>°C
            </span>
          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <p>本教程目的是对上一步生成的BAM文件进行质控，并检测变异，以及对变异进行注释，参考教程：<a href="https://bpa-csiro-workshops.github.io/cancer-manuals/modules/cancer-module-snv/snv/" target="_blank" rel="noopener">Single Nucleotide Variant Calling and Annotation</a>，<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035535912" target="_blank" rel="noopener">Data pre-processing for variant discovery</a>，以及<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-" target="_blank" rel="noopener">Somatic short variant discovery (SNVs + Indels)</a></p>
<h2 id="一、查看BAM信息"><a href="#一、查看BAM信息" class="headerlink" title="一、查看BAM信息"></a>一、查看BAM信息</h2><p>我们先来具体看一下BAM文件，下面是BAM文件主体内容的前4行</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">$ samtools view SRR5478491C.bam | head -n 4</span><br><span class="line"></span><br><span class="line">SRR5478491.1    163     chr1    10037   19      13M1I87M        =       10064   102     TAACCCTAACCCTAAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCAACCCTAACCCTAACCCTAACCCTAGCC   ?@@DDD?BBBAF?GFBAA&lt;AAGEFHIJGGGE?DFIIBHFBBDF;F?FC8B=BFH=CD===E;CEHBBDFE66?5;;38?CC5?9?ADD(8?&lt;&lt;B1(2&lt;A@C   NM:i:2  MD:Z:97A2       MC:Z:75M        AS:i:90 XS:i:91 RG:Z:SRR5478491</span><br><span class="line">SRR5478491.2    163     chr1    10057   35      59M2D39M3S      =       10081   103     ACCCTCACCCTCACCCTCACCCTCACCCTAACCCTAACCCTAACCCTAACCCAACCCTACCCAACCCTAACCCTAACCCTAACCCTAACCCCTAACCCCTA   CCCFFFFFHHGHHJJJIJIGJGEIGIIIIJJIJIIJJIDHGIEHGHIJIHAAFHGIHHECD&gt;DFDCCECCBDDBCCBBD&lt;&lt;CDBBDCDBD&lt;5&lt;@&lt;ADD@@&lt;   NM:i:7  MD:Z:5A5A5A5A35^AC2T36  MC:Z:13S35M2D36M1I6M    AS:i:65</span><br><span class="line">XS:i:61 RG:Z:SRR5478491</span><br><span class="line">SRR5478491.1    83      chr1    10064   19      75M     =       10037   -102    CCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCAACCCTAACCCTAACCCTAACCCTAGCCCT     9?;55-;(A=;.&gt;@?6=)HAC==7CAF@=.&gt;CCB?98@?;B?AFC?*IGHGC9HHE@IIGGFAHFHFD?DDD?;=    NM:i:1   MD:Z:70A4       MC:Z:13M1I87M   AS:i:70 XS:i:65 RG:Z:SRR5478491</span><br><span class="line">SRR5478491.2    83      chr1    10081   35      13S35M2D36M1I6M =       10057   -103    CCCCCTCCCCCTCACCCTAACCCTAACCCTAACCCTAACCCAACCCTACCCAACCCTAACCCTAACCCTAACCCTAACCCCTAACCCCTAA     0&amp;-@&lt;90&amp;?DBDBC@?@B@DDCBBBABCAAACBEDB?BFHHHEHIGF@DGGDBEJIIHFIIHFC?JIHFCEIHEEIJIHHHFFDFFFFBB@     NM:i:4  MD:Z:35^AC2T39  MC:Z:59M2D39M3S AS:i:61 XS:i:58 RG:Z:SRR5478491</span><br></pre></td></tr></table></figure>
<p>SAM/BAM文件至少有11列，内容如下</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1. QNAME, Query template NAME</span></span><br><span class="line"><span class="comment"># 2. FLAG, bitwise FLAG</span></span><br><span class="line"><span class="comment"># 3. RNAME, Reference sequence NAME</span></span><br><span class="line"><span class="comment"># 4. POS, 1-based</span></span><br><span class="line"><span class="comment"># 5. MAPQ, MAPping Quality</span></span><br><span class="line"><span class="comment"># 6. CIGAR, </span></span><br><span class="line"><span class="comment"># 7. RNEXT, Reference name of the mate/next read</span></span><br><span class="line"><span class="comment"># 8. PNEXT, Position of the mate/next read</span></span><br><span class="line"><span class="comment"># 9. TLEN, observed Template LENgth</span></span><br><span class="line"><span class="comment"># 10. SEQ, segment SEQuence</span></span><br><span class="line"><span class="comment"># 11. QUAL, ASCII of Phred-scaled base QUALity+33</span></span><br></pre></td></tr></table></figure>
<p>特别关注一下第6列，此列名为CIGAR，比如，13M1I87M，代表如下信息：匹配13个碱基，第14个碱基是插入，从15至101（87个）碱基匹配，当然，M可以表示匹配也可表示SNP，即SNP也会被标记为M</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">M == base aligns but doesn’t have to be a match. A SNP will have an M even <span class="keyword">if</span> it disagrees with the reference.</span><br><span class="line">I == Insertion</span><br><span class="line">D == Deletion</span><br><span class="line">S == soft-clips. These are handy to find un removed adapters, viral insertions, etc.</span><br></pre></td></tr></table></figure>
<p>关于SAM//BAM文件的格式，可参见其 <a href="http://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank" rel="noopener">说明文件</a>或<a href="https://genome.sph.umich.edu/wiki/SAM" target="_blank" rel="noopener">SAM</a>；关于FLAG，可参看<a href="http://broadinstitute.github.io/picard/explain-flags.html" target="_blank" rel="noopener">FLAG</a>，通过查询，可知上面的163代表:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">read</span> paired (0x1)</span><br><span class="line"><span class="built_in">read</span> mapped <span class="keyword">in</span> proper pair (0x2)</span><br><span class="line">mate reverse strand (0x20)</span><br><span class="line">second <span class="keyword">in</span> pair (0x80)</span><br></pre></td></tr></table></figure>
<p>查看不配对的reads，可以使用samtools view -c -f4实现，若查看配对的reads，可以使用samtools view -c -F4实现</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># -c       print only the count of matching records # 打印数目</span></span><br><span class="line"><span class="comment"># -f INT   only include reads with all  of the FLAGs in INT present [0] # 包含指定FLAG的reads，在此指定4，即read unmapped (0x4)</span></span><br><span class="line"><span class="comment"># -F INT   only include reads with none of the FLAGS in INT present [0] # 不含指定FLAG的reads，在此指定4，即read unmapped (0x4)</span></span><br><span class="line"></span><br><span class="line">$ samtools view -c -f4 SRR5478491C.bam</span><br><span class="line"><span class="comment"># 0</span></span><br><span class="line">$ samtools view -c -F4 SRR5478491C.bam</span><br><span class="line"><span class="comment"># 36533039</span></span><br></pre></td></tr></table></figure>
<h2 id="二、BAM去除DUPLICATES"><a href="#二、BAM去除DUPLICATES" class="headerlink" title="二、BAM去除DUPLICATES"></a>二、BAM去除DUPLICATES</h2><p>现在，我们得到的BAM文件称为RAW BAM，需要进行预处理，主要包括去掉PCR重复和BQSR，如下图</p>
<p><img src="/images/WES/cancer16.png" alt></p>
<p>PCR去重可以通过MarkDuplicates来实现，即使用MarkDuplicates来标记重复，并将其直接去掉，不输出到BAM文件中，再使用samtools index生成索引文件。如果电脑性能比较强悍，建议使用最新的<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360040096212-MarkDuplicatesSpark" target="_blank" rel="noopener">MarkDuplicatesSpark</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">$ gatk MarkDuplicates -I /mnt/d/cancer/bam/SRR5478491C.bam -O /mnt/d/cancer/nodup/SRR5478491C.nodup.bam -M /mnt/d/cancer/nodup/SRR5478491C.metrics.txt --REMOVE_DUPLICATES <span class="literal">true</span> --ASSUME_SORTED <span class="literal">true</span> --CREATE_INDEX <span class="literal">true</span></span><br><span class="line">$ gatk MarkDuplicates -I /mnt/d/cancer/bam/SRR5478492C.bam -O /mnt/d/cancer/nodup/SRR5478492C.nodup.bam -M /mnt/d/cancer/nodup/SRR5478492C.metrics.txt --REMOVE_DUPLICATES <span class="literal">true</span> --ASSUME_SORTED <span class="literal">true</span> --CREATE_INDEX <span class="literal">true</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 记录一下文件大小</span></span><br><span class="line">total 5.5G</span><br><span class="line">-rwxrwxrwx 1 eric eric 3.4K Nov 22 15:21 SRR5478491C.metrics.txt</span><br><span class="line">-rwxrwxrwx 1 eric eric 2.9G Nov 22 15:21 SRR5478491C.nodup.bam</span><br><span class="line">-rwxrwxrwx 1 eric eric 4.1M Nov 22 15:27 SRR5478491C.nodup.bam.bai</span><br><span class="line">-rwxrwxrwx 1 eric eric 3.4K Nov 22 15:21 SRR5478492C.metrics.txt</span><br><span class="line">-rwxrwxrwx 1 eric eric 2.7G Nov 22 15:21 SRR5478492C.nodup.bam</span><br><span class="line">-rwxrwxrwx 1 eric eric 4.0M Nov 22 15:27 SRR5478492C.nodup.bam.bai</span><br><span class="line"></span><br><span class="line"><span class="comment"># metrics.txt记录duplicates的信息，查看可知，重复率0.000003</span></span><br><span class="line"><span class="comment">## METRICS CLASS        picard.sam.DuplicationMetrics</span></span><br><span class="line">LIBRARY UNPAIRED_READS_EXAMINED READ_PAIRS_EXAMINED     SECONDARY_OR_SUPPLEMENTARY_RDS  UNMAPPED_READS  UNPAIRED_READ_DUPLICATES        READ_PAIR_DUPLICATES    READ_PAIR_OPTICAL_DUPLICATES    PERCENT_DUPLICATION     ESTIMATED_LIBRARY_SIZE</span><br><span class="line">P1T     5       18202826        127381  0       1       62      0       0.000003        2672112597763</span><br></pre></td></tr></table></figure>
<h2 id="三、校正碱基质量"><a href="#三、校正碱基质量" class="headerlink" title="三、校正碱基质量"></a>三、校正碱基质量</h2><p>现在，我们得到的是去掉PCR重复的BAM文件，下面进行Base (Quality Score) Recalibration <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890531" target="_blank" rel="noopener">(BQSR)</a>，即碱基质量校正，原因在于，测序过程中不同厂商的碱基读取有不同的倾向，需要重新校正，此步是通过机器学习来对测序结果中的碱基质量进行校正，通过两步实现:</p>
<ol>
<li>Build covariates based on context and known SNP sites. 根据物种已知SNP，生成用于质量校准的metrics</li>
<li>Correct the reads based on these metrics. 根据metrics文件生成新的BAM文件</li>
</ol>
<p>首先，需要先为hg19.fa生成dict文件，这可以通过调用CreateSequenceDictonary模块来实现</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ gatk CreateSequenceDictionary -R hg19.fa -O hg19.dict</span><br></pre></td></tr></table></figure>
<h3 id="3-1-BQSR-step1，BaseRecalibrator"><a href="#3-1-BQSR-step1，BaseRecalibrator" class="headerlink" title="3.1 BQSR step1，BaseRecalibrator"></a>3.1 BQSR step1，BaseRecalibrator</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">$ gatk BaseRecalibrator -I /mnt/d/cancer/nodup/SRR5478491C.nodup.bam -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa --known-sites /mnt/d/ncbi/hg19/bqsr/1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz --known-sites /mnt/d/ncbi/hg19/bqsr/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz  --known-sites /mnt/d/ncbi/hg19/bqsr/dbsnp_138.hg19.vcf.gz -L /mnt/d/ncbi/wxs/agilent/sureSelect_All_Exon_V5/S04380110_Padded.bed -O /mnt/d/cancer/bqsr/recal_SRR5478491C.table</span><br><span class="line"></span><br><span class="line">$ gatk BaseRecalibrator -I /mnt/d/cancer/nodup/SRR5478492C.nodup.bam -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa --known-sites /mnt/d/ncbi/hg19/bqsr/1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz --known-sites /mnt/d/ncbi/hg19/bqsr/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz  --known-sites /mnt/d/ncbi/hg19/bqsr/dbsnp_138.hg19.vcf.gz -L /mnt/d/ncbi/wxs/agilent/sureSelect_All_Exon_V5/S04380110_Padded.bed -O /mnt/d/cancer/bqsr/recal_SRR5478492C.table</span><br></pre></td></tr></table></figure>
<p>–known-sites中用到的三个VCF文件，用于提供物种已知的多态性位点，这些位点将被分析程序跳过，下载地址<a href="ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg19/" target="_blank" rel="noopener">resources bundle of hg19</a>，下载好的文件可以解压后使用，但解压文件空间占用太大，建议重新压缩并生成索引后使用</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">$ bgzip -d ***.vcf.gz</span><br><span class="line">$ bgzip ***.vcf <span class="comment"># 压缩比约4:1 - 8:1，必须使用bgzip压缩</span></span><br><span class="line">$ bcftools index -t ***.vcf.gz <span class="comment"># 生成.tbi格式索引</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># VCF文件如下</span></span><br><span class="line">total 3.3G</span><br><span class="line">-rwxrwxrwx 1 eric eric 1.8G Nov 22 19:03 1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz</span><br><span class="line">-rwxrwxrwx 1 eric eric 2.1M Nov 22 19:10 1000G_phase1.snps.high_confidence.hg19.sites.vcf.gz.tbi</span><br><span class="line">-rwxrwxrwx 1 eric eric  20M Nov 22 18:48 Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz</span><br><span class="line">-rwxrwxrwx 1 eric eric 1.4M Nov 22 19:06 Mills_and_1000G_gold_standard.indels.hg19.sites.vcf.gz.tbi</span><br><span class="line">-rwxrwxrwx 1 eric eric 1.5G Nov 22 18:20 dbsnp_138.hg19.vcf.gz</span><br><span class="line">-rwxrwxrwx 1 eric eric 2.2M Nov 22 18:26 dbsnp_138.hg19.vcf.gz.tbi</span><br></pre></td></tr></table></figure>
<h3 id="3-2-BQSR-step2，ApplyBQSR"><a href="#3-2-BQSR-step2，ApplyBQSR" class="headerlink" title="3.2 BQSR step2，ApplyBQSR"></a>3.2 BQSR step2，ApplyBQSR</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ gatk ApplyBQSR -I /mnt/d/cancer/nodup/SRR5478491C.nodup.bam -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa --bqsr-recal-file /mnt/d/cancer/bqsr/recal_SRR5478491C.table -L /mnt/d/ncbi/wxs/agilent/sureSelect_All_Exon_V5/S04380110_Padded.bed -O /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.bam --create-output-bam-index <span class="literal">true</span></span><br><span class="line">$ gatk ApplyBQSR -I /mnt/d/cancer/nodup/SRR5478492C.nodup.bam -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa --bqsr-recal-file /mnt/d/cancer/bqsr/recal_SRR5478492C.table -L /mnt/d/ncbi/wxs/agilent/sureSelect_All_Exon_V5/S04380110_Padded.bed -O /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam --create-output-bam-index <span class="literal">true</span></span><br></pre></td></tr></table></figure>
<h2 id="四、BAM文件质控"><a href="#四、BAM文件质控" class="headerlink" title="四、BAM文件质控"></a>四、BAM文件质控</h2><h3 id="4-1-计算覆盖度"><a href="#4-1-计算覆盖度" class="headerlink" title="4.1 计算覆盖度"></a>4.1 计算覆盖度</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 使用[DepthOfCoverage](https://gatk.broadinstitute.org/hc/en-us/articles/360046789152-DepthOfCoverage-BETA-)可以计算BAM文件的覆盖度</span></span><br><span class="line">gatk \</span><br><span class="line">   DepthOfCoverage \</span><br><span class="line">   -R reference.fasta \</span><br><span class="line">   -O file_name_base \</span><br><span class="line">   -I input_bams.list</span><br><span class="line">   [-geneList refSeq.sorted.refseq] \</span><br><span class="line">   [-pt readgroup] \</span><br><span class="line">   [-ct 4 -ct 6 -ct 10] \</span><br><span class="line">   [-L my_capture_genes.interval_list]</span><br><span class="line"><span class="comment"># 在本例中，使用如下参数</span></span><br><span class="line">$ gatk DepthOfCoverage -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -O /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.coverage -I /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.bam -L /mnt/d/ncbi/wxs/agilent/sureSelect_All_Exon_V5/S04380110_Padded.bed --omit-depth-output-at-each-base <span class="literal">true</span> --summary-coverage-threshold 10 --summary-coverage-threshold 25 --summary-coverage-threshold 50 --summary-coverage-threshold 100</span><br><span class="line"></span><br><span class="line">$ gatk DepthOfCoverage -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -O /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.coverage -I /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam -L /mnt/d/ncbi/wxs/agilent/sureSelect_All_Exon_V5/S04380110_Padded.bed --omit-depth-output-at-each-base <span class="literal">true</span> --summary-coverage-threshold 10 --summary-coverage-threshold 25 --summary-coverage-threshold 50 --summary-coverage-threshold 100</span><br></pre></td></tr></table></figure>
<p>运行后看一下覆盖度</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">$ less SRR5478491C.bqsr.nodup.coverage.sample_summary</span><br><span class="line">sample_id,total,mean,granular_third_quartile,granular_median,granular_first_quartile,%_bases_above_10,%_bases_above_25,%_bases_above_50,%_bases_above_100</span><br><span class="line">ccRCC,3065081044,34.26,43,12,2,52.9,36.2,21.7,9.6</span><br><span class="line">Total,3065081044,34.26,N/A,N/A,N/A,N/A,N/A,N/A,N/A</span><br><span class="line"></span><br><span class="line">$ less SRR5478492C.bqsr.nodup.coverage.sample_summary</span><br><span class="line">sample_id,total,mean,granular_third_quartile,granular_median,granular_first_quartile,%_bases_above_10,%_bases_above_25,%_bases_above_50,%_bases_above_100</span><br><span class="line">normal,2945165227,32.92,48,18,4,60.3,42.0,23.8,7.5</span><br><span class="line">Total,2945165227,32.92,N/A,N/A,N/A,N/A,N/A,N/A,N/A</span><br><span class="line"></span><br><span class="line"><span class="comment"># 平均覆盖度 &gt; 30X，然而，中位数比均值小很多，数据并非特别理想</span></span><br></pre></td></tr></table></figure>
<h3 id="4-2-计算Insert-Size"><a href="#4-2-计算Insert-Size" class="headerlink" title="4.2 计算Insert Size"></a>4.2 计算Insert Size</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">gatk CollectInsertSizeMetrics \</span><br><span class="line">      I=input.bam \</span><br><span class="line">      O=insert_size_metrics.txt \</span><br><span class="line">      H=insert_size_histogram.pdf \</span><br><span class="line">      M=0.5</span><br><span class="line"></span><br><span class="line"><span class="comment"># 在本例中，使用如下参数：</span></span><br><span class="line">$ gatk CollectInsertSizeMetrics -I /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.bam -O /mnt/d/cancer/bqsr/SRR5478491C.insert_size_metrics.txt -H /mnt/d/cancer/bqsr/SRR5478491C.insert_size_histogram.pdf --VALIDATION_STRINGENCY SILENT -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa</span><br><span class="line"></span><br><span class="line">$ gatk CollectInsertSizeMetrics -I /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam -O /mnt/d/cancer/bqsr/SRR5478492C.insert_size_metrics.txt -H /mnt/d/cancer/bqsr/SRR5478492C.insert_size_histogram.pdf --VALIDATION_STRINGENCY SILENT -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa</span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看一下insert size，Normal的insert size比ccRCC小一点(123 vs 134)</span></span><br><span class="line"></span><br><span class="line">$ less SRR5478491C.insert_size_metrics.txt</span><br><span class="line"><span class="comment">## METRICS CLASS        picard.analysis.InsertSizeMetrics</span></span><br><span class="line">MEDIAN_INSERT_SIZE      MODE_INSERT_SIZE        MEDIAN_ABSOLUTE_DEVIATION       MIN_INSERT_SIZE MAX_INSERT_SIZE MEAN_INSERT_SIZE</span><br><span class="line">        STANDARD_DEVIATION      READ_PAIRS      PAIR_ORIENTATION        WIDTH_OF_10_PERCENT     WIDTH_OF_20_PERCENT     WIDTH_OF_30_PERCENT     WIDTH_OF_40_PERCENT     WIDTH_OF_50_PERCENT     WIDTH_OF_60_PERCENT     WIDTH_OF_70_PERCENT     WIDTH_OF_80_PERCENT     WIDTH_OF_90_PERCENT     WIDTH_OF_95_PERCENT     WIDTH_OF_99_PERCENT     SAMPLE  LIBRARY READ_GROUP</span><br><span class="line">123     117     15      7       210906619       126.082217      22.010779       15536547        FR      7       13      19      25      31      37      45      53      69      91      131</span><br><span class="line"></span><br><span class="line">$ less SRR5478492C.insert_size_metrics.txt</span><br><span class="line"><span class="comment">## METRICS CLASS        picard.analysis.InsertSizeMetrics</span></span><br><span class="line">MEDIAN_INSERT_SIZE      MODE_INSERT_SIZE        MEDIAN_ABSOLUTE_DEVIATION       MIN_INSERT_SIZE MAX_INSERT_SIZE MEAN_INSERT_SIZE</span><br><span class="line">        STANDARD_DEVIATION      READ_PAIRS      PAIR_ORIENTATION        WIDTH_OF_10_PERCENT     WIDTH_OF_20_PERCENT     WIDTH_OF_30_PERCENT     WIDTH_OF_40_PERCENT     WIDTH_OF_50_PERCENT     WIDTH_OF_60_PERCENT     WIDTH_OF_70_PERCENT     WIDTH_OF_80_PERCENT     WIDTH_OF_90_PERCENT     WIDTH_OF_95_PERCENT     WIDTH_OF_99_PERCENT     SAMPLE  LIBRARY READ_GROUP</span><br><span class="line">134     129     18      9       71334417        135.454785      24.140099       14838912        FR      7       15      21      29      37      45      53      63      77      89      125</span><br></pre></td></tr></table></figure>
<p><img src="/images/WES/cancer17.png" alt></p>
<p><img src="/images/WES/cancer18.png" alt></p>
<h3 id="4-3-计算Alignment-metrics"><a href="#4-3-计算Alignment-metrics" class="headerlink" title="4.3 计算Alignment metrics"></a>4.3 计算Alignment metrics</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br></pre></td><td class="code"><pre><span class="line">gatk CollectAlignmentSummaryMetrics \</span><br><span class="line">          R=reference_sequence.fasta \</span><br><span class="line">          I=input.bam \</span><br><span class="line">          O=output.txt</span><br><span class="line"></span><br><span class="line">$ gatk CollectAlignmentSummaryMetrics -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -I /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.bam -O /mnt/d/cancer/bqsr/SRR5478491C.align.txt --VALIDATION_STRINGENCY SILENT</span><br><span class="line"></span><br><span class="line">$ gatk CollectAlignmentSummaryMetrics -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -I /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam -O /mnt/d/cancer/bqsr/SRR5478492C.align.txt --VALIDATION_STRINGENCY SILENT</span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看一下，比对质量相当不错</span></span><br><span class="line">$ less -S SRR5478491C.align.txt</span><br><span class="line"><span class="comment">## METRICS CLASS        picard.analysis.AlignmentSummaryMetrics</span></span><br><span class="line">CATEGORY        TOTAL_READS     PF_READS        PCT_PF_READS    PF_NOISE_READS  PF_READS_ALIGNED        PCT_PF_READS_ALIGNED    P</span><br><span class="line">FIRST_OF_PAIR   15538245        15538245        1       0       15538245        1       1542727791      15532149        154218009</span><br><span class="line">SECOND_OF_PAIR  15537614        15537614        1       0       15537614        1       1533251102      15531520        153271206</span><br><span class="line">PAIR    31075859        31075859        1       0       31075859        1       3075978893      31063669        3074892161      3</span><br><span class="line"></span><br><span class="line">$ less -S SRR5478492C.align.txt</span><br><span class="line"><span class="comment">## METRICS CLASS        picard.analysis.AlignmentSummaryMetrics</span></span><br><span class="line">CATEGORY        TOTAL_READS     PF_READS        PCT_PF_READS    PF_NOISE_READS  PF_READS_ALIGNED        PCT_PF_READS_ALIGNED    P</span><br><span class="line">FIRST_OF_PAIR   14839431        14839431        1       0       14839431        1       1480977340      14834355        148050594</span><br><span class="line">SECOND_OF_PAIR  14838958        14838958        1       0       14838958        1       1473988914      14833866        147352039</span><br><span class="line">PAIR    29678389        29678389        1       0       29678389        1       2954966254      29668221        2954026333      2</span><br></pre></td></tr></table></figure>
<p>关于BAM文件的处理与质控就到此为止，我们现在得到了“Analysis-Ready Reads”的BAM文件，下一步就可以进行MUTATION CALLING。</p>

      
    </div>

<div>


<ul class="post-copyright">
<li class="post-copyright-author">
<strong>本文作者：</strong>括囊无誉
</li>
<li class="post-copyright-link">
<strong>本文链接：</strong>
<a href="/cancer_seq4/" title="肿瘤基因组分析教程：四、BAM文件预处理">cancer_seq4/</a>
</li>
<li class="post-copyright-license">
<strong>版权声明： </strong>
本博客所有文章均为原创作品，转载请注明出处！
</li>
</ul>

</div>

    
    
    


<div>

<div style="text-align:center;color: #ccc;font-size:14px;">------ 本文结束 ------</div>

</div>

    

    
      <div>
        <div style="padding: 10px 0; margin: 20px auto; width: 90%; text-align: center;">
  <div>坚持原创文章分享，您的支持将鼓励我继续创作！</div>
  <button id="rewardButton" disable="enable" onclick="var qr = document.getElementById('QR'); if (qr.style.display === 'none') {qr.style.display='block';} else {qr.style.display='none'}">
    <span>Donate</span>
  </button>
  <div id="QR" style="display: none;">

    
      <div id="wechat" style="display: inline-block">
        <img id="wechat_qr" src="/images/wechatpay.jpg" alt="括囊无誉 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
    

    

    

  </div>
</div>

      </div>
    

    

    <footer class="post-footer">
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/Linux/diskuse/" rel="next" title="LINUX查看磁盘使用情况">
                <i class="fa fa-chevron-left"></i> LINUX查看磁盘使用情况
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/cancer_seq5/" rel="prev" title="肿瘤基因组分析教程：五、检测变异">
                肿瘤基因组分析教程：五、检测变异 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a>
        </noscript>
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image" src="/images/avatar.png" alt="括囊无誉">
          <p class="site-author-name" itemprop="name">括囊无誉</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives/">
                <span class="site-state-item-count">229</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">11</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">224</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        
          <div class="feed-link motion-element">
            <a href="/atom.xml" rel="alternate">
              <i class="fa fa-rss"></i>
              RSS
            </a>
          </div>
        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="http://www.jianshu.com/u/75d3f95b058e" target="_blank" title="简书">
                  
                    <i class="fa fa-fw fa-http://www.jianshu.com/u/75d3f95b058e"></i>
                  
                    
                      简书
                    
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#一、查看BAM信息"><span class="nav-number">1.</span> <span class="nav-text">一、查看BAM信息</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#二、BAM去除DUPLICATES"><span class="nav-number">2.</span> <span class="nav-text">二、BAM去除DUPLICATES</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#三、校正碱基质量"><span class="nav-number">3.</span> <span class="nav-text">三、校正碱基质量</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#3-1-BQSR-step1，BaseRecalibrator"><span class="nav-number">3.1.</span> <span class="nav-text">3.1 BQSR step1，BaseRecalibrator</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3-2-BQSR-step2，ApplyBQSR"><span class="nav-number">3.2.</span> <span class="nav-text">3.2 BQSR step2，ApplyBQSR</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#四、BAM文件质控"><span class="nav-number">4.</span> <span class="nav-text">四、BAM文件质控</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#4-1-计算覆盖度"><span class="nav-number">4.1.</span> <span class="nav-text">4.1 计算覆盖度</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-2-计算Insert-Size"><span class="nav-number">4.2.</span> <span class="nav-text">4.2 计算Insert Size</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4-3-计算Alignment-metrics"><span class="nav-number">4.3.</span> <span class="nav-text">4.3 计算Alignment metrics</span></a></li></ol></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">
  
  &copy;  2017 - 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">括囊无誉</span>
</div>


<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


        
<div class="busuanzi-count">
  <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="site-uv">
      <i class="fa fa-user"></i> 访问人数
      <span class="busuanzi-value" id="busuanzi_value_site_uv"></span>
      
    </span>
  

  
    <span class="site-pv">
      <i class="fa fa-eye"></i> 总访问量
      <span class="busuanzi-value" id="busuanzi_value_site_pv"></span>
      次
    </span>
  
</div>


        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>



  


  

    
      <script id="dsq-count-scr" src="https://uteric-1.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'http://uteric.github.io/cancer_seq4/';
          this.page.identifier = 'cancer_seq4/';
          this.page.title = '肿瘤基因组分析教程：四、BAM文件预处理';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://uteric-1.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script>
    

  




	





  








  

  <script type="text/javascript">
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url);
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x" /></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x" /></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  
<script>
(function(){
    var bp = document.createElement('script');
    var curProtocol = window.location.protocol.split(':')[0];
    if (curProtocol === 'https') {
        bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';        
    }
    else {
        bp.src = 'http://push.zhanzhang.baidu.com/push.js';
    }
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(bp, s);
})();
</script>


  

  

  

  

</body>
</html>
