<!DOCTYPE html>



  


<html class="theme-next mist use-motion" lang="zh-Hans">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">








<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css">




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css">


  <meta name="keywords" content="cancer,">





  <link rel="alternate" href="/atom.xml" title="诸子百家" type="application/atom+xml">




  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.2">






<meta name="description" content="关于变异检测，目前有多个程序可供使用，但很难说哪个程序更好，需要用实验的手段来验证。目前，TCGA采取4款软件，varscan，MuTect，MuSE，SomaticSniper。在这个教程中，我们会陆续介绍这几款主流的程序的使用方法。 一、Varscan123456789101112$ samtools mpileup -d 1000 -B -q 1 -f /mnt/d/ncbi/hg19/BW">
<meta name="keywords" content="cancer">
<meta property="og:type" content="article">
<meta property="og:title" content="肿瘤基因组分析教程：五、检测变异">
<meta property="og:url" content="http://uteric.github.io/WES/cancer_seq5/index.html">
<meta property="og:site_name" content="诸子百家">
<meta property="og:description" content="关于变异检测，目前有多个程序可供使用，但很难说哪个程序更好，需要用实验的手段来验证。目前，TCGA采取4款软件，varscan，MuTect，MuSE，SomaticSniper。在这个教程中，我们会陆续介绍这几款主流的程序的使用方法。 一、Varscan123456789101112$ samtools mpileup -d 1000 -B -q 1 -f /mnt/d/ncbi/hg19/BW">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="http://uteric.github.io/images/WES/cancer19.png">
<meta property="og:updated_time" content="2020-11-23T20:25:14.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="肿瘤基因组分析教程：五、检测变异">
<meta name="twitter:description" content="关于变异检测，目前有多个程序可供使用，但很难说哪个程序更好，需要用实验的手段来验证。目前，TCGA采取4款软件，varscan，MuTect，MuSE，SomaticSniper。在这个教程中，我们会陆续介绍这几款主流的程序的使用方法。 一、Varscan123456789101112$ samtools mpileup -d 1000 -B -q 1 -f /mnt/d/ncbi/hg19/BW">
<meta name="twitter:image" content="http://uteric.github.io/images/WES/cancer19.png">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '9e233f3fa2f200ca831529de7ba10253',
      indexName: 'uteric',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://uteric.github.io/WES/cancer_seq5/">





  <title>肿瘤基因组分析教程：五、检测变异 | 诸子百家</title>
  




<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-104081724-1', 'auto');
ga('send', 'pageview');

</script>











</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <meta name="baidu-site-verification" content="bfQpbnNmEL">

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">诸子百家</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br>
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br>
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br>
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br>
            
            关于
          </a>
        </li>
      

      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br>
            
            搜索
          </a>
        </li>
      
    </ul>
  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
<div class="local-search-header clearfix">
<span class="search-icon">
<i class="fa fa-search"></i>
</span>
<span class="popup-btn-close">
<i class="fa fa-times-circle"></i>
</span>
<div class="local-search-input-wrapper">
<input autocomplete="off" placeholder="搜索..." spellcheck="false" type="text" id="local-search-input">
</div>
</div>
<div id="local-search-result"></div>
</div>


    </div>
  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://uteric.github.io/WES/cancer_seq5/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="括囊无誉">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="诸子百家">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">肿瘤基因组分析教程：五、检测变异</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-11-23T14:25:14-06:00">
                2020-11-23
              </time>
            

            

            
          </span>

          
            <span class="post-category">
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/编程语言/" itemprop="url" rel="index">
                    <span itemprop="name">编程语言</span>
                  </a>
                </span>

                
                
              
            </span>
          


          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/WES/cancer_seq5/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count" data-disqus-identifier="WES/cancer_seq5/" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          

          
            <span class="post-meta-divider">|</span>
            <span class="page-pv"><i class="fa fa-file-o"></i> 热度
            <span class="busuanzi-value" id="busuanzi_value_page_pv"></span>°C
            </span>
          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <p>关于变异检测，目前有多个程序可供使用，但很难说哪个程序更好，需要用实验的手段来验证。目前，TCGA采取4款软件，varscan，MuTect，MuSE，SomaticSniper。在这个教程中，我们会陆续介绍这几款主流的程序的使用方法。</p>
<h2 id="一、Varscan"><a href="#一、Varscan" class="headerlink" title="一、Varscan"></a>一、Varscan</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">$ samtools mpileup -d 1000 -B -q 1 -f /mnt/d/ncbi/hg19/BWAIndex/hg19.fa /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.bam &gt; /mnt/d/cancer/variant/normal-tumor.mpileup</span><br><span class="line"></span><br><span class="line"><span class="comment"># 关于生成mpileup文件，varscan推荐生成normal-tumor.mpileup，同时也有助于缩小空间占用</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># -d, --max-depth INT     max per-file depth; avoids excessive memory usage [8000]</span></span><br><span class="line"><span class="comment"># -B, --no-BAQ            disable BAQ (per-Base Alignment Quality)</span></span><br><span class="line"><span class="comment"># -q, --min-MQ INT        skip alignments with mapQ smaller than INT [0]</span></span><br><span class="line"><span class="comment"># -f, --fasta-ref FILE    faidx indexed reference sequence file</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># mpileup文件很大</span></span><br><span class="line">total 9.7G</span><br><span class="line">-rwxrwxrwx 1 eric eric 9.7G Nov 24 11:15 normal-tumor.mpileup</span><br></pre></td></tr></table></figure>
<p>现在，可以使用进行VARIANT CALLING</p>
<p>varscan是基于启发式的算法，并通过对一组样本进行统计检验来检测变异，可用于检测somatic/germline variants或CNV</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">***NON-COMMERCIAL VERSION***</span><br><span class="line"></span><br><span class="line">USAGE: java -jar VarScan.jar [COMMAND] [OPTIONS]</span><br><span class="line"></span><br><span class="line">COMMANDS:</span><br><span class="line">        pileup2snp              Identify SNPs from a pileup file</span><br><span class="line">        pileup2indel            Identify indels a pileup file</span><br><span class="line">        pileup2cns              Call consensus and variants from a pileup file</span><br><span class="line">        mpileup2snp             Identify SNPs from an mpileup file</span><br><span class="line">        mpileup2indel           Identify indels an mpileup file</span><br><span class="line">        mpileup2cns             Call consensus and variants from an mpileup file</span><br><span class="line"></span><br><span class="line">        somatic                 Call germline/somatic variants from tumor-normal pileups</span><br><span class="line">        copynumber              Determine relative tumor copy number from tumor-normal pileups</span><br><span class="line">        readcounts              Obtain <span class="built_in">read</span> counts <span class="keyword">for</span> a list of variants from a pileup file</span><br><span class="line"></span><br><span class="line">        filter                  Filter SNPs by coverage, frequency, p-value, etc.</span><br><span class="line">        somaticFilter           Filter somatic variants <span class="keyword">for</span> clusters/indels</span><br><span class="line">        fpfilter                Apply the <span class="literal">false</span>-positive filter</span><br><span class="line"></span><br><span class="line">        processSomatic          Isolate Germline/LOH/Somatic calls from output</span><br><span class="line">        copyCaller              GC-adjust and process copy number changes from VarScan copynumber output</span><br><span class="line">        compare                 Compare two lists of positions/variants</span><br><span class="line">        <span class="built_in">limit</span>                   Restrict pileup/snps/indels to ROI positions</span><br></pre></td></tr></table></figure>
<p>对于somatic/germline variants，可以使用varscan somatic命令</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br></pre></td><td class="code"><pre><span class="line">USAGE: VarScan somatic [normal_pileup] [tumor_pileup] [Opt: output] OPTIONS</span><br><span class="line">        normal_pileup - The SAMtools pileup file <span class="keyword">for</span> Normal</span><br><span class="line">        tumor_pileup - The SAMtools pileup file <span class="keyword">for</span> Tumor</span><br><span class="line">        output - Output base name <span class="keyword">for</span> SNP and indel output</span><br><span class="line"></span><br><span class="line">OPTIONS:</span><br><span class="line">        --output-snp - Output file <span class="keyword">for</span> SNP calls [output.snp]</span><br><span class="line">        --output-indel - Output file <span class="keyword">for</span> indel calls [output.indel]</span><br><span class="line">        --min-coverage - Minimum coverage <span class="keyword">in</span> normal and tumor to call variant [8]</span><br><span class="line">        --min-coverage-normal - Minimum coverage <span class="keyword">in</span> normal to call somatic [8]</span><br><span class="line">        --min-coverage-tumor - Minimum coverage <span class="keyword">in</span> tumor to call somatic [6]</span><br><span class="line">        --min-var-freq - Minimum variant frequency to call a heterozygote [0.20]</span><br><span class="line">        --min-freq-for-hom      Minimum frequency to call homozygote [0.75]</span><br><span class="line">        --normal-purity - Estimated purity (non-tumor content) of normal sample [1.00]</span><br><span class="line">        --tumor-purity - Estimated purity (tumor content) of tumor sample [1.00]</span><br><span class="line">        --p-value - P-value threshold to call a heterozygote [0.99]</span><br><span class="line">        --somatic-p-value - P-value threshold to call a somatic site [0.05]</span><br><span class="line">        --strand-filter - If <span class="built_in">set</span> to 1, removes variants with &gt;90% strand bias [0]</span><br><span class="line">        --validation - If <span class="built_in">set</span> to 1, outputs all compared positions even <span class="keyword">if</span> non-variant</span><br><span class="line">        --output-vcf - If <span class="built_in">set</span> to 1, output VCF instead of VarScan native format</span><br><span class="line"></span><br><span class="line"><span class="comment"># 在本例中，使用如下参数：</span></span><br><span class="line">$ $ varscan somatic /mnt/d/cancer/variant/normal-tumor.mpileup /mnt/d/cancer/variant/varscan/varscan --mpileup 1 --output-vcf 1 --strand-filter 1 --somatic-p-value 0.001</span><br><span class="line"></span><br><span class="line">Min coverage:   8x <span class="keyword">for</span> Normal, 6x <span class="keyword">for</span> Tumor</span><br><span class="line">Min reads2:     2</span><br><span class="line">Min strands2:   1</span><br><span class="line">Min var freq:   0.2</span><br><span class="line">Min freq <span class="keyword">for</span> hom:       0.75</span><br><span class="line">Normal purity:  1.0</span><br><span class="line">Tumor purity:   1.0</span><br><span class="line">Min avg qual:   15</span><br><span class="line">P-value thresh: 0.99</span><br><span class="line">Somatic p-value:        0.001</span><br><span class="line">Reading input from /mnt/d/cancer/variant/normal-tumor.mpileup</span><br><span class="line">Reading mpileup input...</span><br><span class="line">87538592 positions <span class="keyword">in</span> mpileup file</span><br><span class="line">43808558 had sufficient coverage <span class="keyword">for</span> comparison</span><br><span class="line">43766686 were called Reference</span><br><span class="line">0 were mixed SNP-indel calls and filtered</span><br><span class="line">0 were removed by the strand filter</span><br><span class="line">41675 were called Germline</span><br><span class="line">16 were called LOH</span><br><span class="line">181 were called Somatic</span><br><span class="line">0 were called Unknown</span><br><span class="line">0 were called Variant</span><br><span class="line"></span><br><span class="line"><span class="comment"># 在mpileup文件中有87538592个位置，其中43808558个位置有足够的测序深度进行对比，43766686个位置是Reference，41675个是Germline，16个LOH，181个是Somatic。我们来看一下生成的VCF文件的内容，前面已经介绍过对于VCF文件的操作</span></span><br><span class="line"></span><br><span class="line">total 5.9M</span><br><span class="line">-rwxrwxrwx 1 eric eric 305K Nov 24 12:18 varscan.indel.vcf</span><br><span class="line">-rwxrwxrwx 1 eric eric 5.6M Nov 24 12:18 varscan.snp.vcf</span><br><span class="line"></span><br><span class="line">$ wc -l *</span><br><span class="line">1974 varscan.indel.vcf</span><br><span class="line">37579 varscan.snp.vcf</span><br><span class="line">39553 total</span><br><span class="line"></span><br><span class="line">$ grep -v <span class="string">'^#'</span> varscan.indel.vcf | wc -l</span><br><span class="line">1956</span><br><span class="line">$ grep -v <span class="string">'^#'</span> varscan.snp.vcf | wc -l</span><br><span class="line">37561</span><br><span class="line"></span><br><span class="line">$ grep -v <span class="string">'^#'</span> varscan.indel.vcf | head</span><br><span class="line">chr1    866511  .       C       CCCCT   .       PASS    DP=22;SS=1;SSC=3;GPV=5.3885E-2;SPV=4.4962E-1    GT:GQ:DP:RD:AD:FREQ:DP4 0/0:.:9:8:1:11.11%:3,5,0,1       0/1:.:13:10:3:23.08%:4,6,2,1</span><br><span class="line">chr1    948846  .       T       TA      .       PASS    DP=26;SS=1;SSC=4;GPV=1.0968E-2;SPV=3.202E-1     GT:GQ:DP:RD:AD:FREQ:DP4 0/0:.:13:10:2:16.67%:10,0,2,0    0/1:.:13:8:4:33.33%:8,0,4,0</span><br><span class="line">chr1    1110987 .       T       TC      .       PASS    DP=109;SS=1;SSC=10;GPV=5.5481E-19;SPV=9.2343E-2 GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:50:31:19:38%:27,4,18,1     0/1:.:59:28:31:52.54%:28,0,31,0</span><br><span class="line">chr1    1177918 .       CT      C       .       PASS    DP=81;SS=1;SSC=2;GPV=5.0311E-18;SPV=5.4934E-1   GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:40:18:22:55%:18,0,21,1     0/1:.:41:18:23:56.1%:18,0,23,0</span><br><span class="line">chr1    1276930 .       GCA     G       .       PASS    DP=111;SS=1;SSC=2;GPV=3.1066E-64;SPV=5.8559E-1  GT:GQ:DP:RD:AD:FREQ:DP4 1/1:.:46:0:46:100%:0,0,46,0      1/1:.:65:1:64:98.46%:1,0,64,0</span><br><span class="line">chr1    1276973 .       G       GACAC   .       PASS    DP=344;SS=1;SSC=2;GPV=2.5102E-171;SPV=5.1236E-1 GT:GQ:DP:RD:AD:FREQ:DP4 1/1:.:147:6:133:95.68%:6,0,125,8 1/1:.:197:7:178:96.22%:6,1,164,14</span><br><span class="line">chr1    1289367 .       CTG     C       .       PASS    DP=45;SS=1;SSC=6;GPV=1.6658E-22;SPV=2.3087E-1   GT:GQ:DP:RD:AD:FREQ:DP4 1/1:.:28:3:25:89.29%:0,3,19,6    1/1:.:17:0:17:100%:0,0,15,2</span><br><span class="line">chr1    1647893 .       C       CTTTCTT .       PASS    DP=148;SS=1;SSC=1;GPV=1.5376E-11;SPV=6.6262E-1  GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:61:46:14:23.33%:27,19,6,8  0/1:.:87:68:19:21.84%:41,27,11,8</span><br><span class="line">chr1    1887091 .       CG      C       .       PASS    DP=83;SS=1;SSC=9;GPV=5.6303E-12;SPV=1.1544E-1   GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:34:24:10:29.41%:18,6,8,2   0/1:.:49:27:22:44.9%:26,1,21,1</span><br><span class="line">chr1    1887111 .       GC      G       .       PASS    DP=101;SS=1;SSC=5;GPV=3.6902E-16;SPV=2.8695E-1  GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:42:26:16:38.1%:20,6,12,4   0/1:.:59:32:27:45.76%:27,5,23,4</span><br><span class="line"></span><br><span class="line">$ grep -v <span class="string">'^#'</span> varscan.snp.vcf | head</span><br><span class="line">chr1    69511   .       A       G       .       PASS    DP=60;SS=1;SSC=0;GPV=1.035E-35;SPV=1E0  GT:GQ:DP:RD:AD:FREQ:DP4         1/1:.:36:0:36:100%:0,0,33,3      1/1:.:24:0:24:100%:0,0,22,2</span><br><span class="line">chr1    762273  .       G       A       .       PASS    DP=66;SS=1;SSC=0;GPV=2.6498E-39;SPV=1E0 GT:GQ:DP:RD:AD:FREQ:DP4         1/1:.:37:0:37:100%:0,0,12,25     1/1:.:29:0:29:100%:0,0,14,15</span><br><span class="line">chr1    777232  .       C       T       .       PASS    DP=28;SS=1;SSC=0;GPV=2.3037E-6;SPV=9.2937E-1    GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:18:7:11:61.11%:7,0,9,2     0/1:.:10:6:4:40%:6,0,3,1</span><br><span class="line">chr1    792480  .       C       T       .       PASS    DP=27;SS=1;SSC=0;GPV=5.1363E-16;SPV=1E0 GT:GQ:DP:RD:AD:FREQ:DP4         1/1:.:16:0:16:100%:0,0,6,10      1/1:.:11:0:11:100%:0,0,5,6</span><br><span class="line">chr1    866319  .       G       A       .       PASS    DP=49;SS=1;SSC=0;GPV=3.925E-29;SPV=1E0  GT:GQ:DP:RD:AD:FREQ:DP4         1/1:.:21:0:21:100%:0,0,21,0      1/1:.:28:0:28:100%:0,0,28,0</span><br><span class="line">chr1    876499  .       A       G       .       PASS    DP=31;SS=1;SSC=0;GPV=2.1486E-18;SPV=1E0 GT:GQ:DP:RD:AD:FREQ:DP4         1/1:.:19:0:19:100%:0,0,18,1      1/1:.:12:0:12:100%:0,0,10,2</span><br><span class="line">chr1    880238  .       A       G       .       PASS    DP=78;SS=1;SSC=0;GPV=1.7165E-46;SPV=1E0 GT:GQ:DP:RD:AD:FREQ:DP4         1/1:.:56:0:56:100%:0,0,37,19     1/1:.:22:0:22:100%:0,0,14,8</span><br><span class="line">chr1    881627  .       G       A       .       PASS    DP=57;SS=1;SSC=0;GPV=6.4572E-34;SPV=1E0 GT:GQ:DP:RD:AD:FREQ:DP4         1/1:.:22:0:22:100%:0,0,19,3      1/1:.:35:0:35:100%:0,0,32,3</span><br><span class="line">chr1    883625  .       A       G       .       PASS    DP=87;SS=1;SSC=0;GPV=6.9142E-52;SPV=1E0 GT:GQ:DP:RD:AD:FREQ:DP4         1/1:.:32:0:32:100%:0,0,26,6      1/1:.:55:0:55:100%:0,0,42,13</span><br><span class="line">chr1    884101  .       A       C       .       PASS    DP=117;SS=1;SSC=0;GPV=7.359E-48;SPV=9.3085E-1   GT:GQ:DP:RD:AD:FREQ:DP4 1/1:.:61:7:54:88.52%:4,3,21,33   1/1:.:56:11:45:80.36%:9,2,20,25</span><br><span class="line"></span><br><span class="line">$ grep -v <span class="string">'^#'</span> varscan.snp.vcf | cut -f 1 | sort | uniq -c</span><br><span class="line">   3781 chr1</span><br><span class="line">   1640 chr10</span><br><span class="line">   2783 chr11</span><br><span class="line">   1844 chr12</span><br><span class="line">    476 chr13</span><br><span class="line">   1289 chr14</span><br><span class="line">   1210 chr15</span><br><span class="line">   2049 chr16</span><br><span class="line">   2432 chr17</span><br><span class="line">    489 chr18</span><br><span class="line">   3112 chr19</span><br><span class="line">   2336 chr2</span><br><span class="line">   1091 chr20</span><br><span class="line">    572 chr21</span><br><span class="line">   1110 chr22</span><br><span class="line">   1720 chr3</span><br><span class="line">   1239 chr4</span><br><span class="line">   1509 chr5</span><br><span class="line">   1492 chr6</span><br><span class="line">   2162 chr7</span><br><span class="line">   1263 chr8</span><br><span class="line">   1635 chr9</span><br><span class="line">    322 chrX</span><br><span class="line">      5 chrY</span><br><span class="line">$ grep -v <span class="string">'^#'</span> varscan.indel.vcf | cut -f 1 | sort | uniq -c</span><br><span class="line">189 chr1</span><br><span class="line">     69 chr10</span><br><span class="line">    137 chr11</span><br><span class="line">    116 chr12</span><br><span class="line">     20 chr13</span><br><span class="line">     76 chr14</span><br><span class="line">     67 chr15</span><br><span class="line">    100 chr16</span><br><span class="line">    162 chr17</span><br><span class="line">     25 chr18</span><br><span class="line">    140 chr19</span><br><span class="line">    102 chr2</span><br><span class="line">     56 chr20</span><br><span class="line">     40 chr21</span><br><span class="line">     58 chr22</span><br><span class="line">     90 chr3</span><br><span class="line">     57 chr4</span><br><span class="line">     77 chr5</span><br><span class="line">     91 chr6</span><br><span class="line">    115 chr7</span><br><span class="line">     59 chr8</span><br><span class="line">     83 chr9</span><br><span class="line">     27 chrX</span><br><span class="line"></span><br><span class="line"><span class="comment"># chr1没有INDEL，但有最多的SNP</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 只抓取SOMATIC</span></span><br><span class="line">$ grep SOMATIC varscan.indel.vcf</span><br><span class="line"><span class="comment">##INFO=&lt;ID=SOMATIC,Number=0,Type=Flag,Description="Indicates if record is a somatic mutation"&gt;</span></span><br><span class="line">chr1    67154730        .       CAAA    C       .       PASS    DP=21;SOMATIC;SS=2;SSC=9;GPV=1E0;SPV=1.1905E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:9:3:0:0%:3,0,0,0  0/1:.:12:2:4:66.67%:2,0,4,0</span><br><span class="line">chr1    116940496       .       TG      T       .       PASS    DP=30;SOMATIC;SS=2;SSC=9;GPV=1E0;SPV=1.0345E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:20:20:0:0%:18,2,0,0       0/1:.:10:8:2:20%:8,0,2,0</span><br><span class="line">chr11   134054705       .       TAC     T       .       PASS    DP=17;SOMATIC;SS=2;SSC=7;GPV=1E0;SPV=1.9231E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:9:7:0:0%:4,3,0,0  0/1:.:8:4:2:33.33%:4,0,2,0</span><br><span class="line">chr12   104732900       .       AT      A       .       PASS    DP=18;SOMATIC;SS=2;SSC=7;GPV=1E0;SPV=1.8301E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:10:10:0:0%:10,0,0,0       0/1:.:8:6:2:25%:6,0,2,0</span><br><span class="line">chr12   122064778       .       CA      C       .       PASS    DP=116;SOMATIC;SS=2;SSC=7;GPV=1E0;SPV=1.8182E-1 GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:43:6:0:0%:4,2,0,0 0/1:.:73:3:2:40%:2,1,1,1</span><br><span class="line">chr12   132445122       .       AT      A       .       PASS    DP=22;SOMATIC;SS=2;SSC=9;GPV=1E0;SPV=1.2121E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:14:14:0:0%:11,3,0,0       0/1:.:8:6:2:25%:6,0,2,0</span><br><span class="line">chr14   50623681        .       GTA     G       .       PASS    DP=18;SOMATIC;SS=2;SSC=9;GPV=1E0;SPV=1.0294E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:9:9:0:0%:9,0,0,0  0/1:.:9:6:3:33.33%:6,0,3,0</span><br><span class="line">chr15   50782434        .       GT      G       .       PASS    DP=20;SOMATIC;SS=2;SSC=6;GPV=1E0;SPV=2.3684E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:10:10:0:0%:10,0,0,0       0/1:.:10:8:2:20%:8,0,2,0</span><br><span class="line">chr17   54981571        .       CA      C       .       PASS    DP=24;SOMATIC;SS=2;SSC=5;GPV=1E0;SPV=2.7368E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:9:7:0:0%:7,0,0,0  0/1:.:15:11:3:21.43%:11,0,3,0</span><br><span class="line">chr19   6469042 .       CT      C       .       PASS    DP=35;SOMATIC;SS=2;SSC=10;GPV=1E0;SPV=9.4721E-2 GT:GQ:DP:RD:AD:FREQ:DP4 0/0:.:15:14:0:0%:10,4,0,0        0/1:.:20:15:4:21.05%:8,7,3,1</span><br><span class="line">chr19   56000261        .       G       GTACA   .       PASS    DP=16;SOMATIC;SS=2;SSC=6;GPV=1E0;SPV=2.3333E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:8:8:0:0%:5,3,0,0  0/1:.:8:6:2:25%:4,2,1,1</span><br><span class="line">chr2    116593690       .       TGTGA   T       .       PASS    DP=20;SOMATIC;SS=2;SSC=8;GPV=1E0;SPV=1.4737E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:12:12:0:0%:11,1,0,0       0/1:.:8:6:2:25%:6,0,2,0</span><br><span class="line">chr3    10183699        .       CG      C       .       PASS    DP=87;SOMATIC;SS=2;SSC=37;GPV=1E0;SPV=1.9809E-4 GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:31:31:0:0%:30,1,0,0       0/1:.:56:39:17:30.36%:37,2,17,0</span><br><span class="line">chr3    186393073       .       GGTGT   G       .       PASS    DP=16;SOMATIC;SS=2;SSC=5;GPV=1E0;SPV=2.6667E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:8:7:0:0%:3,4,0,0  0/1:.:8:6:2:25%:4,2,1,1</span><br><span class="line">chr4    4190507 .       C       CTTGTTAGTCA     .       PASS    DP=24;SOMATIC;SS=2;SSC=10;GPV=1E0;SPV=9.4203E-2 GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:10:10:0:0%:10,0,0,0       0/1:.:14:10:4:28.57%:10,0,4,0</span><br><span class="line">chr4    4190511 .       G       GTTA    .       PASS    DP=27;SOMATIC;SS=2;SSC=16;GPV=1E0;SPV=2.19E-2   GT:GQ:DP:RD:AD:FREQ:DP4 0/0:.:10:10:0:0%:10,0,0,0        0/1:.:17:10:7:41.18%:9,1,4,3</span><br><span class="line">chr4    88048315        .       GC      G       .       PASS    DP=27;SOMATIC;SS=2;SSC=12;GPV=1E0;SPV=5.641E-2  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:16:16:0:0%:0,16,0,0       0/1:.:11:8:3:27.27%:4,4,2,1</span><br><span class="line">chr7    56087291        .       AC      A       .       PASS    DP=24;SOMATIC;SS=2;SSC=6;GPV=1E0;SPV=2.248E-1   GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:9:9:0:0%:9,0,0,0  0/1:.:15:12:3:20%:11,1,3,0</span><br><span class="line">chr7    114293360       .       AT      A       .       PASS    DP=27;SOMATIC;SS=2;SSC=9;GPV=1E0;SPV=1.2E-1     GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:17:16:0:0%:16,0,0,0       0/1:.:10:7:2:22.22%:7,0,2,0</span><br><span class="line">chr8    145738581       .       TGGG    T       .       PASS    DP=22;SOMATIC;SS=2;SSC=7;GPV=1E0;SPV=1.8947E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:13:11:0:0%:10,1,0,0       0/1:.:9:7:2:22.22%:7,0,1,1</span><br><span class="line">chr9    125932152       .       CAT     C       .       PASS    DP=24;SOMATIC;SS=2;SSC=6;GPV=1E0;SPV=2.1053E-1  GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:15:10:0:0%:9,1,0,0        0/1:.:9:7:2:22.22%:7,0,2,0</span><br><span class="line">chrX    47107214        .       TC      T       .       PASS    DP=165;SOMATIC;SS=2;SSC=98;GPV=1E0;SPV=1.4237E-10      GT:GQ:DP:RD:AD:FREQ:DP4   0/0:.:59:58:0:0%:37,21,0,0      0/1:.:106:63:43:40.57%:42,21,30,13</span><br><span class="line">chrX    100666907       .       CT      C       .       PASS    DP=21;SOMATIC;SS=2;SSC=7;GPV=1E0;SPV=1.75E-1    GT:GQ:DP:RD:AD:FREQ:DP4  0/0:.:12:9:0:0%:9,0,0,0 0/1:.:9:5:2:28.57%:5,0,2,0</span><br><span class="line"></span><br><span class="line">$ grep SOMATIC varscan.indel.vcf | wc -l</span><br><span class="line">24</span><br><span class="line">$ grep <span class="string">'SS=2'</span> varscan.indel.vcf |wc -l <span class="comment"># 体会一下这两个参数的区别，都是指SOMATIC</span></span><br><span class="line">23</span><br><span class="line"></span><br><span class="line">$ grep SOMATIC varscan.snp.vcf  | wc -l</span><br><span class="line">108</span><br><span class="line">$ grep <span class="string">'SS=2'</span> varscan.snp.vcf | wc -l</span><br><span class="line">107</span><br><span class="line"></span><br><span class="line"><span class="comment"># 同样，可以看一下Germline有多少</span></span><br><span class="line">$ grep <span class="string">'SS=1'</span> varscan.indel.vcf | wc -l</span><br><span class="line">1933</span><br><span class="line">$ grep <span class="string">'SS=1'</span> varscan.snp.vcf | wc -l</span><br><span class="line">37442</span><br><span class="line"></span><br><span class="line"><span class="comment"># 看一下LOH</span></span><br><span class="line">$ grep <span class="string">'SS=3'</span> varscan.indel.vcf | wc -l</span><br><span class="line">0</span><br><span class="line">$ grep <span class="string">'SS=3'</span> varscan.snp.vcf | wc -l</span><br><span class="line">12</span><br><span class="line">$ grep <span class="string">'SS=3'</span> varscan.snp.vcf | head</span><br><span class="line">chr1    45218895        .       A       T       .       PASS    DP=52;SS=3;SSC=35;GPV=1E0;SPV=3.0463E-4 GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:28:20:8:28.57%:17,3,8,0    1/1:.:24:5:19:79.17%:5,0,18,1</span><br><span class="line">chr11   72316344        .       G       T       .       PASS    DP=63;SS=3;SSC=32;GPV=1E0;SPV=5.3356E-4 GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:44:25:19:43.18%:9,16,7,12  1/1:.:19:2:17:89.47%:1,1,10,7</span><br><span class="line">chr14   105935577       .       C       A       .       PASS    DP=75;SS=3;SSC=31;GPV=1E0;SPV=6.9789E-4 GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:33:24:8:25%:24,0,0,8       0/0:.:42:42:0:0%:38,4,0,0</span><br><span class="line">chr16   71956593        .       T       G       .       PASS    DP=151;SS=3;SSC=66;GPV=1E0;SPV=2.2714E-7        GT:GQ:DP:RD:AD:FREQ:DP4  0/1:.:74:37:37:50%:10,27,8,29   0/0:.:77:64:8:11.11%:17,47,3,5</span><br><span class="line">chr16   71956596        .       A       G       .       PASS    DP=146;SS=3;SSC=66;GPV=1E0;SPV=2.2714E-7        GT:GQ:DP:RD:AD:FREQ:DP4  0/1:.:74:37:37:50%:10,27,8,29   0/0:.:72:64:8:11.11%:17,47,3,5</span><br><span class="line">chr16   71956597        .       A       T       .       PASS    DP=146;SS=3;SSC=37;GPV=1E0;SPV=1.6498E-4        GT:GQ:DP:RD:AD:FREQ:DP4  0/1:.:74:51:23:31.08%:24,27,8,15        0/0:.:72:67:5:6.94%:20,47,3,2</span><br><span class="line">chr3    19959756        .       G       A       .       PASS    DP=90;SS=3;SSC=37;GPV=1E0;SPV=1.6442E-4 GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:61:25:36:59.02%:19,6,28,8  0/0:.:29:24:5:17.24%:18,6,2,3</span><br><span class="line">chr3    48732390        .       A       G       .       PASS    DP=31;SS=3;SSC=34;GPV=1E0;SPV=3.5706E-4 GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:19:7:12:63.16%:4,3,11,1    0/0:.:12:12:0:0%:8,4,0,0</span><br><span class="line">chr3    52474480        .       A       G       .       PASS    DP=69;SS=3;SSC=39;GPV=1E0;SPV=1.2352E-4 GT:GQ:DP:RD:AD:FREQ:DP4 0/1:.:37:21:16:43.24%:20,1,16,0  1/1:.:32:4:28:87.5%:4,0,28,0</span><br><span class="line">chr3    52860936        .       C       T       .       PASS    DP=153;SS=3;SSC=46;GPV=1E0;SPV=2.3344E-5        GT:GQ:DP:RD:AD:FREQ:DP4  0/1:.:91:43:48:52.75%:25,18,26,22       0/0:.:62:50:12:19.35%:35,15,5,7</span><br><span class="line"><span class="comment"># 这些VCF文件我们后面要作图进行可视化，先将VCF文件压缩并生成索引</span></span><br><span class="line">$ bgzip varscan.indel.vcf</span><br><span class="line">$ bgzip varscan.snp.vcf</span><br><span class="line">$ bcftools index -t varscan.indel.vcf.gz</span><br><span class="line">$ bcftools index -t varscan.snp.vcf.gz</span><br><span class="line"><span class="comment"># 查看</span></span><br><span class="line">total 1.6M</span><br><span class="line">-rwxrwxrwx 1 eric eric  86K Nov 24 22:33 varscan.indel.vcf.gz</span><br><span class="line">-rwxrwxrwx 1 eric eric  26K Nov 24 22:34 varscan.indel.vcf.gz.tbi</span><br><span class="line">-rwxrwxrwx 1 eric eric 1.4M Nov 24 22:33 varscan.snp.vcf.gz</span><br><span class="line">-rwxrwxrwx 1 eric eric 175K Nov 24 22:34 varscan.snp.vcf.gz.tbi</span><br></pre></td></tr></table></figure>
<h2 id="二、MuTect2"><a href="#二、MuTect2" class="headerlink" title="二、MuTect2"></a>二、MuTect2</h2><p>MuTect2使用起来非常繁琐，软件更新十分频繁，因此流程变来变去，不推荐使用。详情敬请移步官方教程：<br><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531132" target="_blank" rel="noopener">Call somatic mutations using GATK4 Mutect2</a>；<br><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360051306691-Mutect2" target="_blank" rel="noopener">Mutect2</a>;<br><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-" target="_blank" rel="noopener">Somatic short variant discovery (SNVs + Indels)</a></p>
<p><img src="/images/WES/cancer19.png" alt></p>
<h3 id="2-1-创建PON-panel-of-normals"><a href="#2-1-创建PON-panel-of-normals" class="headerlink" title="2.1 创建PON (panel of normals)"></a>2.1 创建PON (panel of normals)</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">$ gatk Mutect2 -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -I /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam --max-mnp-distance 0 -O /mnt/d/cancer/variant/mutect/normal.vcf.gz</span><br><span class="line"></span><br><span class="line"><span class="comment"># 运行完成后会生成3个文件</span></span><br><span class="line"><span class="comment"># Step 1: Run Mutect2 in tumor-only mode for each normal sample</span></span><br><span class="line">total 5.2M</span><br><span class="line">-rwxrwxrwx 1 eric eric 4.8M Nov 26 22:01 normal.vcf.gz</span><br><span class="line">-rwxrwxrwx 1 eric eric   37 Nov 26 22:01 normal.vcf.gz.stats</span><br><span class="line">-rwxrwxrwx 1 eric eric 414K Nov 26 22:01 normal.vcf.gz.tbi</span><br><span class="line"></span><br><span class="line"><span class="comment"># Step 2: Create a GenomicsDB from the normal Mutect2 calls</span></span><br><span class="line">$ gatk GenomicsDBImport -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -L /mnt/d/ncbi/wxs/agilent/sureSelect_All_Exon_V5/S04380110_Padded.bed --genomicsdb-workspace-path pon_db -V /mnt/d/cancer/variant/mutect/normal.vcf.gz</span><br><span class="line"><span class="comment"># 注意，此处"--genomicsdb-workspace-path pon_db"是为GenomicsDB提供工作目录，即在运行目录下会新建一个pon_db目录</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Step 3: Combine the normal calls using CreateSomaticPanelOfNormals</span></span><br><span class="line">$ gatk CreateSomaticPanelOfNormals -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa --germline-resource /mnt/d/ncbi/wxs/gnomAD/af-only-gnomad.hg19.vcf.gz -V gendb://pon_db -O /mnt/d/cancer/variant/mutect/pon.vcf.gz</span><br><span class="line"><span class="comment"># 现在，得到了PON文件，这个文件将作为下一步的输入文件</span></span><br></pre></td></tr></table></figure>
<p>第三步中，需要用到一个af-only-gnomad.hg19.vcf.gz文件，如果使用hg38参考基因组，这个文件可以从gatk的resource of bundle上下载，但是对于hg19，并不提供这个文件。但是，我们可以找到gnomAD的完全版，8.3G，我们可以使用bcftools来提取AF信息，并生成新的文件，过程如下：</p>
<p>下载地址: <a href="https://hgdownload.soe.ucsc.edu/gbdb/hg19/gnomAD/vcf/" target="_blank" rel="noopener">hg19</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 下载后的文件</span></span><br><span class="line">-rwxrwxrwx 1 eric eric 8.3G Nov 27 09:45 gnomad.exomes.r2.0.2.sites.vcf.gz</span><br><span class="line">-rwxrwxrwx 1 eric eric 902K Nov 27 10:13 gnomad.exomes.r2.0.2.sites.vcf.gz.tbi</span><br><span class="line"></span><br><span class="line"><span class="comment"># 提取AF，压缩并索引</span></span><br><span class="line">$ bcftools annotate -x ^INFO/AF gnomad.exomes.r2.0.2.sites.vcf.gz &gt; af-only-gnomad.hg19.vcf</span><br><span class="line">$ bgzip af-only-gnomad.hg19.vcf</span><br><span class="line">$ bcftools index -t af-only-gnomad.hg19.vcf.gz</span><br><span class="line"></span><br><span class="line">total 212M</span><br><span class="line">-rwxrwxrwx 1 eric eric 211M Nov 27 22:13 af-only-gnomad.hg19.vcf.gz</span><br><span class="line">-rwxrwxrwx 1 eric eric 534K Nov 27 22:15 af-only-gnomad.hg19.vcf.gz.tbi</span><br></pre></td></tr></table></figure>
<h3 id="2-2-获取原始的候选变异"><a href="#2-2-获取原始的候选变异" class="headerlink" title="2.2 获取原始的候选变异"></a>2.2 获取原始的候选变异</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk Mutect2 -R /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -I /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.bam -I /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam -normal SRR5478492C.bqsr --germline-resource af-only-gnomad.vcf.gz --panel-of-normals pon.vcf.gz -O somatic.vcf.gz</span><br></pre></td></tr></table></figure>
<p><a href="https://hgdownload.soe.ucsc.edu/gbdb/hg19/gnomAD/vcf/" target="_blank" rel="noopener">https://hgdownload.soe.ucsc.edu/gbdb/hg19/gnomAD/vcf/</a></p>
<p><a href="https://console.cloud.google.com/storage/browser/gatk-best-practices/somatic-b37;tab=objects?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&amp;prefix=&amp;forceOnObjectsSortingFiltering=false" target="_blank" rel="noopener">https://console.cloud.google.com/storage/browser/gatk-best-practices/somatic-b37;tab=objects?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&amp;prefix=&amp;forceOnObjectsSortingFiltering=false</a><br><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890711-GRCh37-hg19-b37-humanG1Kv37-Human-Reference-Discrepancies" target="_blank" rel="noopener">https://gatk.broadinstitute.org/hc/en-us/articles/360035890711-GRCh37-hg19-b37-humanG1Kv37-Human-Reference-Discrepancies</a><br><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19" target="_blank" rel="noopener">https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19</a><br><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-" target="_blank" rel="noopener">https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-</a><br><a href="https://blog.csdn.net/viancheng/article/details/106332585" target="_blank" rel="noopener">https://blog.csdn.net/viancheng/article/details/106332585</a></p>
<p>拷贝数变异</p>
<p><a href="https://sequenza-utils.readthedocs.io/en/latest/guide.html#generate-gc-reference-file" target="_blank" rel="noopener">sequenza</a></p>
<p>生成GC content文件</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ sequenza-utils gc_wiggle --fasta /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -w 50 -o /mnt/d/ncbi/wxs/sequenza/hg19_gc50.wig.gz</span><br></pre></td></tr></table></figure>
<p>sequenza-utils bam2seqz –normal normal_sample.bam –tumor tumor_sample.bam \<br>    –fasta genome.fa.gz -gc genome_gc50.wig.gz –output sample.seqz.gz</p>
<p>sequenza bam2seqz \<br>-n data/normal.chr5.60Mb.bam \<br>-t data/tumour.chr5.60Mb.bam \<br>–fasta assets/human_g1k_v37.fasta \<br>-gc assets/human_g1k_v37.gc50Base.txt.gz \<br>-C 5:1-60000000 | gzip &gt; stage1.seqz.gz</p>
<p>-n: the normal BAM<br>-t: the tumour BAM<br>–fasta: the reference genome used for mapping (b37 here)<br>-gc: GC content as windows through the genome (pre-generated and downloadable from the Sequenza website)<br>-C: specifies the genomic location to process</p>
<p>$ sequenza-utils bam2seqz –normal /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam -t /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.bam –fasta /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -gc /mnt/d/ncbi/wxs/sequenza/hg19_gc50.wig.gz –output /mnt/d/cancer/cnv/stage1.seqz.gz</p>
<p>$ sequenza-utils bam2seqz –normal /mnt/d/cancer/bqsr/SRR5478492C.bqsr.nodup.bam -t /mnt/d/cancer/bqsr/SRR5478491C.bqsr.nodup.bam –fasta /mnt/d/ncbi/hg19/BWAIndex/hg19.fa -gc /mnt/d/ncbi/wxs/sequenza/hg19_gc50.wig.gz –output /mnt/d/cancer/cnv/stage1.seqz.gz<br>[mpileup] 1 samples in 1 input files<br>[mpileup] 1 samples in 1 input files</p>
<p>total 75M<br>-rwxrwxrwx 1 eric eric  75M Nov 27 22:47 stage1.seqz.gz<br>-rwxrwxrwx 1 eric eric 126K Nov 27 22:47 stage1.seqz.gz.tbi</p>
<p>$ zcat stage1.seqz.gz | head -n 20<br>chromosome      position        base.ref        depth.normal    depth.tumor     depth.ratio     Af      Bf      zygosity.normal GC.percent      good.reads      AB.normal       AB.tumor        tumor.strand<br>chr1    69457   C       13      7       0.538   1.0     0       hom     48      7       C       .       0<br>chr1    69458   T       13      7       0.538   1.0     0       hom     48      7       T       .       0<br>chr1    69459   A       13      7       0.538   1.0     0       hom     48      7       A       .       0<br>chr1    69460   C       14      8       0.571   1.0     0       hom     48      8       C       .       0<br>chr1    69461   A       14      8       0.571   1.0     0       hom     48      8       A       .       0<br>chr1    69462   C       14      8       0.571   1.0     0       hom     48      8       C       .       0<br>chr1    69463   T       14      8       0.571   1.0     0       hom     48      8       T       .       0<br>chr1    69464   A       15      8       0.533   1.0     0       hom     48      8       A       .       0<br>chr1    69465   C       16      8       0.5     1.0     0       hom     48      8       C       .       0<br>chr1    69466   A       17      9       0.529   1.0     0       hom     48      9       A       .       0<br>chr1    69467   C       17      9       0.529   1.0     0       hom     48      9       C       .       0<br>chr1    69468   T       17      9       0.529   1.0     0       hom     48      9       T       .       0<br>chr1    69469   A       16      9       0.562   1.0     0       hom     48      9       A       .       0<br>chr1    69470   C       18      9       0.5     1.0     0       hom     48      9       C       .       0<br>chr1    69471   A       20      9       0.45    1.0     0       hom     48      9       A       .       0<br>chr1    69472   A       22      9       0.409   1.0     0       hom     48      9       A       .       0<br>chr1    69473   T       22      11      0.5     1.0     0       hom     48      11      T       .       0<br>chr1    69474   T       22      14      0.636   1.0     0       hom     48      14      T       .       0<br>chr1    69475   A       22      14      0.636   1.0     0       hom     48      14      A       .       0</p>
<p>sequenza-utils seqz-binning \<br>-w 200 \<br>-s stage1.seqz.gz | gzip &gt; stage2.seqz.gz</p>
<p>sequenza-utils seqz_binning -w 200 -s stage1.seqz.gz -o stage2.seqz.gz</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">&gt; library(sequenza)</span><br><span class="line">&gt; data.file &lt;- <span class="string">"./stage2.seqz.gz"</span></span><br><span class="line">&gt; seqzdata &lt;- sequenza.extract(data.file)</span><br><span class="line">Collecting GC information . <span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">Processing chr1:</span><br><span class="line">   1 variant calls.</span><br><span class="line">   2 copy-number segments.</span><br><span class="line">   2255 heterozygous positions.</span><br><span class="line">   107658 homozygous positions.</span><br><span class="line">Processing chr10:</span><br><span class="line">   1 variant calls.</span><br><span class="line">   2 copy-number segments.</span><br><span class="line">   1005 heterozygous positions.</span><br><span class="line">   38009 homozygous positions.</span><br><span class="line">Processing chr11:</span><br><span class="line">   2 variant calls.</span><br><span class="line">   3 copy-number segments.</span><br><span class="line">   1744 heterozygous positions.</span><br><span class="line">   71520 homozygous positions.</span><br></pre></td></tr></table></figure>

      
    </div>

<div>


<ul class="post-copyright">
<li class="post-copyright-author">
<strong>本文作者：</strong>括囊无誉
</li>
<li class="post-copyright-link">
<strong>本文链接：</strong>
<a href="/WES/cancer_seq5/" title="肿瘤基因组分析教程：五、检测变异">WES/cancer_seq5/</a>
</li>
<li class="post-copyright-license">
<strong>版权声明： </strong>
本博客所有文章均为原创作品，转载请注明出处！
</li>
</ul>

</div>

    
    
    


<div>

<div style="text-align:center;color: #ccc;font-size:14px;">------ 本文结束 ------</div>

</div>

    

    
      <div>
        <div style="padding: 10px 0; margin: 20px auto; width: 90%; text-align: center;">
  <div>坚持原创文章分享，您的支持将鼓励我继续创作！</div>
  <button id="rewardButton" disable="enable" onclick="var qr = document.getElementById('QR'); if (qr.style.display === 'none') {qr.style.display='block';} else {qr.style.display='none'}">
    <span>Donate</span>
  </button>
  <div id="QR" style="display: none;">

    
      <div id="wechat" style="display: inline-block">
        <img id="wechat_qr" src="/images/wechatpay.jpg" alt="括囊无誉 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
    

    

    

  </div>
</div>

      </div>
    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/cancer/" rel="tag"># cancer</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/WES/cancer_seq4/" rel="next" title="肿瘤基因组分析教程：四、BAM文件预处理">
                <i class="fa fa-chevron-left"></i> 肿瘤基因组分析教程：四、BAM文件预处理
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/WES/wxs_interval/" rel="prev" title="外显子测序分析中使用区间文件INTERVAL">
                外显子测序分析中使用区间文件INTERVAL <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a>
        </noscript>
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image" src="/images/avatar.png" alt="括囊无誉">
          <p class="site-author-name" itemprop="name">括囊无誉</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives/">
                <span class="site-state-item-count">234</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">11</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">226</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        
          <div class="feed-link motion-element">
            <a href="/atom.xml" rel="alternate">
              <i class="fa fa-rss"></i>
              RSS
            </a>
          </div>
        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="http://www.jianshu.com/u/75d3f95b058e" target="_blank" title="简书">
                  
                    <i class="fa fa-fw fa-http://www.jianshu.com/u/75d3f95b058e"></i>
                  
                    
                      简书
                    
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#一、Varscan"><span class="nav-number">1.</span> <span class="nav-text">一、Varscan</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#二、MuTect2"><span class="nav-number">2.</span> <span class="nav-text">二、MuTect2</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#2-1-创建PON-panel-of-normals"><span class="nav-number">2.1.</span> <span class="nav-text">2.1 创建PON (panel of normals)</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-2-获取原始的候选变异"><span class="nav-number">2.2.</span> <span class="nav-text">2.2 获取原始的候选变异</span></a></li></ol></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">
  
  &copy;  2017 - 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">括囊无誉</span>
</div>


<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


        
<div class="busuanzi-count">
  <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="site-uv">
      <i class="fa fa-user"></i> 访问人数
      <span class="busuanzi-value" id="busuanzi_value_site_uv"></span>
      
    </span>
  

  
    <span class="site-pv">
      <i class="fa fa-eye"></i> 总访问量
      <span class="busuanzi-value" id="busuanzi_value_site_pv"></span>
      次
    </span>
  
</div>


        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>



  


  

    
      <script id="dsq-count-scr" src="https://uteric-1.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'http://uteric.github.io/WES/cancer_seq5/';
          this.page.identifier = 'WES/cancer_seq5/';
          this.page.title = '肿瘤基因组分析教程：五、检测变异';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://uteric-1.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script>
    

  




	





  








  

  <script type="text/javascript">
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url);
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x" /></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x" /></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  
<script>
(function(){
    var bp = document.createElement('script');
    var curProtocol = window.location.protocol.split(':')[0];
    if (curProtocol === 'https') {
        bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';        
    }
    else {
        bp.src = 'http://push.zhanzhang.baidu.com/push.js';
    }
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(bp, s);
})();
</script>


  

  

  

  

</body>
</html>
