<!DOCTYPE html>



  


<html class="theme-next mist use-motion" lang="zh-Hans">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">








<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css">




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css">


  <meta name="keywords" content="MAFTOOLS,">





  <link rel="alternate" href="/atom.xml" title="诸子百家" type="application/atom+xml">




  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.2">






<meta name="description" content="我们以TCGA-KIRC数据为例，来介绍肿瘤突变数据与临床数据的下载方法、预处理及可视化 一、数据下载与处理在TCGA-GDC官网，在CASES选项下，选择CTGA-KIRC，这里共包含537个CASE，回到FILES选项下，选择Data Category[simple nucleotide variation]+Data Type[Masked Somatic Mutation]+Workflo">
<meta name="keywords" content="MAFTOOLS">
<meta property="og:type" content="article">
<meta property="og:title" content="使用MAFTOOLS分析肿瘤变异数据">
<meta property="og:url" content="http://uteric.github.io/R/maltools/index.html">
<meta property="og:site_name" content="诸子百家">
<meta property="og:description" content="我们以TCGA-KIRC数据为例，来介绍肿瘤突变数据与临床数据的下载方法、预处理及可视化 一、数据下载与处理在TCGA-GDC官网，在CASES选项下，选择CTGA-KIRC，这里共包含537个CASE，回到FILES选项下，选择Data Category[simple nucleotide variation]+Data Type[Masked Somatic Mutation]+Workflo">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc01.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc02.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc03.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc04.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc05.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc06.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc07.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc08.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc09.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc10.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc11.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc12.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc13.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc14.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc15.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc16.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc17.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc18.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc19.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc20.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc21.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc22.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc23.png">
<meta property="og:image" content="http://uteric.github.io/images/R/kirc24.png">
<meta property="og:updated_time" content="2020-10-27T18:57:39.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="使用MAFTOOLS分析肿瘤变异数据">
<meta name="twitter:description" content="我们以TCGA-KIRC数据为例，来介绍肿瘤突变数据与临床数据的下载方法、预处理及可视化 一、数据下载与处理在TCGA-GDC官网，在CASES选项下，选择CTGA-KIRC，这里共包含537个CASE，回到FILES选项下，选择Data Category[simple nucleotide variation]+Data Type[Masked Somatic Mutation]+Workflo">
<meta name="twitter:image" content="http://uteric.github.io/images/R/kirc01.png">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '9e233f3fa2f200ca831529de7ba10253',
      indexName: 'uteric',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://uteric.github.io/R/maltools/">





  <title>使用MAFTOOLS分析肿瘤变异数据 | 诸子百家</title>
  




<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-104081724-1', 'auto');
ga('send', 'pageview');

</script>











</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <meta name="baidu-site-verification" content="bfQpbnNmEL">

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">诸子百家</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br>
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br>
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br>
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br>
            
            关于
          </a>
        </li>
      

      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br>
            
            搜索
          </a>
        </li>
      
    </ul>
  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
<div class="local-search-header clearfix">
<span class="search-icon">
<i class="fa fa-search"></i>
</span>
<span class="popup-btn-close">
<i class="fa fa-times-circle"></i>
</span>
<div class="local-search-input-wrapper">
<input autocomplete="off" placeholder="搜索..." spellcheck="false" type="text" id="local-search-input">
</div>
</div>
<div id="local-search-result"></div>
</div>


    </div>
  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://uteric.github.io/R/maltools/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="括囊无誉">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="诸子百家">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">使用MAFTOOLS分析肿瘤变异数据</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-10-27T13:57:39-05:00">
                2020-10-27
              </time>
            

            

            
          </span>

          
            <span class="post-category">
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/编程语言/" itemprop="url" rel="index">
                    <span itemprop="name">编程语言</span>
                  </a>
                </span>

                
                
              
            </span>
          


          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/R/maltools/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count" data-disqus-identifier="R/maltools/" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          

          
            <span class="post-meta-divider">|</span>
            <span class="page-pv"><i class="fa fa-file-o"></i> 热度
            <span class="busuanzi-value" id="busuanzi_value_page_pv"></span>°C
            </span>
          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <p>我们以TCGA-KIRC数据为例，来介绍肿瘤突变数据与临床数据的下载方法、预处理及可视化</p>
<h1 id="一、数据下载与处理"><a href="#一、数据下载与处理" class="headerlink" title="一、数据下载与处理"></a>一、数据下载与处理</h1><p>在<a href="https://portal.gdc.cancer.gov/" target="_blank" rel="noopener">TCGA-GDC</a>官网，在CASES选项下，选择CTGA-KIRC，这里共包含537个CASE，回到FILES选项下，选择Data Category[simple nucleotide variation]+Data Type[Masked Somatic Mutation]+Workflow Type[MuTect2 Variant Aggregation and Masking]，至此，数据选择就完成了，点击Add Files to Cart，回到Cart，FILE=1，CASES=339，FILE SIZE=7.12 MB。下载Clinical/TSV文件，下载Download/Cart文件，前一个文件包含临床数据，后一个文件包含突变数据。</p>
<p>将这两个文件移至R的工作目录之下，使用命令对这两个tat.gz文件进行解压，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">tar -zxvf clinical.cart.2020-10-27.tar.gz <span class="comment"># 解压后，得到三个文件：clinical.tsv，family_history.tsv，exposure.tsv</span></span><br><span class="line">tar -zxvf gdc_download_20201027_190832.301544.tar.gz <span class="comment"># 解压后，得到maf.gz文件，再使用gzip -d filename.gz进行解压，得到我们需要的.maf文件，重命名为mutect.maf</span></span><br></pre></td></tr></table></figure>
<p>在下一步操作之前，我们需要对下载好的数据作预处理，以使临床数据与突变数据能够通过Tumor_Sample_Barcode联系起来，分两步进行：</p>
<p>第一、打开clinical.tsv，将第二列“case_submitter_id”修改为“Tumor_Sample_Barcode”；<br>第二、打开mutect.maf，将“Tumor_Sample_Barcode”这一列中的内容修改为与clinical.tsv中的”Tumor_Sample_Barcode”一致，即只保留前12个字符，在EXCEL中可以新建一列，并使用[LEFT(text,12)]这个函数进行操作，操作完成后，使用CTRL+SHIFT+DOWN向下选择，再使用CTRL+D快速填充，填充完毕，将新列命名为”Tumor_Sample_Barcode”，并将旧的”Tumor_Sample_Barcode”删除，保存退出。</p>
<p>至此，数据下载与预处理就完成了，后面的过程会比较简单。</p>
<h1 id="二、读取MAF文件并进行统计"><a href="#二、读取MAF文件并进行统计" class="headerlink" title="二、读取MAF文件并进行统计"></a>二、读取MAF文件并进行统计</h1><h2 id="2-1-MAF文件读取"><a href="#2-1-MAF文件读取" class="headerlink" title="2.1 MAF文件读取"></a>2.1 MAF文件读取</h2><p>下面的操作都将在RStudio/R中进行，代码如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">&gt; setwd(<span class="string">"D:/mutation/KIRC/mutect"</span>) <span class="comment"># 切换至工作目录下</span></span><br><span class="line">&gt; library(maftools) <span class="comment"># 加载maftools</span></span><br><span class="line">&gt; kirc &lt;- read.maf(maf=<span class="string">"mutect.maf"</span>, clinicalData=<span class="string">"clinical.tsv"</span>)</span><br></pre></td></tr></table></figure>
<p>文件读取过程</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">-Reading</span><br><span class="line">|--------------------------------------------------|</span><br><span class="line">|==================================================|</span><br><span class="line">|--------------------------------------------------|</span><br><span class="line">|==================================================|</span><br><span class="line">-Validating</span><br><span class="line">--Removed 1021876 duplicated variants</span><br><span class="line">--Found  1  variants with no Gene Symbols</span><br><span class="line">--Annotating them as <span class="string">'UnknownGene'</span> <span class="keyword">for</span> convenience</span><br><span class="line">--Non MAF specific values <span class="keyword">in</span> Variant_Classification column:</span><br><span class="line">  </span><br><span class="line">--Non MAF specific values <span class="keyword">in</span> Variant_Type column:</span><br><span class="line">  </span><br><span class="line">-Silent variants: 8384 </span><br><span class="line">-Summarizing</span><br><span class="line">--Mutiple centers found</span><br><span class="line">BI;BCM--Possible FLAGS among top ten genes:</span><br><span class="line">  TTN</span><br><span class="line">  MUC16</span><br><span class="line">  HMCN1</span><br><span class="line">-Processing clinical data</span><br><span class="line">--Annotation missing <span class="keyword">for</span> below samples <span class="keyword">in</span> MAF:</span><br><span class="line">  </span><br><span class="line">-Finished <span class="keyword">in</span> 39.7s elapsed (16.7s cpu)</span><br></pre></td></tr></table></figure>
<h2 id="2-2-MAF文件统计"><a href="#2-2-MAF文件统计" class="headerlink" title="2.2 MAF文件统计"></a>2.2 MAF文件统计</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">&gt; kirc <span class="comment"># 查看MAF文件基本信息</span></span><br><span class="line">An object of class  MAF </span><br><span class="line">                        ID summary   Mean Median</span><br><span class="line"> 1:             NCBI_Build  GRCh38     NA     NA</span><br><span class="line"> 2:                 Center  BI;BCM     NA     NA</span><br><span class="line"> 3:                Samples     337     NA     NA</span><br><span class="line"> 4:                 nGenes    9444     NA     NA</span><br><span class="line"> 5:        Frame_Shift_Del    1732  5.155      4</span><br><span class="line"> 6:        Frame_Shift_Ins    1201  3.574      1</span><br><span class="line"> 7:           In_Frame_Del     238  0.708      0</span><br><span class="line"> 8:           In_Frame_Ins     350  1.042      0</span><br><span class="line"> 9:      Missense_Mutation   12997 38.682     36</span><br><span class="line">10:      Nonsense_Mutation    1259  3.747      2</span><br><span class="line">11:       Nonstop_Mutation      18  0.054      0</span><br><span class="line">12:            Splice_Site     490  1.458      1</span><br><span class="line">13: Translation_Start_Site      25  0.074      0</span><br><span class="line">14:                  total   18310 54.494     47</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br></pre></td><td class="code"><pre><span class="line">&gt; getSampleSummary(kirc) <span class="comment"># 对每一个样本的情况进行统计（336个样本）</span></span><br><span class="line">     Tumor_Sample_Barcode Frame_Shift_Del Frame_Shift_Ins In_Frame_Del</span><br><span class="line">  1:         TCGA-B8-4143              39             721           13</span><br><span class="line">  2:         TCGA-B0-5098              88              25            5</span><br><span class="line">  3:         TCGA-A3-A8OV               7               3            1</span><br><span class="line">  4:         TCGA-CJ-4920              16               4            0</span><br><span class="line">  5:         TCGA-CZ-5468               4               2            0</span><br><span class="line"> ---                                                                  </span><br><span class="line">332:         TCGA-B0-5117               0               1            0</span><br><span class="line">333:         TCGA-BP-4760               1               0            0</span><br><span class="line">334:         TCGA-DV-5576               0               0            0</span><br><span class="line">335:         TCGA-B8-5546               0               0            0</span><br><span class="line">336:         TCGA-DV-A4VZ               1               0            0</span><br><span class="line">     In_Frame_Ins Missense_Mutation Nonsense_Mutation Nonstop_Mutation</span><br><span class="line">  1:          305               201               320                1</span><br><span class="line">  2:            1               390                20                0</span><br><span class="line">  3:            0                99                 7                0</span><br><span class="line">  4:            0                87                 6                0</span><br><span class="line">  5:            0                91                 4                0</span><br><span class="line"> ---                                                                  </span><br><span class="line">332:            0                 7                 0                0</span><br><span class="line">333:            0                 7                 0                0</span><br><span class="line">334:            0                 7                 1                0</span><br><span class="line">335:            0                 7                 0                0</span><br><span class="line">336:            0                 4                 0                0</span><br><span class="line">     Splice_Site Translation_Start_Site total</span><br><span class="line">  1:          11                      0  1611</span><br><span class="line">  2:          20                      1   550</span><br><span class="line">  3:           3                      0   120</span><br><span class="line">  4:           2                      2   117</span><br><span class="line">  5:           1                      0   102</span><br><span class="line"> ---                                         </span><br><span class="line">332:           0                      0     8</span><br><span class="line">333:           0                      0     8</span><br><span class="line">334:           0                      0     8</span><br><span class="line">335:           0                      0     7</span><br><span class="line">336:           0                      0     5</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br></pre></td><td class="code"><pre><span class="line">&gt; getGeneSummary(kirc) <span class="comment"># 对每一个基因进行统计（9444个基因）</span></span><br><span class="line">      Hugo_Symbol Frame_Shift_Del Frame_Shift_Ins In_Frame_Del In_Frame_Ins</span><br><span class="line">   1:         VHL              41              22            2            0</span><br><span class="line">   2:       PBRM1              64              12            1            1</span><br><span class="line">   3:         TTN               4               4            0            1</span><br><span class="line">   4:       SETD2              12               2            0            0</span><br><span class="line">   5:        BAP1              13               1            0            0</span><br><span class="line">  ---                                                                      </span><br><span class="line">9440:       ZWINT               0               0            0            0</span><br><span class="line">9441:        ZXDA               0               0            0            0</span><br><span class="line">9442:        ZXDB               0               0            0            0</span><br><span class="line">9443:        ZXDC               0               0            0            0</span><br><span class="line">9444:         ZYX               0               0            0            0</span><br><span class="line">      Missense_Mutation Nonsense_Mutation Nonstop_Mutation Splice_Site</span><br><span class="line">   1:                60                27                1          16</span><br><span class="line">   2:                19                38                0          13</span><br><span class="line">   3:                58                 9                0           1</span><br><span class="line">   4:                10                18                0           4</span><br><span class="line">   5:                 8                 8                0           4</span><br><span class="line">  ---                                                                 </span><br><span class="line">9440:                 1                 0                0           0</span><br><span class="line">9441:                 1                 0                0           0</span><br><span class="line">9442:                 0                 1                0           0</span><br><span class="line">9443:                 0                 1                0           0</span><br><span class="line">9444:                 1                 0                0           0</span><br><span class="line">      Translation_Start_Site total MutatedSamples AlteredSamples</span><br><span class="line">   1:                      0   169            166            166</span><br><span class="line">   2:                      0   148            141            141</span><br><span class="line">   3:                      0    77             59             59</span><br><span class="line">   4:                      0    46             42             42</span><br><span class="line">   5:                      3    37             35             35</span><br><span class="line">  ---                                                           </span><br><span class="line">9440:                      0     1              1              1</span><br><span class="line">9441:                      0     1              1              1</span><br><span class="line">9442:                      0     1              1              1</span><br><span class="line">9443:                      0     1              1              1</span><br><span class="line">9444:                      0     1              1              1</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; getClinicalData(kirc) <span class="comment"># 临床信息，即clinical.tsv文件中的内容，相同的case_id会输出一行，内容很长，不展示</span></span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br></pre></td><td class="code"><pre><span class="line">&gt; getFields(kirc) <span class="comment"># 显示MAF文件中的所有字段，第16列就是我们预先修改过的"Tumor_Sample_Barcode"</span></span><br><span class="line">  [1] <span class="string">"Hugo_Symbol"</span>                   <span class="string">"Entrez_Gene_Id"</span>               </span><br><span class="line">  [3] <span class="string">"Center"</span>                        <span class="string">"NCBI_Build"</span>                   </span><br><span class="line">  [5] <span class="string">"Chromosome"</span>                    <span class="string">"Start_Position"</span>               </span><br><span class="line">  [7] <span class="string">"End_Position"</span>                  <span class="string">"Strand"</span>                       </span><br><span class="line">  [9] <span class="string">"Variant_Classification"</span>        <span class="string">"Variant_Type"</span>                 </span><br><span class="line"> [11] <span class="string">"Reference_Allele"</span>              <span class="string">"Tumor_Seq_Allele1"</span>            </span><br><span class="line"> [13] <span class="string">"Tumor_Seq_Allele2"</span>             <span class="string">"dbSNP_RS"</span>                     </span><br><span class="line"> [15] <span class="string">"dbSNP_Val_Status"</span>              <span class="string">"Tumor_Sample_Barcode"</span>         </span><br><span class="line"> [17] <span class="string">"Matched_Norm_Sample_Barcode"</span>   <span class="string">"Match_Norm_Seq_Allele1"</span>       </span><br><span class="line"> [19] <span class="string">"Match_Norm_Seq_Allele2"</span>        <span class="string">"Tumor_Validation_Allele1"</span>     </span><br><span class="line"> [21] <span class="string">"Tumor_Validation_Allele2"</span>      <span class="string">"Match_Norm_Validation_Allele1"</span></span><br><span class="line"> [23] <span class="string">"Match_Norm_Validation_Allele2"</span> <span class="string">"Verification_Status"</span>          </span><br><span class="line"> [25] <span class="string">"Validation_Status"</span>             <span class="string">"Mutation_Status"</span>              </span><br><span class="line"> [27] <span class="string">"Sequencing_Phase"</span>              <span class="string">"Sequence_Source"</span>              </span><br><span class="line"> [29] <span class="string">"Validation_Method"</span>             <span class="string">"Score"</span>                        </span><br><span class="line"> [31] <span class="string">"BAM_File"</span>                      <span class="string">"Sequencer"</span>                    </span><br><span class="line"> [33] <span class="string">"Tumor_Sample_UUID"</span>             <span class="string">"Matched_Norm_Sample_UUID"</span>     </span><br><span class="line"> [35] <span class="string">"HGVSc"</span>                         <span class="string">"HGVSp"</span>                        </span><br><span class="line"> [37] <span class="string">"HGVSp_Short"</span>                   <span class="string">"Transcript_ID"</span>                </span><br><span class="line"> [39] <span class="string">"Exon_Number"</span>                   <span class="string">"t_depth"</span>                      </span><br><span class="line"> [41] <span class="string">"t_ref_count"</span>                   <span class="string">"t_alt_count"</span>                  </span><br><span class="line"> [43] <span class="string">"n_depth"</span>                       <span class="string">"n_ref_count"</span>                  </span><br><span class="line"> [45] <span class="string">"n_alt_count"</span>                   <span class="string">"all_effects"</span>                  </span><br><span class="line"> [47] <span class="string">"Allele"</span>                        <span class="string">"Gene"</span>                         </span><br><span class="line"> [49] <span class="string">"Feature"</span>                       <span class="string">"Feature_type"</span>                 </span><br><span class="line"> [51] <span class="string">"One_Consequence"</span>               <span class="string">"Consequence"</span>                  </span><br><span class="line"> [53] <span class="string">"cDNA_position"</span>                 <span class="string">"CDS_position"</span>                 </span><br><span class="line"> [55] <span class="string">"Protein_position"</span>              <span class="string">"Amino_acids"</span>                  </span><br><span class="line"> [57] <span class="string">"Codons"</span>                        <span class="string">"Existing_variation"</span>           </span><br><span class="line"> [59] <span class="string">"ALLELE_NUM"</span>                    <span class="string">"DISTANCE"</span>                     </span><br><span class="line"> [61] <span class="string">"TRANSCRIPT_STRAND"</span>             <span class="string">"SYMBOL"</span>                       </span><br><span class="line"> [63] <span class="string">"SYMBOL_SOURCE"</span>                 <span class="string">"HGNC_ID"</span>                      </span><br><span class="line"> [65] <span class="string">"BIOTYPE"</span>                       <span class="string">"CANONICAL"</span>                    </span><br><span class="line"> [67] <span class="string">"CCDS"</span>                          <span class="string">"ENSP"</span>                         </span><br><span class="line"> [69] <span class="string">"SWISSPROT"</span>                     <span class="string">"TREMBL"</span>                       </span><br><span class="line"> [71] <span class="string">"UNIPARC"</span>                       <span class="string">"RefSeq"</span>                       </span><br><span class="line"> [73] <span class="string">"SIFT"</span>                          <span class="string">"PolyPhen"</span>                     </span><br><span class="line"> [75] <span class="string">"EXON"</span>                          <span class="string">"INTRON"</span>                       </span><br><span class="line"> [77] <span class="string">"DOMAINS"</span>                       <span class="string">"GMAF"</span>                         </span><br><span class="line"> [79] <span class="string">"AFR_MAF"</span>                       <span class="string">"AMR_MAF"</span>                      </span><br><span class="line"> [81] <span class="string">"ASN_MAF"</span>                       <span class="string">"EAS_MAF"</span>                      </span><br><span class="line"> [83] <span class="string">"EUR_MAF"</span>                       <span class="string">"SAS_MAF"</span>                      </span><br><span class="line"> [85] <span class="string">"AA_MAF"</span>                        <span class="string">"EA_MAF"</span>                       </span><br><span class="line"> [87] <span class="string">"CLIN_SIG"</span>                      <span class="string">"SOMATIC"</span>                      </span><br><span class="line"> [89] <span class="string">"PUBMED"</span>                        <span class="string">"MOTIF_NAME"</span>                   </span><br><span class="line"> [91] <span class="string">"MOTIF_POS"</span>                     <span class="string">"HIGH_INF_POS"</span>                 </span><br><span class="line"> [93] <span class="string">"MOTIF_SCORE_CHANGE"</span>            <span class="string">"IMPACT"</span>                       </span><br><span class="line"> [95] <span class="string">"PICK"</span>                          <span class="string">"VARIANT_CLASS"</span>                </span><br><span class="line"> [97] <span class="string">"TSL"</span>                           <span class="string">"HGVS_OFFSET"</span>                  </span><br><span class="line"> [99] <span class="string">"PHENO"</span>                         <span class="string">"MINIMISED"</span>                    </span><br><span class="line">[101] <span class="string">"ExAC_AF"</span>                       <span class="string">"ExAC_AF_Adj"</span>                  </span><br><span class="line">[103] <span class="string">"ExAC_AF_AFR"</span>                   <span class="string">"ExAC_AF_AMR"</span>                  </span><br><span class="line">[105] <span class="string">"ExAC_AF_EAS"</span>                   <span class="string">"ExAC_AF_FIN"</span>                  </span><br><span class="line">[107] <span class="string">"ExAC_AF_NFE"</span>                   <span class="string">"ExAC_AF_OTH"</span>                  </span><br><span class="line">[109] <span class="string">"ExAC_AF_SAS"</span>                   <span class="string">"GENE_PHENO"</span>                   </span><br><span class="line">[111] <span class="string">"FILTER"</span>                        <span class="string">"CONTEXT"</span>                      </span><br><span class="line">[113] <span class="string">"src_vcf_id"</span>                    <span class="string">"tumor_bam_uuid"</span>               </span><br><span class="line">[115] <span class="string">"normal_bam_uuid"</span>               <span class="string">"case_id"</span>                      </span><br><span class="line">[117] <span class="string">"GDC_FILTER"</span>                    <span class="string">"COSMIC"</span>                       </span><br><span class="line">[119] <span class="string">"MC3_Overlap"</span>                   <span class="string">"GDC_Validation_Status"</span></span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; write.mafSummary(maf=kirc, basename=<span class="string">"kirc"</span>) <span class="comment"># 输出一个kirc_sampleSummary.txt文件，内容就是getSampleSummary()输出的内容</span></span><br></pre></td></tr></table></figure>
<h1 id="三、突变的可视化"><a href="#三、突变的可视化" class="headerlink" title="三、突变的可视化"></a>三、突变的可视化</h1><h2 id="3-1-MAF文件汇总统计图"><a href="#3-1-MAF文件汇总统计图" class="headerlink" title="3.1 MAF文件汇总统计图"></a>3.1 MAF文件汇总统计图</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; plotmafSummary(maf=kirc, rmOutlier=FALSE, addStat=<span class="string">"median"</span>, dashboard=TRUE, titvRaw = FALSE)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc01.png" alt></p>
<h2 id="3-2-ONCOPLOT（瀑布图）"><a href="#3-2-ONCOPLOT（瀑布图）" class="headerlink" title="3.2 ONCOPLOT（瀑布图）"></a>3.2 ONCOPLOT（瀑布图）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">oncoplot(maf=kirc,top=10,fontSize = 0.6,showTumorSampleBarcodes = F)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc02.png" alt></p>
<p>若对某些基因感兴趣，则使用genes参数指定基因，如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">oncoplot(maf=kirc,genes=c(<span class="string">"VHL"</span>,<span class="string">"PBRM1"</span>,<span class="string">"SETD2"</span>,<span class="string">"BAP1"</span>,<span class="string">"KDM5C"</span>,<span class="string">"TP53"</span>,<span class="string">"ATM"</span>,<span class="string">"ARID1A"</span>,<span class="string">"PTEN"</span>,<span class="string">"TSC1"</span>,<span class="string">"NF2"</span>,<span class="string">"CDKN2A"</span>,<span class="string">"CDKN2B"</span>),fontSize = 0.6,showTumorSampleBarcodes = F)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc03.png" alt></p>
<h2 id="3-3-基因突变共存与排斥"><a href="#3-3-基因突变共存与排斥" class="headerlink" title="3.3 基因突变共存与排斥"></a>3.3 基因突变共存与排斥</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br></pre></td><td class="code"><pre><span class="line">&gt; somaticInteractions(maf=kirc,top=10,pvalue = c(0.01,0.05),fontSize = 0.5)</span><br><span class="line">    gene1 gene2       pValue oddsRatio  00 11  01  10              Event</span><br><span class="line"> 1:   VHL PBRM1 0.0006345551 2.1499701 115 85  56  81       Co_Occurence</span><br><span class="line"> 2: SETD2 PBRM1 0.0069684692 2.5362176 180 26 115  16       Co_Occurence</span><br><span class="line"> 3:  BAP1 MUC16 0.0177253442 3.6761737 286  6  16  29       Co_Occurence</span><br><span class="line"> 4:   TTN HMCN1 0.0231208832 3.2520719 267  7  11  52       Co_Occurence</span><br><span class="line"> 5:  MTOR   TTN 0.0811830227 2.3527658 263  7  52  15       Co_Occurence</span><br><span class="line"> 6: DNAH9  BAP1 0.0869868888 2.8556913 289  4  31  13       Co_Occurence</span><br><span class="line"> 7: DNAH9  MTOR 0.0902657408 3.3752125 301  3  19  14       Co_Occurence</span><br><span class="line"> 8:   TTN  BAP1 0.0963286074 2.0600380 253 10  25  49       Co_Occurence</span><br><span class="line"> 9:   TTN SETD2 0.1287443528 1.8222905 247 11  31  48       Co_Occurence</span><br><span class="line">10: MUC16  MTOR 0.1649934265 2.4505615 296  3  19  19       Co_Occurence</span><br><span class="line">11: DNAH9   TTN 0.1924226420 2.0471964 266  5  54  12       Co_Occurence</span><br><span class="line">12:  BAP1 PBRM1 0.2090438361 0.6072762 172 11 130  24 Mutually_Exclusive</span><br><span class="line">13: MUC16   TTN 0.2426284078 1.8496958 262  6  53  16       Co_Occurence</span><br><span class="line">14: KDM5C  BAP1 0.2453372474 0.0000000 282  0  35  20 Mutually_Exclusive</span><br><span class="line">15: DNAH9 SETD2 0.2460022133 2.2760862 282  4  38  13       Co_Occurence</span><br><span class="line">16: HMCN1 SETD2 0.2586292953 2.1068377 281  4  38  14       Co_Occurence</span><br><span class="line">17:   VHL SETD2 0.3232858924 1.4350657 153 24  18 142       Co_Occurence</span><br><span class="line">18:  MTOR HMCN1 0.3314867116 1.8644128 299  2  16  20       Co_Occurence</span><br><span class="line">19: MUC16 HMCN1 0.3314867116 1.8644128 299  2  16  20       Co_Occurence</span><br><span class="line">20: KDM5C   TTN 0.3647966906 1.6207908 263  5  54  15       Co_Occurence</span><br><span class="line">21: MUC16 KDM5C 0.3814379116 1.6470297 297  2  18  20       Co_Occurence</span><br><span class="line">22: SETD2  BAP1 0.4145364481 1.5265571 266  6  29  36       Co_Occurence</span><br><span class="line">23: HMCN1  BAP1 0.4153981852 1.7899322 287  3  32  15       Co_Occurence</span><br><span class="line">24: PBRM1 DNAH9 0.4501215062 1.5999432 188  9   8 132       Co_Occurence</span><br><span class="line">25:  BAP1  MTOR 0.4880885198 1.3948288 283  3  19  32       Co_Occurence</span><br><span class="line">26: KDM5C SETD2 0.4881372857 0.3550657 276  1  41  19 Mutually_Exclusive</span><br><span class="line">27: DNAH9   VHL 0.6206385050 0.7095260 161  7 159  10 Mutually_Exclusive</span><br><span class="line">28:   TTN PBRM1 0.6649326479 0.8666566 160 23 118  36 Mutually_Exclusive</span><br><span class="line">29: MUC16 SETD2 0.7451002458 1.1170630 276  3  39  19       Co_Occurence</span><br><span class="line">30: HMCN1   VHL 0.8097726647 0.8156840 161  8 158  10 Mutually_Exclusive</span><br><span class="line">31: PBRM1 HMCN1 0.8113574779 1.1184198 186  8  10 133       Co_Occurence</span><br><span class="line">32:  MTOR PBRM1 0.8239686926 1.1699269 184 10 131  12       Co_Occurence</span><br><span class="line">33: MUC16 PBRM1 0.8239686926 1.1699269 184 10 131  12       Co_Occurence</span><br><span class="line">34:  MTOR   VHL 0.8264941510 0.8497706 159 10 156  12 Mutually_Exclusive</span><br><span class="line">35:   TTN   VHL 1.0000000000 0.9949054 141 29 137  30 Mutually_Exclusive</span><br><span class="line">36:  BAP1   VHL 1.0000000000 0.9698796 153 17 149  18 Mutually_Exclusive</span><br><span class="line">37: MUC16   VHL 1.0000000000 1.0321697 160 11 155  11       Co_Occurence</span><br><span class="line">38: KDM5C   VHL 1.0000000000 1.0319633 161 10 156  10       Co_Occurence</span><br><span class="line">39: KDM5C PBRM1 1.0000000000 0.9225070 184  8 133  12 Mutually_Exclusive</span><br><span class="line">40:  MTOR SETD2 1.0000000000 0.6881771 275  2  40  20 Mutually_Exclusive</span><br><span class="line">41: KDM5C  MTOR 1.0000000000 0.7424513 296  1  21  19 Mutually_Exclusive</span><br><span class="line">42: DNAH9 MUC16 1.0000000000 0.8901653 299  1  21  16 Mutually_Exclusive</span><br><span class="line">43: HMCN1 KDM5C 1.0000000000 0.9289862 300  1  19  17 Mutually_Exclusive</span><br><span class="line">44: DNAH9 KDM5C 1.0000000000 0.9901605 301  1  19  16 Mutually_Exclusive</span><br><span class="line">45: DNAH9 HMCN1 1.0000000000 1.1135821 303  1  17  16       Co_Occurence</span><br><span class="line">    gene1 gene2       pValue oddsRatio  00 11  01  10              Event</span><br><span class="line">            pair event_ratio</span><br><span class="line"> 1:   PBRM1, VHL      85/137</span><br><span class="line"> 2: PBRM1, SETD2      26/131</span><br><span class="line"> 3:  BAP1, MUC16        6/45</span><br><span class="line"> 4:   HMCN1, TTN        7/63</span><br><span class="line"> 5:    MTOR, TTN        7/67</span><br><span class="line"> 6:  BAP1, DNAH9        4/44</span><br><span class="line"> 7:  DNAH9, MTOR        3/33</span><br><span class="line"> 8:    BAP1, TTN       10/74</span><br><span class="line"> 9:   SETD2, TTN       11/79</span><br><span class="line">10:  MTOR, MUC16        3/38</span><br><span class="line">11:   DNAH9, TTN        5/66</span><br><span class="line">12:  BAP1, PBRM1      11/154</span><br><span class="line">13:   MUC16, TTN        6/69</span><br><span class="line">14:  BAP1, KDM5C        0/55</span><br><span class="line">15: DNAH9, SETD2        4/51</span><br><span class="line">16: HMCN1, SETD2        4/52</span><br><span class="line">17:   SETD2, VHL      24/160</span><br><span class="line">18:  HMCN1, MTOR        2/36</span><br><span class="line">19: HMCN1, MUC16        2/36</span><br><span class="line">20:   KDM5C, TTN        5/69</span><br><span class="line">21: KDM5C, MUC16        2/38</span><br><span class="line">22:  BAP1, SETD2        6/65</span><br><span class="line">23:  BAP1, HMCN1        3/47</span><br><span class="line">24: DNAH9, PBRM1       9/140</span><br><span class="line">25:   BAP1, MTOR        3/51</span><br><span class="line">26: KDM5C, SETD2        1/60</span><br><span class="line">27:   DNAH9, VHL       7/169</span><br><span class="line">28:   PBRM1, TTN      23/154</span><br><span class="line">29: MUC16, SETD2        3/58</span><br><span class="line">30:   HMCN1, VHL       8/168</span><br><span class="line">31: HMCN1, PBRM1       8/143</span><br><span class="line">32:  MTOR, PBRM1      10/143</span><br><span class="line">33: MUC16, PBRM1      10/143</span><br><span class="line">34:    MTOR, VHL      10/168</span><br><span class="line">35:     TTN, VHL      29/167</span><br><span class="line">36:    BAP1, VHL      17/167</span><br><span class="line">37:   MUC16, VHL      11/166</span><br><span class="line">38:   KDM5C, VHL      10/166</span><br><span class="line">39: KDM5C, PBRM1       8/145</span><br><span class="line">40:  MTOR, SETD2        2/60</span><br><span class="line">41:  KDM5C, MTOR        1/40</span><br><span class="line">42: DNAH9, MUC16        1/37</span><br><span class="line">43: HMCN1, KDM5C        1/36</span><br><span class="line">44: DNAH9, KDM5C        1/35</span><br><span class="line">45: DNAH9, HMCN1        1/33</span><br><span class="line">            pair event_ratio</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc04.png" alt></p>
<p>同理，也可以指定感兴趣的基因</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br></pre></td><td class="code"><pre><span class="line">&gt; somaticInteractions(maf=kirc,genes=c(<span class="string">"VHL"</span>,<span class="string">"PBRM1"</span>,<span class="string">"SETD2"</span>,<span class="string">"BAP1"</span>,<span class="string">"KDM5C"</span>,<span class="string">"TP53"</span>,<span class="string">"ATM"</span>,<span class="string">"ARID1A"</span>,<span class="string">"PTEN"</span>,<span class="string">"TSC1"</span>,<span class="string">"NF2"</span>,<span class="string">"CDKN2A"</span>,<span class="string">"CDKN2B"</span>),pvalue = c(0.01,0.05),fontSize = 0.5)</span><br><span class="line">     gene1  gene2       pValue  oddsRatio  00 11  01  10              Event</span><br><span class="line"> 1:    VHL  PBRM1 0.0006345551  2.1499701 115 85  56  81       Co_Occurence</span><br><span class="line"> 2:  SETD2  PBRM1 0.0069684692  2.5362176 180 26 115  16       Co_Occurence</span><br><span class="line"> 3:  PBRM1    ATM 0.0358546719  3.2210326 191 11   5 130       Co_Occurence</span><br><span class="line"> 4:    NF2    VHL 0.0607849993  0.0000000 166  0 166   5 Mutually_Exclusive</span><br><span class="line"> 5:   TP53    VHL 0.0673114551  0.1426375 164  1 165   7 Mutually_Exclusive</span><br><span class="line"> 6:    NF2  PBRM1 0.0774205829  0.0000000 191  0 141   5 Mutually_Exclusive</span><br><span class="line"> 7:    VHL   PTEN 0.0826438335  3.1999405 168  9   3 157       Co_Occurence</span><br><span class="line"> 8:    NF2   TP53 0.1138366801 11.3372992 325  1   7   4       Co_Occurence</span><br><span class="line"> 9:   TP53  PBRM1 0.1457726032  0.1935716 189  1 140   7 Mutually_Exclusive</span><br><span class="line">10:    ATM   TSC1 0.1775445900  6.9716044 318  1   3  15       Co_Occurence</span><br><span class="line">11:   PTEN  SETD2 0.1778634395  2.4357280 286  3  39   9       Co_Occurence</span><br><span class="line">12:   BAP1  PBRM1 0.2090438361  0.6072762 172 11 130  24 Mutually_Exclusive</span><br><span class="line">13:  KDM5C   BAP1 0.2453372474  0.0000000 282  0  35  20 Mutually_Exclusive</span><br><span class="line">14:   TP53   PTEN 0.2541098432  4.0970807 318  1  11   7       Co_Occurence</span><br><span class="line">15:    VHL ARID1A 0.2853701129  1.8996157 166  9   5 157       Co_Occurence</span><br><span class="line">16:    ATM    VHL 0.3144443178  1.7598874 165 10 156   6       Co_Occurence</span><br><span class="line">17:    VHL  SETD2 0.3232858924  1.4350657 153 24  18 142       Co_Occurence</span><br><span class="line">18:   TSC1  SETD2 0.4143213860  2.3659439 292  1  41   3       Co_Occurence</span><br><span class="line">19:  SETD2   BAP1 0.4145364481  1.5265571 266  6  29  36       Co_Occurence</span><br><span class="line">20:  KDM5C  SETD2 0.4881372857  0.3550657 276  1  41  19 Mutually_Exclusive</span><br><span class="line">21:  SETD2    NF2 0.4881907608  1.7706936 291  1   4  41       Co_Occurence</span><br><span class="line">22:  KDM5C   PTEN 0.5261326770  1.4621664 306  1  11  19       Co_Occurence</span><br><span class="line">23:   PTEN  PBRM1 0.5665516486  1.4059303 190  6 135   6       Co_Occurence</span><br><span class="line">24: ARID1A  PBRM1 0.5860353120  1.4089576 189  7 134   7       Co_Occurence</span><br><span class="line">25:    ATM  KDM5C 0.6110121591  0.0000000 301  0  20  16 Mutually_Exclusive</span><br><span class="line">26:   TSC1    VHL 0.6229248936  0.3403639 168  1 165   3 Mutually_Exclusive</span><br><span class="line">27: ARID1A   BAP1 0.6463374918  1.4627394 290  2  33  12       Co_Occurence</span><br><span class="line">28:    ATM   BAP1 0.6765457171  1.2458698 288  2  33  14       Co_Occurence</span><br><span class="line">29:  SETD2 ARID1A 0.6890948911  1.1785374 283  2  12  40       Co_Occurence</span><br><span class="line">30:    ATM  SETD2 0.7037518548  0.4560963 280  1  41  15 Mutually_Exclusive</span><br><span class="line">31:   TSC1   BAP1 1.0000000000  0.0000000 298  0  35   4 Mutually_Exclusive</span><br><span class="line">32:   TP53   BAP1 1.0000000000  0.0000000 294  0  35   8 Mutually_Exclusive</span><br><span class="line">33: ARID1A  KDM5C 1.0000000000  0.0000000 303  0  20  14 Mutually_Exclusive</span><br><span class="line">34:   TSC1  KDM5C 1.0000000000  0.0000000 313  0  20   4 Mutually_Exclusive</span><br><span class="line">35:   TP53    ATM 1.0000000000  0.0000000 313  0  16   8 Mutually_Exclusive</span><br><span class="line">36:   TP53 ARID1A 1.0000000000  0.0000000 315  0  14   8 Mutually_Exclusive</span><br><span class="line">37:    NF2 ARID1A 1.0000000000  0.0000000 318  0  14   5 Mutually_Exclusive</span><br><span class="line">38:   BAP1    NF2 1.0000000000  0.0000000 297  0   5  35 Mutually_Exclusive</span><br><span class="line">39:  KDM5C    NF2 1.0000000000  0.0000000 312  0   5  20 Mutually_Exclusive</span><br><span class="line">40:   PTEN    NF2 1.0000000000  0.0000000 320  0   5  12 Mutually_Exclusive</span><br><span class="line">41:   PTEN   TSC1 1.0000000000  0.0000000 321  0   4  12 Mutually_Exclusive</span><br><span class="line">42:   TP53   TSC1 1.0000000000  0.0000000 325  0   4   8 Mutually_Exclusive</span><br><span class="line">43:   BAP1    VHL 1.0000000000  0.9698796 153 17 149  18 Mutually_Exclusive</span><br><span class="line">44:  KDM5C    VHL 1.0000000000  1.0319633 161 10 156  10       Co_Occurence</span><br><span class="line">45:  KDM5C  PBRM1 1.0000000000  0.9225070 184  8 133  12 Mutually_Exclusive</span><br><span class="line">46:   TSC1  PBRM1 1.0000000000  1.3942563 194  2 139   2       Co_Occurence</span><br><span class="line">47:   TP53  SETD2 1.0000000000  1.0034740 288  1  41   7       Co_Occurence</span><br><span class="line">48:   PTEN   BAP1 1.0000000000  0.7786031 291  1  34  11 Mutually_Exclusive</span><br><span class="line">49:   TP53  KDM5C 1.0000000000  0.0000000 309  0  20   8 Mutually_Exclusive</span><br><span class="line">50: ARID1A    ATM 1.0000000000  0.0000000 307  0  16  14 Mutually_Exclusive</span><br><span class="line">51:   PTEN    ATM 1.0000000000  0.0000000 309  0  16  12 Mutually_Exclusive</span><br><span class="line">52:    NF2    ATM 1.0000000000  0.0000000 316  0  16   5 Mutually_Exclusive</span><br><span class="line">53:   PTEN ARID1A 1.0000000000  0.0000000 311  0  14  12 Mutually_Exclusive</span><br><span class="line">54:   TSC1 ARID1A 1.0000000000  0.0000000 319  0  14   4 Mutually_Exclusive</span><br><span class="line">55:   TSC1    NF2 1.0000000000  0.0000000 328  0   5   4 Mutually_Exclusive</span><br><span class="line">     gene1  gene2       pValue  oddsRatio  00 11  01  10              Event</span><br><span class="line">             pair event_ratio</span><br><span class="line"> 1:    PBRM1, VHL      85/137</span><br><span class="line"> 2:  PBRM1, SETD2      26/131</span><br><span class="line"> 3:    ATM, PBRM1      11/135</span><br><span class="line"> 4:      NF2, VHL       0/171</span><br><span class="line"> 5:     TP53, VHL       1/172</span><br><span class="line"> 6:    NF2, PBRM1       0/146</span><br><span class="line"> 7:     PTEN, VHL       9/160</span><br><span class="line"> 8:     NF2, TP53        1/11</span><br><span class="line"> 9:   PBRM1, TP53       1/147</span><br><span class="line">10:     ATM, TSC1        1/18</span><br><span class="line">11:   PTEN, SETD2        3/48</span><br><span class="line">12:   BAP1, PBRM1      11/154</span><br><span class="line">13:   BAP1, KDM5C        0/55</span><br><span class="line">14:    PTEN, TP53        1/18</span><br><span class="line">15:   ARID1A, VHL       9/162</span><br><span class="line">16:      ATM, VHL      10/162</span><br><span class="line">17:    SETD2, VHL      24/160</span><br><span class="line">18:   SETD2, TSC1        1/44</span><br><span class="line">19:   BAP1, SETD2        6/65</span><br><span class="line">20:  KDM5C, SETD2        1/60</span><br><span class="line">21:    NF2, SETD2        1/45</span><br><span class="line">22:   KDM5C, PTEN        1/30</span><br><span class="line">23:   PBRM1, PTEN       6/141</span><br><span class="line">24: ARID1A, PBRM1       7/141</span><br><span class="line">25:    ATM, KDM5C        0/36</span><br><span class="line">26:     TSC1, VHL       1/168</span><br><span class="line">27:  ARID1A, BAP1        2/45</span><br><span class="line">28:     ATM, BAP1        2/47</span><br><span class="line">29: ARID1A, SETD2        2/52</span><br><span class="line">30:    ATM, SETD2        1/56</span><br><span class="line">31:    BAP1, TSC1        0/39</span><br><span class="line">32:    BAP1, TP53        0/43</span><br><span class="line">33: ARID1A, KDM5C        0/34</span><br><span class="line">34:   KDM5C, TSC1        0/24</span><br><span class="line">35:     ATM, TP53        0/24</span><br><span class="line">36:  ARID1A, TP53        0/22</span><br><span class="line">37:   ARID1A, NF2        0/19</span><br><span class="line">38:     BAP1, NF2        0/40</span><br><span class="line">39:    KDM5C, NF2        0/25</span><br><span class="line">40:     NF2, PTEN        0/17</span><br><span class="line">41:    PTEN, TSC1        0/16</span><br><span class="line">42:    TP53, TSC1        0/12</span><br><span class="line">43:     BAP1, VHL      17/167</span><br><span class="line">44:    KDM5C, VHL      10/166</span><br><span class="line">45:  KDM5C, PBRM1       8/145</span><br><span class="line">46:   PBRM1, TSC1       2/141</span><br><span class="line">47:   SETD2, TP53        1/48</span><br><span class="line">48:    BAP1, PTEN        1/45</span><br><span class="line">49:   KDM5C, TP53        0/28</span><br><span class="line">50:   ARID1A, ATM        0/30</span><br><span class="line">51:     ATM, PTEN        0/28</span><br><span class="line">52:      ATM, NF2        0/21</span><br><span class="line">53:  ARID1A, PTEN        0/26</span><br><span class="line">54:  ARID1A, TSC1        0/18</span><br><span class="line">55:     NF2, TSC1         0/9</span><br><span class="line">             pair event_ratio</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc05.png" alt></p>
<h2 id="3-4-文字云"><a href="#3-4-文字云" class="headerlink" title="3.4 文字云"></a>3.4 文字云</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; geneCloud(input=kirc,top=10)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc06.png" alt></p>
<h2 id="3-5-ONCOSTRIP"><a href="#3-5-ONCOSTRIP" class="headerlink" title="3.5 ONCOSTRIP"></a>3.5 ONCOSTRIP</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; oncostrip(maf=kirc,top=10) <span class="comment"># oncostrip与oncoplot非常相似，但没有右侧的统计结果，按需选择吧</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc07.png" alt></p>
<p>或按一组基因作图</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">oncostrip(maf=kirc,genes=c(<span class="string">"VHL"</span>,<span class="string">"PBRM1"</span>,<span class="string">"SETD2"</span>,<span class="string">"BAP1"</span>,<span class="string">"KDM5C"</span>,<span class="string">"TP53"</span>,<span class="string">"ATM"</span>,<span class="string">"ARID1A"</span>,<span class="string">"PTEN"</span>,<span class="string">"TSC1"</span>,<span class="string">"NF2"</span>,<span class="string">"CDKN2A"</span>,<span class="string">"CDKN2B"</span>))</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc08.png" alt></p>
<h2 id="3-6-转换（transition）和颠换（transversion）"><a href="#3-6-转换（transition）和颠换（transversion）" class="headerlink" title="3.6 转换（transition）和颠换（transversion）"></a>3.6 转换（transition）和颠换（transversion）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; titv(maf=kirc, plot=TRUE, useSyn=TRUE) <span class="comment"># titv函数的结果是对SNV_Class的另外一种诠释</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc09.png" alt></p>
<h2 id="3-7-LOLLIPOPPLOT"><a href="#3-7-LOLLIPOPPLOT" class="headerlink" title="3.7 LOLLIPOPPLOT"></a>3.7 LOLLIPOPPLOT</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">&gt; lollipopPlot(maf=kirc, gene=<span class="string">"TP53"</span>, AACol=<span class="string">"HGVSp_Short"</span>, showMutationRate=TRUE)</span><br><span class="line">8 transcripts available. Use arguments refSeqID or proteinID to manually specify tx name.</span><br><span class="line">   HGNC    refseq.ID   protein.ID aa.length</span><br><span class="line">1: TP53    NM_000546    NP_000537       393</span><br><span class="line">2: TP53 NM_001126112 NP_001119584       393</span><br><span class="line">3: TP53 NM_001126118 NP_001119590       354</span><br><span class="line">4: TP53 NM_001126115 NP_001119587       261</span><br><span class="line">5: TP53 NM_001126113 NP_001119585       346</span><br><span class="line">6: TP53 NM_001126117 NP_001119589       214</span><br><span class="line">7: TP53 NM_001126114 NP_001119586       341</span><br><span class="line">8: TP53 NM_001126116 NP_001119588       209</span><br><span class="line">Using longer transcript NM_000546 <span class="keyword">for</span> now.</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc10.png" alt></p>
<p>上面提示有报错信息，原因是TP53这个基因有8个已知的转录本，程序自动选择了最长的转录本，即NM_000546，我们可以通过refSeqID或proteinID手动指定转录本，如：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">lollipopPlot(maf=kirc, gene=<span class="string">"TP53"</span>,refSeqID=<span class="string">"NM_001126115"</span>, AACol=<span class="string">"HGVSp_Short"</span>, showMutationRate=TRUE)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc11.png" alt></p>
<h2 id="3-8-RAINFALLPLOT（降雨图）"><a href="#3-8-RAINFALLPLOT（降雨图）" class="headerlink" title="3.8 RAINFALLPLOT（降雨图）"></a>3.8 RAINFALLPLOT（降雨图）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; rainfallPlot(maf=kirc, detectChangePoints=TRUE, pointSize=0.6) <span class="comment"># 通过绘制染色体尺度的突变间距来展示hyper mutated genomic region</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc12.png" alt></p>
<h2 id="3-9-TMB（肿瘤突变负荷）"><a href="#3-9-TMB（肿瘤突变负荷）" class="headerlink" title="3.9 TMB（肿瘤突变负荷）"></a>3.9 TMB（肿瘤突变负荷）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">tcgaCompare(maf=kirc, cohortName=<span class="string">"kirc"</span>) <span class="comment"># cohortName是自定义的名称</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc13.png" alt></p>
<h2 id="3-10-VAF（Variant-Allele-Frequency，变异等位基因频率）"><a href="#3-10-VAF（Variant-Allele-Frequency，变异等位基因频率）" class="headerlink" title="3.10 VAF（Variant Allele Frequency，变异等位基因频率）"></a>3.10 VAF（Variant Allele Frequency，变异等位基因频率）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plotVaf(maf=kirc,top=10) <span class="comment"># 默认top=10，可以手动指定，也可以使用genes指定基因名称</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc14.png" alt></p>
<h1 id="四、生存分析"><a href="#四、生存分析" class="headerlink" title="四、生存分析"></a>四、生存分析</h1><p>生存分析是基于MAF文件中给定基因或样本的突变状态进行预测，可以通过参数genes指定一个或一组基因，可以通过参数samples指定样本，参数中的status需要特别注意，我这里使用的是”vital_status”，这个参数必须是“TRUE”/“FALSE”或者”1”/“0”，因此，需要在clinical.tsv中对相应的列进行修改，具体到不同的分析软件这一列的名称会大不相同。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">&gt; mafSurvival(maf=kirc, genes=c(<span class="string">"VHL"</span>), time=<span class="string">"days_to_last_follow_up"</span>, Status=<span class="string">"vital_status"</span>, isTCGA=TRUE)</span><br><span class="line">Looking <span class="keyword">for</span> clinical data <span class="keyword">in</span> annoatation slot of MAF..</span><br><span class="line">Number of mutated samples <span class="keyword">for</span> given genes: </span><br><span class="line">VHL </span><br><span class="line">166 </span><br><span class="line">Removed 69 samples with NA<span class="string">'s</span></span><br><span class="line"><span class="string">Median survival..</span></span><br><span class="line"><span class="string">    Group medianTime   N</span></span><br><span class="line"><span class="string">1: Mutant     1261.5 130</span></span><br><span class="line"><span class="string">2:     WT     1217.0 137</span></span><br><span class="line"><span class="string">Warning messages:</span></span><br><span class="line"><span class="string">1: In survival::Surv(time = Time, event = Status) :</span></span><br><span class="line"><span class="string">  Invalid status value, converted to NA</span></span><br><span class="line"><span class="string">2: In survival::Surv(time = Time, event = Status) :</span></span><br><span class="line"><span class="string">  Invalid status value, converted to NA</span></span><br><span class="line"><span class="string">3: In survival::Surv(time = Time, event = Status) :</span></span><br><span class="line"><span class="string">  Invalid status value, converted to NA</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc15.png" alt></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br></pre></td><td class="code"><pre><span class="line">&gt; mafSurvGroup(maf=kirc,geneSet=c(<span class="string">"SETD2"</span>,<span class="string">"PBRM1"</span>), time=<span class="string">"days_to_last_follow_up"</span>, Status=<span class="string">"vital_status"</span>)</span><br><span class="line">Looking <span class="keyword">for</span> clinical data <span class="keyword">in</span> annoatation slot of MAF..</span><br><span class="line">Removed 69 samples with NA<span class="string">'s</span></span><br><span class="line"><span class="string">Median survival..</span></span><br><span class="line"><span class="string">    Group medianTime   N</span></span><br><span class="line"><span class="string">1: Mutant       2231  19</span></span><br><span class="line"><span class="string">2:     WT       1176 248</span></span><br><span class="line"><span class="string">&gt; prog_geneset = survGroup(maf = kirc, top = 20, minSamples=5,geneSetSize = 2, time = "days_to_last_follow_up", Status="vital_status", verbose = TRUE)</span></span><br><span class="line"><span class="string">------------------</span></span><br><span class="line"><span class="string">genes: 20</span></span><br><span class="line"><span class="string">geneset size: 2</span></span><br><span class="line"><span class="string">190 combinations</span></span><br><span class="line"><span class="string">Looking for clinical data in annoatation slot of MAF..</span></span><br><span class="line"><span class="string">Removed 69 samples with NA'</span>s</span><br><span class="line">Geneset:  VHL,PBRM1 [N= 85 ]</span><br><span class="line">Geneset:  VHL,TTN [N= 29 ]</span><br><span class="line">Geneset:  VHL,SETD2 [N= 24 ]</span><br><span class="line">Geneset:  VHL,BAP1 [N= 17 ]</span><br><span class="line">Geneset:  VHL,MUC16 [N= 11 ]</span><br><span class="line">Geneset:  VHL,MTOR [N= 10 ]</span><br><span class="line">Geneset:  VHL,KDM5C [N= 10 ]</span><br><span class="line">Geneset:  VHL,HMCN1 [N= 8 ]</span><br><span class="line">Geneset:  VHL,DNAH9 [N= 7 ]</span><br><span class="line">Geneset:  VHL,LRP2 [N= 7 ]</span><br><span class="line">Geneset:  VHL,ATM [N= 10 ]</span><br><span class="line">Geneset:  VHL,ARID1A [N= 9 ]</span><br><span class="line">Geneset:  VHL,CSMD3 [N= 8 ]</span><br><span class="line">Geneset:  VHL,DST [N= 6 ]</span><br><span class="line">Geneset:  VHL,KMT2C [N= 7 ]</span><br><span class="line">Geneset:  VHL,ERBB4 [N= 7 ]</span><br><span class="line">Geneset:  VHL,SMARCA4 [N= 6 ]</span><br><span class="line">Geneset:  VHL,USH2A [N= 10 ]</span><br><span class="line">Geneset:  VHL,PCLO [N= 5 ]</span><br><span class="line">Geneset:  PBRM1,TTN [N= 23 ]</span><br><span class="line">Geneset:  PBRM1,SETD2 [N= 26 ]</span><br><span class="line">Geneset:  PBRM1,BAP1 [N= 11 ]</span><br><span class="line">Geneset:  PBRM1,MUC16 [N= 10 ]</span><br><span class="line">Geneset:  PBRM1,MTOR [N= 10 ]</span><br><span class="line">Geneset:  PBRM1,KDM5C [N= 8 ]</span><br><span class="line">Geneset:  PBRM1,HMCN1 [N= 8 ]</span><br><span class="line">Geneset:  PBRM1,DNAH9 [N= 9 ]</span><br><span class="line">Geneset:  PBRM1,LRP2 [N= 11 ]</span><br><span class="line">Geneset:  PBRM1,ATM [N= 11 ]</span><br><span class="line">Geneset:  PBRM1,ARID1A [N= 7 ]</span><br><span class="line">Geneset:  PBRM1,CSMD3 [N= 8 ]</span><br><span class="line">Geneset:  PBRM1,KMT2C [N= 7 ]</span><br><span class="line">Geneset:  PBRM1,ERBB4 [N= 6 ]</span><br><span class="line">Geneset:  PBRM1,SMARCA4 [N= 6 ]</span><br><span class="line">Geneset:  PBRM1,USH2A [N= 10 ]</span><br><span class="line">Geneset:  PBRM1,PCLO [N= 5 ]</span><br><span class="line">Geneset:  TTN,SETD2 [N= 11 ]</span><br><span class="line">Geneset:  TTN,BAP1 [N= 10 ]</span><br><span class="line">Geneset:  TTN,MUC16 [N= 6 ]</span><br><span class="line">Geneset:  TTN,MTOR [N= 7 ]</span><br><span class="line">Geneset:  TTN,KDM5C [N= 5 ]</span><br><span class="line">Geneset:  TTN,HMCN1 [N= 7 ]</span><br><span class="line">Geneset:  TTN,DNAH9 [N= 5 ]</span><br><span class="line">Geneset:  TTN,USH2A [N= 5 ]</span><br><span class="line">Geneset:  SETD2,BAP1 [N= 6 ]</span><br><span class="line">Geneset:  SETD2,LRP2 [N= 5 ]</span><br><span class="line">Geneset:  BAP1,MUC16 [N= 6 ]</span><br><span class="line">There were 34 warnings (use warnings() to see them)</span><br><span class="line">&gt; <span class="built_in">print</span>(prog_geneset)</span><br><span class="line">    Gene_combination  P_value       hr  WT Mutant</span><br><span class="line"> 1:        VHL_USH2A 0.000465 9.42e+00 259      8</span><br><span class="line"> 2:        TTN_USH2A 0.001190 1.36e+01 264      3</span><br><span class="line"> 3:      PBRM1_USH2A 0.001380 8.17e+00 259      8</span><br><span class="line"> 4:        VHL_KMT2C 0.026500 7.30e+00 263      4</span><br><span class="line"> 5:      PBRM1_ERBB4 0.029000 7.12e+00 263      4</span><br><span class="line"> 6:        VHL_PBRM1 0.046000 3.14e+00 198     69</span><br><span class="line"> 7:        VHL_ERBB4 0.065300 5.58e+00 261      6</span><br><span class="line"> 8:       PBRM1_MTOR 0.085400 5.11e+00 261      6</span><br><span class="line"> 9:         VHL_MTOR 0.156000 3.97e+00 261      6</span><br><span class="line">10:       VHL_ARID1A 0.523000 3.89e-08 260      7</span><br><span class="line">11:      PBRM1_MUC16 0.556000 3.94e-08 258      9</span><br><span class="line">12:        PBRM1_ATM 0.558000 3.94e-08 258      9</span><br><span class="line">13:         TTN_BAP1 0.559000 3.92e-08 262      5</span><br><span class="line">14:          VHL_ATM 0.562000 3.93e-08 260      7</span><br><span class="line">15:     PBRM1_ARID1A 0.563000 3.93e-08 261      6</span><br><span class="line">16:        VHL_MUC16 0.568000 3.95e-08 260      7</span><br><span class="line">17:         VHL_BAP1 0.585000 1.08e-07 260      7</span><br><span class="line">18:       PBRM1_LRP2 0.596000 1.08e-07 259      8</span><br><span class="line">19:    PBRM1_SMARCA4 0.603000 1.08e-07 262      5</span><br><span class="line">20:        TTN_SETD2 0.608000 1.08e-07 260      7</span><br><span class="line">21:          VHL_DST 0.636000 1.09e-07 261      6</span><br><span class="line">22:       PBRM1_BAP1 0.637000 1.09e-07 262      5</span><br><span class="line">23:      PBRM1_CSMD3 0.643000 1.09e-07 261      6</span><br><span class="line">24:        TTN_MUC16 0.646000 1.09e-07 264      3</span><br><span class="line">25:      PBRM1_KMT2C 0.647000 1.09e-07 262      5</span><br><span class="line">26:        VHL_DNAH9 0.656000 1.09e-07 261      6</span><br><span class="line">27:        VHL_KDM5C 0.660000 1.09e-07 261      6</span><br><span class="line">28:        VHL_CSMD3 0.665000 1.09e-07 259      8</span><br><span class="line">29:         VHL_LRP2 0.666000 1.09e-07 262      5</span><br><span class="line">30:      PBRM1_KDM5C 0.671000 1.09e-07 262      5</span><br><span class="line">31:        VHL_SETD2 0.672000 1.55e+00 251     16</span><br><span class="line">32:      VHL_SMARCA4 0.672000 1.10e-07 263      4</span><br><span class="line">33:      PBRM1_DNAH9 0.683000 1.10e-07 260      7</span><br><span class="line">34:       SETD2_LRP2 0.684000 1.10e-07 264      3</span><br><span class="line">35:         VHL_PCLO 0.689000 1.10e-07 263      4</span><br><span class="line">36:       SETD2_BAP1 0.693000 1.10e-07 264      3</span><br><span class="line">37:         TTN_MTOR 0.697000 1.10e-07 263      4</span><br><span class="line">38:        VHL_HMCN1 0.715000 1.10e-07 263      4</span><br><span class="line">39:        TTN_DNAH9 0.724000 1.10e-07 264      3</span><br><span class="line">40:       PBRM1_PCLO 0.727000 1.10e-07 263      4</span><br><span class="line">41:        TTN_KDM5C 0.727000 1.10e-07 264      3</span><br><span class="line">42:        TTN_HMCN1 0.743000 3.01e-07 263      4</span><br><span class="line">43:        PBRM1_TTN 0.751000 1.39e+00 250     17</span><br><span class="line">44:       BAP1_MUC16 0.793000 3.02e-07 266      1</span><br><span class="line">45:      PBRM1_HMCN1 0.898000 8.26e-07 265      2</span><br><span class="line">46:          VHL_TTN 0.913000 1.12e+00 246     21</span><br><span class="line">47:      PBRM1_SETD2 0.955000 1.06e+00 248     19</span><br><span class="line">    Gene_combination  P_value       hr  WT Mutant</span><br><span class="line"></span><br><span class="line"><span class="comment"># 接着，可以使用mafSurvGroup进行作图</span></span><br><span class="line"></span><br><span class="line">mafSurvGroup(maf=kirc,geneSet=c(<span class="string">"VHL"</span>,<span class="string">"USH2A"</span>), time=<span class="string">"days_to_last_follow_up"</span>, Status=<span class="string">"vital_status"</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc16.png" alt></p>
<h1 id="五、对比两个MAF文件"><a href="#五、对比两个MAF文件" class="headerlink" title="五、对比两个MAF文件"></a>五、对比两个MAF文件</h1><p>这一功能使得我们可以通过对临床数据进行分类而比较其差异，比如同一疾病中不同性别是否有显著差异，比如不同种族是否有显著差异，或者不同年龄阶段是否有显著差异，等等。下面，我们以性别将KIRC分为两类，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">kirc &lt;- read.maf(maf=<span class="string">"mutect.maf"</span>)</span><br><span class="line">require(data.table)</span><br><span class="line"></span><br><span class="line">clin&lt;-as.data.frame(fread(<span class="string">"clinical.tsv"</span>))</span><br><span class="line">clin.female &lt;- subset(clin, gender==<span class="string">"female"</span>)<span class="variable">$Tumor_Sample_Barcode</span></span><br><span class="line">clin.male &lt;- subset(clin, gender==<span class="string">"male"</span>)<span class="variable">$Tumor_Sample_Barcode</span></span><br><span class="line"></span><br><span class="line">kirc.male &lt;- subsetMaf(maf=kirc, tsb=clin.male, isTCGA=TRUE)</span><br><span class="line">kirc.female &lt;- subsetMaf(maf=kirc, tsb=clin.female, isTCGA=TRUE)</span><br></pre></td></tr></table></figure>
<p>至此，基于性别分类的两个MAF文件就准备好了，下面可以对这两组数据进行作图，</p>
<h2 id="5-1-FORESTPLOT（森林图）"><a href="#5-1-FORESTPLOT（森林图）" class="headerlink" title="5.1 FORESTPLOT（森林图）"></a>5.1 FORESTPLOT（森林图）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">comp &lt;- mafCompare(m1=kirc.male, m2=kirc.female, m1Name=<span class="string">"Male"</span>, m2Name=<span class="string">"Female"</span>, minMut=5)</span><br><span class="line">forestPlot(mafCompareRes=comp, pVal=0.05, color=c(<span class="string">"maroon"</span>, <span class="string">"royalblue"</span>), geneFontSize=0.8)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc17.png" alt></p>
<h2 id="5-2-ONCOPLOT（双MAF瀑布图）"><a href="#5-2-ONCOPLOT（双MAF瀑布图）" class="headerlink" title="5.2 ONCOPLOT（双MAF瀑布图）"></a>5.2 ONCOPLOT（双MAF瀑布图）</h2><p>oncoplot中基因的选择，可根据mafCompare()的结果，选择有显著差异的基因，进行作图</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">genes &lt;- c(<span class="string">"ZFHX4"</span>, <span class="string">"ERBB4"</span>, <span class="string">"TGM5"</span>, <span class="string">"TENM1"</span>, <span class="string">"BRWD3"</span>, <span class="string">"ARID1A"</span>, <span class="string">"KDM5C"</span>)</span><br><span class="line">coOncoplot(m1=kirc.female, m2=kirc.male, m1Name=<span class="string">"Female"</span>, m2Name=<span class="string">"male"</span>, genes=genes)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc18.png" alt></p>
<h2 id="5-3-LOLLIPOPOLOT2（双MAF棒棒糖图）"><a href="#5-3-LOLLIPOPOLOT2（双MAF棒棒糖图）" class="headerlink" title="5.3 LOLLIPOPOLOT2（双MAF棒棒糖图）"></a>5.3 LOLLIPOPOLOT2（双MAF棒棒糖图）</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">lollipopPlot2(m1=luad.female, m2=luad.male, m1_name=<span class="string">"Female"</span>, m2_name=<span class="string">"Male"</span>, gene=<span class="string">"IQGAP2"</span>, AACol1 = <span class="string">"HGVSp_Short"</span>, AACol2 = <span class="string">"HGVSp_Short"</span>)</span><br><span class="line"><span class="comment"># 同前面的棒棒糖图一样，直接运行会提示有多个转录本，可以通过refSeqID/ proteinID来手动指定转录本进行绘图</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc19.png" alt></p>
<h1 id="六、信号通路富集"><a href="#六、信号通路富集" class="headerlink" title="六、信号通路富集"></a>六、信号通路富集</h1><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">&gt; OncogenicPathways(maf = laml)</span><br><span class="line"><span class="comment"># 这个功能可说得上是简单粗暴，没有可选参数，只需要一个INPUT即可</span></span><br><span class="line">Pathway alteration fractions</span><br><span class="line">       Pathway  N n_affected_genes fraction_affected</span><br><span class="line"> 1:    RTK-RAS 85               60         0.7058824</span><br><span class="line"> 2:      NOTCH 71               43         0.6056338</span><br><span class="line"> 3:        WNT 68               38         0.5588235</span><br><span class="line"> 4:      Hippo 38               25         0.6578947</span><br><span class="line"> 5:       PI3K 29               21         0.7241379</span><br><span class="line"> 6:        MYC 13                7         0.5384615</span><br><span class="line"> 7:       TP53  6                5         0.8333333</span><br><span class="line"> 8: Cell_Cycle 15                5         0.3333333</span><br><span class="line"> 9:       NRF2  3                3         1.0000000</span><br><span class="line">10:   TGF-Beta  7                3         0.4285714</span><br><span class="line"><span class="comment"># 同时绘制图形</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc20.png" alt></p>
<p>从上面输出的统计结果，可以选择感兴趣的信号通路进行作图，注意，图中的红色表示抑癌基因，蓝色表示致癌基因。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&gt; PlotOncogenicPathways(maf=kirc, pathways = <span class="string">"RTK-RAS"</span>)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc21.png" alt></p>
<h1 id="七、临床富集分析"><a href="#七、临床富集分析" class="headerlink" title="七、临床富集分析"></a>七、临床富集分析</h1><p>clinicalEnrichment可以对指定的临床特征进行富集分析，比如指定”ajcc_pathologic_stage”，命令如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">&gt; stage = clinicalEnrichment(maf=kirc, clinicalFeature = <span class="string">'ajcc_pathologic_stage'</span>)</span><br><span class="line"></span><br><span class="line">Sample size per factor <span class="keyword">in</span> ajcc_pathologic_stage:</span><br><span class="line"></span><br><span class="line">      <span class="string">'--   Stage I  Stage II Stage III  Stage IV </span></span><br><span class="line"><span class="string">        2       192        33        68        41</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string"># 查看符合p_value &lt; 0.01时的富集情况</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">&gt; stage$groupwise_comparision[p_value &lt; 0.01]</span></span><br><span class="line"><span class="string">   Hugo_Symbol    Group1 Group2 n_mutated_group1 n_mutated_group2</span></span><br><span class="line"><span class="string">1:       SETD2  Stage IV   Rest         13 of 41        29 of 295</span></span><br><span class="line"><span class="string">2:       CSMD3 Stage III   Rest          8 of 68         6 of 268</span></span><br><span class="line"><span class="string">3:      COL1A2  Stage II   Rest          4 of 33         3 of 303</span></span><br><span class="line"><span class="string">4:     PKHD1L1  Stage IV   Rest          5 of 41         5 of 295</span></span><br><span class="line"><span class="string">5:      ANKS1B  Stage IV   Rest          4 of 41         3 of 295</span></span><br><span class="line"><span class="string">6:       STAG2 Stage III   Rest          6 of 68         4 of 268</span></span><br><span class="line"><span class="string">7:       PRKDC  Stage IV   Rest          4 of 41         4 of 295</span></span><br><span class="line"><span class="string">        p_value OR_low   OR_high       fdr</span></span><br><span class="line"><span class="string">1: 0.0004091851      0 0.4862509 0.4828384</span></span><br><span class="line"><span class="string">2: 0.0020655103      0 0.4967966 0.8691319</span></span><br><span class="line"><span class="string">3: 0.0022096573      0 0.3539644 0.8691319</span></span><br><span class="line"><span class="string">4: 0.0033918860      0 0.4593002 1.0000000</span></span><br><span class="line"><span class="string">5: 0.0051372257      0 0.4581794 1.0000000</span></span><br><span class="line"><span class="string">6: 0.0059701059      0 0.5596025 1.0000000</span></span><br><span class="line"><span class="string">7: 0.0093632230      0 0.5595471 1.0000000</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string"># 通过函数plotEnrichmentResults来作图</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">&gt; plotEnrichmentResults(enrich_res=stage, pVal = 0.01)</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc22.png" alt></p>
<p>同理，也可对其他临床特征进行富集分析，比如性别</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">&gt; gender = clinicalEnrichment(maf=kirc, clinicalFeature = <span class="string">'gender'</span>)</span><br><span class="line">Sample size per factor <span class="keyword">in</span> gender:</span><br><span class="line"></span><br><span class="line">female   male </span><br><span class="line">   120    216 </span><br><span class="line">&gt; gender<span class="variable">$groupwise_comparision</span>[p_value &lt; 0.01]</span><br><span class="line">   Hugo_Symbol Group1 Group2 n_mutated_group1 n_mutated_group2     p_value</span><br><span class="line">1:      ARID1A female   Rest        11 of 120         3 of 216 0.001073289</span><br><span class="line">2:     DYNC2H1 female   Rest         7 of 120         1 of 216 0.003695386</span><br><span class="line">3:       KDM5C   male   Rest        18 of 216         2 of 120 0.008638251</span><br><span class="line">4:   KIAA1549L female   Rest         6 of 120         1 of 216 0.009419279</span><br><span class="line">5:       TENM1 female   Rest         6 of 120         1 of 216 0.009419279</span><br><span class="line">   OR_low   OR_high       fdr</span><br><span class="line">1:      0 0.4594307 0.5065922</span><br><span class="line">2:      0 0.4787417 0.8721112</span><br><span class="line">3:      0 0.6794989 0.8891799</span><br><span class="line">4:      0 0.5892996 0.8891799</span><br><span class="line">5:      0 0.5892996 0.8891799</span><br><span class="line">&gt; plotEnrichmentResults(enrich_res=gender, pVal = 0.01)</span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc23.png" alt></p>
<h1 id="八、肿瘤异质性分析"><a href="#八、肿瘤异质性分析" class="headerlink" title="八、肿瘤异质性分析"></a>八、肿瘤异质性分析</h1><p>一般而言，肿瘤是异质的，包含不同的亚型的细胞，可以通过inferHeterogeneity来对VAF进行评估以推断肿瘤异质性</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br></pre></td><td class="code"><pre><span class="line">&gt; inferHeterogeneity(maf=kirc)</span><br><span class="line">t_vaf field is missing, but found t_ref_count &amp; t_alt_count columns. Estimating vaf..</span><br><span class="line">Processing TCGA-B8-4143..</span><br><span class="line">Processing TCGA-B0-5098..</span><br><span class="line">Processing TCGA-A3-A8OV..</span><br><span class="line">Processing TCGA-CJ-4920..</span><br><span class="line">Processing TCGA-CZ-5468..</span><br><span class="line"><span class="variable">$clusterData</span></span><br><span class="line">      Hugo_Symbol Chromosome Start_Position End_Position</span><br><span class="line">   1:      CDK11B          1        1638569      1638569</span><br><span class="line">   2:      CDK11B          1        1638570      1638570</span><br><span class="line">   3:       PLOD1          1       11949869     11949869</span><br><span class="line">   4:       PLOD1          1       11949870     11949871</span><br><span class="line">   5:     DNAJC16          1       15565965     15565966</span><br><span class="line">  ---                                                   </span><br><span class="line">2495:       UBOX5         20        3122571      3122571</span><br><span class="line">2496:       PHF20         20       35871808     35871808</span><br><span class="line">2497:        LIPI         21       14206820     14206820</span><br><span class="line">2498:        MARS         12       57488179     57488179</span><br><span class="line">2499:       MXRA5          X        3317692      3317692</span><br><span class="line">      Tumor_Sample_Barcode      t_vaf cluster     MATH</span><br><span class="line">   1:         TCGA-B8-4143 0.07500000       1 41.61684</span><br><span class="line">   2:         TCGA-B8-4143 0.07500000       1 41.61684</span><br><span class="line">   3:         TCGA-B8-4143 0.07500000       1 41.61684</span><br><span class="line">   4:         TCGA-B8-4143 0.07692308       1 41.61684</span><br><span class="line">   5:         TCGA-B8-4143 0.06217617       1 41.61684</span><br><span class="line">  ---                                                 </span><br><span class="line">2495:         TCGA-CZ-5468 0.18994413       1 54.05899</span><br><span class="line">2496:         TCGA-CZ-5468 0.08011869       1 54.05899</span><br><span class="line">2497:         TCGA-CZ-5468 0.11620795       1 54.05899</span><br><span class="line">2498:         TCGA-CZ-5468 0.78947368 outlier 54.05899</span><br><span class="line">2499:         TCGA-CZ-5468 0.71794872 outlier 54.05899</span><br><span class="line">      MedianAbsoluteDeviation</span><br><span class="line">   1:                1.538092</span><br><span class="line">   2:                1.538092</span><br><span class="line">   3:                1.538092</span><br><span class="line">   4:                1.538092</span><br><span class="line">   5:                1.538092</span><br><span class="line">  ---                        </span><br><span class="line">2495:               10.526316</span><br><span class="line">2496:               10.526316</span><br><span class="line">2497:               10.526316</span><br><span class="line">2498:               10.526316</span><br><span class="line">2499:               10.526316</span><br><span class="line"></span><br><span class="line"><span class="variable">$clusterMeans</span></span><br><span class="line">    Tumor_Sample_Barcode cluster    meanVaf</span><br><span class="line"> 1:         TCGA-B8-4143       1 0.05095599</span><br><span class="line"> 2:         TCGA-B8-4143       2 0.11837474</span><br><span class="line"> 3:         TCGA-B8-4143       3 0.30865454</span><br><span class="line"> 4:         TCGA-B8-4143 outlier 0.60606061</span><br><span class="line"> 5:         TCGA-B0-5098       3 0.53075245</span><br><span class="line"> 6:         TCGA-B0-5098       1 0.10022037</span><br><span class="line"> 7:         TCGA-B0-5098       2 0.32285970</span><br><span class="line"> 8:         TCGA-B0-5098 outlier 0.64120134</span><br><span class="line"> 9:         TCGA-A3-A8OV       2 0.37256626</span><br><span class="line">10:         TCGA-A3-A8OV       1 0.06902849</span><br><span class="line">11:         TCGA-A3-A8OV       3 0.71787855</span><br><span class="line">12:         TCGA-A3-A8OV outlier 0.58955224</span><br><span class="line">13:         TCGA-CJ-4920       2 0.27004853</span><br><span class="line">14:         TCGA-CJ-4920       1 0.06939873</span><br><span class="line">15:         TCGA-CJ-4920       3 0.43948759</span><br><span class="line">16:         TCGA-CJ-4920 outlier 0.13793103</span><br><span class="line">17:         TCGA-CZ-5468       2 0.35860649</span><br><span class="line">18:         TCGA-CZ-5468       1 0.14729743</span><br><span class="line">19:         TCGA-CZ-5468 outlier 0.75371120</span><br></pre></td></tr></table></figure>
<p>上面，对5个barcode进行了整体分析，下面，可以指定某一个Barcode来具体看一下，比如指定”TCGA-B8-4143”</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">&gt; barcode1 &lt;- inferHeterogeneity(maf=kirc, tsb=<span class="string">"TCGA-B8-4143"</span>)</span><br><span class="line">t_vaf field is missing, but found t_ref_count &amp; t_alt_count columns. Estimating vaf..</span><br><span class="line">Processing TCGA-B8-4143..</span><br><span class="line"></span><br><span class="line"><span class="comment"># 可以通过$clusterMeans来查看聚类情况与VAF均值，或通过$clusterData来查看基因的详细信息</span></span><br><span class="line">&gt; barcode1<span class="variable">$clusterMeans</span></span><br><span class="line">   Tumor_Sample_Barcode cluster    meanVaf</span><br><span class="line">1:         TCGA-B8-4143       1 0.05095599</span><br><span class="line">2:         TCGA-B8-4143       2 0.11837474</span><br><span class="line">3:         TCGA-B8-4143       3 0.30865454</span><br><span class="line">4:         TCGA-B8-4143 outlier 0.60606061</span><br><span class="line"></span><br><span class="line"><span class="comment"># 使用plotClusters来作图</span></span><br><span class="line">&gt; plotClusters(clusters=barcode1)</span><br><span class="line"><span class="comment"># 这个肿瘤异质性聚类结果是比较差的，一方面主峰的频率只有5%，另一方面MATH scores太高，分数越高结果越差</span></span><br></pre></td></tr></table></figure>
<p><img src="/images/R/kirc24.png" alt></p>

      
    </div>

<div>


<ul class="post-copyright">
<li class="post-copyright-author">
<strong>本文作者：</strong>括囊无誉
</li>
<li class="post-copyright-link">
<strong>本文链接：</strong>
<a href="/R/maltools/" title="使用MAFTOOLS分析肿瘤变异数据">R/maltools/</a>
</li>
<li class="post-copyright-license">
<strong>版权声明： </strong>
本博客所有文章均为原创作品，转载请注明出处！
</li>
</ul>

</div>

    
    
    


<div>

<div style="text-align:center;color: #ccc;font-size:14px;">------ 本文结束 ------</div>

</div>

    

    
      <div>
        <div style="padding: 10px 0; margin: 20px auto; width: 90%; text-align: center;">
  <div>坚持原创文章分享，您的支持将鼓励我继续创作！</div>
  <button id="rewardButton" disable="enable" onclick="var qr = document.getElementById('QR'); if (qr.style.display === 'none') {qr.style.display='block';} else {qr.style.display='none'}">
    <span>Donate</span>
  </button>
  <div id="QR" style="display: none;">

    
      <div id="wechat" style="display: inline-block">
        <img id="wechat_qr" src="/images/wechatpay.jpg" alt="括囊无誉 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
    

    

    

  </div>
</div>

      </div>
    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/MAFTOOLS/" rel="tag"># MAFTOOLS</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/wes/" rel="next" title="外显子测序分析教程">
                <i class="fa fa-chevron-left"></i> 外显子测序分析教程
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/R/readTsv/" rel="prev" title="R基础命令：读取TSV文件">
                R基础命令：读取TSV文件 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a>
        </noscript>
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image" src="/images/avatar.png" alt="括囊无誉">
          <p class="site-author-name" itemprop="name">括囊无誉</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives/">
                <span class="site-state-item-count">203</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">11</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">216</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        
          <div class="feed-link motion-element">
            <a href="/atom.xml" rel="alternate">
              <i class="fa fa-rss"></i>
              RSS
            </a>
          </div>
        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="http://www.jianshu.com/u/75d3f95b058e" target="_blank" title="简书">
                  
                    <i class="fa fa-fw fa-http://www.jianshu.com/u/75d3f95b058e"></i>
                  
                    
                      简书
                    
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#一、数据下载与处理"><span class="nav-number">1.</span> <span class="nav-text">一、数据下载与处理</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#二、读取MAF文件并进行统计"><span class="nav-number">2.</span> <span class="nav-text">二、读取MAF文件并进行统计</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#2-1-MAF文件读取"><span class="nav-number">2.1.</span> <span class="nav-text">2.1 MAF文件读取</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2-2-MAF文件统计"><span class="nav-number">2.2.</span> <span class="nav-text">2.2 MAF文件统计</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#三、突变的可视化"><span class="nav-number">3.</span> <span class="nav-text">三、突变的可视化</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#3-1-MAF文件汇总统计图"><span class="nav-number">3.1.</span> <span class="nav-text">3.1 MAF文件汇总统计图</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-2-ONCOPLOT（瀑布图）"><span class="nav-number">3.2.</span> <span class="nav-text">3.2 ONCOPLOT（瀑布图）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-3-基因突变共存与排斥"><span class="nav-number">3.3.</span> <span class="nav-text">3.3 基因突变共存与排斥</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-4-文字云"><span class="nav-number">3.4.</span> <span class="nav-text">3.4 文字云</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-5-ONCOSTRIP"><span class="nav-number">3.5.</span> <span class="nav-text">3.5 ONCOSTRIP</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-6-转换（transition）和颠换（transversion）"><span class="nav-number">3.6.</span> <span class="nav-text">3.6 转换（transition）和颠换（transversion）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-7-LOLLIPOPPLOT"><span class="nav-number">3.7.</span> <span class="nav-text">3.7 LOLLIPOPPLOT</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-8-RAINFALLPLOT（降雨图）"><span class="nav-number">3.8.</span> <span class="nav-text">3.8 RAINFALLPLOT（降雨图）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-9-TMB（肿瘤突变负荷）"><span class="nav-number">3.9.</span> <span class="nav-text">3.9 TMB（肿瘤突变负荷）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3-10-VAF（Variant-Allele-Frequency，变异等位基因频率）"><span class="nav-number">3.10.</span> <span class="nav-text">3.10 VAF（Variant Allele Frequency，变异等位基因频率）</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#四、生存分析"><span class="nav-number">4.</span> <span class="nav-text">四、生存分析</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#五、对比两个MAF文件"><span class="nav-number">5.</span> <span class="nav-text">五、对比两个MAF文件</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#5-1-FORESTPLOT（森林图）"><span class="nav-number">5.1.</span> <span class="nav-text">5.1 FORESTPLOT（森林图）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#5-2-ONCOPLOT（双MAF瀑布图）"><span class="nav-number">5.2.</span> <span class="nav-text">5.2 ONCOPLOT（双MAF瀑布图）</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#5-3-LOLLIPOPOLOT2（双MAF棒棒糖图）"><span class="nav-number">5.3.</span> <span class="nav-text">5.3 LOLLIPOPOLOT2（双MAF棒棒糖图）</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#六、信号通路富集"><span class="nav-number">6.</span> <span class="nav-text">六、信号通路富集</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#七、临床富集分析"><span class="nav-number">7.</span> <span class="nav-text">七、临床富集分析</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#八、肿瘤异质性分析"><span class="nav-number">8.</span> <span class="nav-text">八、肿瘤异质性分析</span></a></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">
  
  &copy;  2017 - 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">括囊无誉</span>
</div>


<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


        
<div class="busuanzi-count">
  <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="site-uv">
      <i class="fa fa-user"></i> 访问人数
      <span class="busuanzi-value" id="busuanzi_value_site_uv"></span>
      
    </span>
  

  
    <span class="site-pv">
      <i class="fa fa-eye"></i> 总访问量
      <span class="busuanzi-value" id="busuanzi_value_site_pv"></span>
      次
    </span>
  
</div>


        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>



  


  

    
      <script id="dsq-count-scr" src="https://uteric-1.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'http://uteric.github.io/R/maltools/';
          this.page.identifier = 'R/maltools/';
          this.page.title = '使用MAFTOOLS分析肿瘤变异数据';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://uteric-1.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script>
    

  




	





  








  

  <script type="text/javascript">
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url);
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x" /></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x" /></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  
<script>
(function(){
    var bp = document.createElement('script');
    var curProtocol = window.location.protocol.split(':')[0];
    if (curProtocol === 'https') {
        bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';        
    }
    else {
        bp.src = 'http://push.zhanzhang.baidu.com/push.js';
    }
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(bp, s);
})();
</script>


  

  

  

  

</body>
</html>
