<!DOCTYPE html>



  


<html class="theme-next mist use-motion" lang="zh-Hans">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">








<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css">




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css">


  <meta name="keywords" content="RNASEQ,">





  <link rel="alternate" href="/atom.xml" title="诸子百家" type="application/atom+xml">




  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.2">






<meta name="description" content="HTseq统计基因reads数HTseq是用于分析基因表达量的软件，能够根据基因结构注释GTF文件对排序过的SAM/BAM文件进行基因reads数的统计。 语法：htseq-count [options] 1mkdir -p ncbi/matrix/ 在ncbi目录下创建matrix文件夹，用于存放HTseq生成的.count文件。-p (-parents)，加上此选项后，系统将自动建立那些尚不存">
<meta name="keywords" content="RNASEQ">
<meta property="og:type" content="article">
<meta property="og:title" content="RNASEQ分析入门笔记7- HTSeq定量基因表达水平">
<meta property="og:url" content="http://uteric.github.io/RNASEQ/RNASEQ HTSeq定量基因表达水平/index.html">
<meta property="og:site_name" content="诸子百家">
<meta property="og:description" content="HTseq统计基因reads数HTseq是用于分析基因表达量的软件，能够根据基因结构注释GTF文件对排序过的SAM/BAM文件进行基因reads数的统计。 语法：htseq-count [options] 1mkdir -p ncbi/matrix/ 在ncbi目录下创建matrix文件夹，用于存放HTseq生成的.count文件。-p (-parents)，加上此选项后，系统将自动建立那些尚不存">
<meta property="og:locale" content="zh-Hans">
<meta property="og:updated_time" content="2018-03-28T03:24:05.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="RNASEQ分析入门笔记7- HTSeq定量基因表达水平">
<meta name="twitter:description" content="HTseq统计基因reads数HTseq是用于分析基因表达量的软件，能够根据基因结构注释GTF文件对排序过的SAM/BAM文件进行基因reads数的统计。 语法：htseq-count [options] 1mkdir -p ncbi/matrix/ 在ncbi目录下创建matrix文件夹，用于存放HTseq生成的.count文件。-p (-parents)，加上此选项后，系统将自动建立那些尚不存">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '9e233f3fa2f200ca831529de7ba10253',
      indexName: 'uteric',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://uteric.github.io/RNASEQ/RNASEQ HTSeq定量基因表达水平/">





  <title>RNASEQ分析入门笔记7- HTSeq定量基因表达水平 | 诸子百家</title>
  




<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-104081724-1', 'auto');
ga('send', 'pageview');

</script>











</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <meta name="baidu-site-verification" content="bfQpbnNmEL">

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">诸子百家</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br>
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br>
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br>
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br>
            
            关于
          </a>
        </li>
      

      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br>
            
            搜索
          </a>
        </li>
      
    </ul>
  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
<div class="local-search-header clearfix">
<span class="search-icon">
<i class="fa fa-search"></i>
</span>
<span class="popup-btn-close">
<i class="fa fa-times-circle"></i>
</span>
<div class="local-search-input-wrapper">
<input autocomplete="off" placeholder="搜索..." spellcheck="false" type="text" id="local-search-input">
</div>
</div>
<div id="local-search-result"></div>
</div>


    </div>
  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://uteric.github.io/RNASEQ/RNASEQ HTSeq定量基因表达水平/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="括囊无誉">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="诸子百家">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">RNASEQ分析入门笔记7- HTSeq定量基因表达水平</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2018-03-27T22:24:05-05:00">
                2018-03-27
              </time>
            

            

            
          </span>

          
            <span class="post-category">
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/编程语言/" itemprop="url" rel="index">
                    <span itemprop="name">编程语言</span>
                  </a>
                </span>

                
                
              
            </span>
          


          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/RNASEQ/RNASEQ HTSeq定量基因表达水平/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count" data-disqus-identifier="RNASEQ/RNASEQ HTSeq定量基因表达水平/" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          

          
            <span class="post-meta-divider">|</span>
            <span class="page-pv"><i class="fa fa-file-o"></i> 热度
            <span class="busuanzi-value" id="busuanzi_value_page_pv"></span>°C
            </span>
          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <h2 id="HTseq统计基因reads数"><a href="#HTseq统计基因reads数" class="headerlink" title="HTseq统计基因reads数"></a>HTseq统计基因reads数</h2><p><strong>HTseq</strong>是用于分析基因表达量的软件，能够根据基因结构注释GTF文件对排序过的SAM/BAM文件进行基因reads数的统计。</p>
<p><em>语法：htseq-count [options]</em></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p ncbi/matrix/ 在ncbi目录下创建matrix文件夹，用于存放HTseq生成的.count文件。-p (-parents)，加上此选项后，系统将自动建立那些尚不存在的目录。</span><br></pre></td></tr></table></figure>
<p>运行HTseq-count命令：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">htseq-count ~/ncbi/aligned/SRR3589959_count.sam ~/ncbi/reference/mm10/gencode.vM10.annotation.gtf &gt; ~/ncbi/matrix/SRR3589959.count</span><br><span class="line">htseq-count ~/ncbi/aligned/SRR3589960_count.sam ~/ncbi/reference/mm10/gencode.vM10.annotation.gtf &gt; ~/ncbi/matrix/SRR3589960.count</span><br><span class="line">htseq-count ~/ncbi/aligned/SRR3589961_count.sam ~/ncbi/reference/mm10/gencode.vM10.annotation.gtf &gt; ~/ncbi/matrix/SRR3589961.count</span><br><span class="line">htseq-count ~/ncbi/aligned/SRR3589962_count.sam ~/ncbi/reference/mm10/gencode.vM10.annotation.gtf &gt; ~/ncbi/matrix/SRR3589962.count</span><br></pre></td></tr></table></figure>
<p>上述语法是将排序过的SAM文件（count.sam）进行reads数统计，并输出为.count文件（不用担心存储空间问题，每个文件仅1M左右）。此处，使用了绝对路径，也可以根据自己的情况选择使用相对路径。</p>
<p>为节省时间，可以将上述命令在4个终端中同时执行，大约需要2个小时左右可以完成，生成4个文件：</p>
<blockquote>
<p>SRR3589959.count<br>SRR3589960.count<br>SRR3589961.count<br>SRR3589962.count</p>
</blockquote>
<p>简单介绍一下.count文件的格式，在命令行下查看：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> ~/ncbi/matrix <span class="comment"># 切换到.count文件所在的matrix目录</span></span><br><span class="line">head -n 10 SRR3589959.count <span class="comment"># 使用head命令查看SRR3589959.count文件的前10行</span></span><br></pre></td></tr></table></figure>
<p>输出如下：</p>
<blockquote>
<p>ENSMUSG00000000001.4    1648<br>ENSMUSG00000000003.15    0<br>ENSMUSG00000000028.14    835<br>ENSMUSG00000000031.15    65<br>ENSMUSG00000000037.16    70<br>ENSMUSG00000000049.11    0<br>ENSMUSG00000000056.7 366<br>ENSMUSG00000000058.6    7<br>ENSMUSG00000000078.6    558<br>ENSMUSG00000000085.16    286</p>
</blockquote>
<p><strong>.count</strong>文件分为2列，第一列是基因名称（ENSMUSG00000000001.4），第二列是reads数（1648），由此可见，htseq-count命令的目的是将排序过的SAM文件（001，003，0028，0031，0037……）逐个统计reads数，但是，此时生成的.count文件只有两列。</p>
<p>因为所有的.count文件都排过序，并且共享相同的基因名称，所以下一步是将所有的.count文件合并起来，生成一个统一的文件，需要使用R语言脚本将4个.count文件合并为一个表达矩阵。</p>
<h2 id="R语言脚本合并表达矩阵"><a href="#R语言脚本合并表达矩阵" class="headerlink" title="R语言脚本合并表达矩阵"></a>R语言脚本合并表达矩阵</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">R <span class="comment"># 在命令行下输入R，进入R环境</span></span><br><span class="line">options(stringsAsFactors = FALSE) <span class="comment">#默认为TRUE，不要辨认为factors</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 59，61为对照组，60，62为处理组，首先将四个文件分别赋值为：control1，control2，rep1，rep2</span></span><br><span class="line">control1 &lt;- read.table(<span class="string">"/Users/Administrator/ncbi/matrix/SRR3589959.count"</span>, sep=<span class="string">"\t"</span>, col.names = c(<span class="string">"gene_id"</span>,<span class="string">"control1"</span>))</span><br><span class="line">control2 &lt;- read.table(<span class="string">"/Users/Administrator/ncbi/matrix/SRR3589961.count"</span>, sep=<span class="string">"\t"</span>, col.names = c(<span class="string">"gene_id"</span>,<span class="string">"control2"</span>))</span><br><span class="line">rep1 &lt;- read.table(<span class="string">"/Users/Administrator/ncbi/matrix/SRR3589960.count"</span>, sep=<span class="string">"\t"</span>, col.names = c(<span class="string">"gene_id"</span>,<span class="string">"rep1"</span>))</span><br><span class="line">rep2 &lt;- read.table(<span class="string">"/Users/Administrator/ncbi/matrix/SRR3589962.count"</span>, sep=<span class="string">"\t"</span>,col.names = c(<span class="string">"gene_id"</span>,<span class="string">"rep2"</span>))</span><br><span class="line"></span><br><span class="line"><span class="comment"># 将4个矩阵文件按照gene_id进行合并，首先组内合并，接着组间合并，并且赋值给raw_count</span></span><br><span class="line">raw_count &lt;- merge(merge(control1, control2, by=<span class="string">"gene_id"</span>), merge(rep1,rep2, by=<span class="string">"gene_id"</span>))</span><br></pre></td></tr></table></figure>
<p>对raw_count进行赋值后，我们可以查看生成的合并矩阵：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">raw_count <span class="comment"># 查看合并矩阵</span></span><br></pre></td></tr></table></figure>
<p>但是raw_count文件太长，只能输出19999行，显示达到最大输出量：</p>
<blockquote>
<p>[ reached getOption(“max.print”) – omitted 28826 rows ]</p>
</blockquote>
<p>因此，我们只显示前10行数据，来查看一下生成的矩阵的格式：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">head (raw_count, n=10) <span class="comment">#查看raw_count的前10行</span></span><br></pre></td></tr></table></figure>
<p>输出结果如下：</p>
<blockquote>
<p> gene_id control1 control2 rep1 rep2<br>1   ENSMUSG00000000001.4     1648     2306 2941 2780<br>2  ENSMUSG00000000003.15        0        0    0    0<br>3  ENSMUSG00000000028.14      835      950 1366 1051<br>4  ENSMUSG00000000031.15       65       83   52   53<br>5  ENSMUSG00000000037.16       70       53   94   66<br>6  ENSMUSG00000000049.11        0        3    4    5<br>7   ENSMUSG00000000056.7      366      386  722  301<br>8   ENSMUSG00000000058.6        7       11    7    3<br>9   ENSMUSG00000000078.6      558      896 1265 1342<br>10 ENSMUSG00000000085.16      286      313  343  348</p>
</blockquote>
<p>合并后的矩阵共有5列，第一列是基因名称，其余四列分别是control1，control2，rep1，rep2，从上表中可以很直观地看到每个基因的表达情况。</p>
<p>我们再来看看raw_count的最后5行，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">tail (raw_count, n=5)</span><br></pre></td></tr></table></figure>
<p>输出如下：</p>
<blockquote>
<p>gene_id control1 control2     rep1     rep2<br>48821 <strong>alignment_not_unique 11278964 14080481 18440618 14495860<br>48822            </strong>ambiguous   295498   425708   498973   381237<br>48823           <strong>no_feature 12642161 15042888 22357626 18675857<br>48824          </strong>not_aligned  1025857  1067038  1675660  1529309<br>48825        __too_low_aQual  1107674  1303141  1799176  1542845</p>
</blockquote>
<p>这5行（48821， 48822， 48823， 48824， 48825）没有匹配到相应的基因，需要删除，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">raw_count_filt &lt;- raw_count[-48821:-48825, ] <span class="comment"># 删除5行，重新赋值给raw_count_filt</span></span><br></pre></td></tr></table></figure>
<p>查看新矩阵raw_count_filt的最后5行，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">tail (raw_count_filt, n=5)</span><br></pre></td></tr></table></figure>
<p>可以看到，48821：48825已经被删除了，</p>
<blockquote>
<p>gene_id control1 control2 rep1 rep2<br>48816 ENSMUSG00000110420.1        0        0    0    0<br>48817 ENSMUSG00000110421.1        0        0    0    1<br>48818 ENSMUSG00000110422.1        1        0    1    2<br>48819 ENSMUSG00000110423.1        0        0    0    0<br>48820 ENSMUSG00000110424.1       26       20   48   24</p>
</blockquote>
<p>以行48820为例，第一列是基因名称ENSMUSG00000110424.1，但是这个名称在<a href="https://www.ebi.ac.uk/" target="_blank" rel="noopener">EBI</a>数据库上是不能直接检索到的，只能是整数形式的ENSMUSG00000110424，因此，需要进一步替换为整数的形式，并且赋值给ENSEMBL。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ENSEMBL &lt;- gsub(<span class="string">"\\.\\d*"</span>, <span class="string">""</span>, raw_count_filt<span class="variable">$gene_id</span>)</span><br><span class="line">tail (ENSEMBL, n=5) <span class="comment"># 查看ENSEMBL的最后5行</span></span><br></pre></td></tr></table></figure>
<p>输出如下：</p>
<blockquote>
<p>[1] “ENSMUSG00000110420” “ENSMUSG00000110421” “ENSMUSG00000110422”<br>[4] “ENSMUSG00000110423” “ENSMUSG00000110424”</p>
</blockquote>
<p>ENSEMBL只有一列，即修改为整数的基因名称，下一步再将ENSEMBL添加到raw_count_filt矩阵:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">row.names(raw_count_filt) &lt;- ENSEMBL <span class="comment"># 添加ENSENBL到raw_count_filt矩阵</span></span><br><span class="line">tail (raw_count_filt, n=5) <span class="comment"># 查看raw_count_filt的最后5行</span></span><br></pre></td></tr></table></figure>
<blockquote>
<p>gene_id control1 control2 rep1 rep2<br>ENSMUSG00000110420 ENSMUSG00000110420.1        0        0    0    0<br>ENSMUSG00000110421 ENSMUSG00000110421.1        0        0    0    1<br>ENSMUSG00000110422 ENSMUSG00000110422.1        1        0    1    2<br>ENSMUSG00000110423 ENSMUSG00000110423.1        0        0    0    0<br>ENSMUSG00000110424 ENSMUSG00000110424.1       26       20   48   24</p>
</blockquote>
<p>[optional] 这一步选做，即把raw_count_filt矩阵的第二列删掉，并赋值给raw_count_filt2：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">raw_count_filt2 &lt;- raw_count_filt [ ,-1] <span class="comment"># 删除raw_count_filt的第二列</span></span><br><span class="line">tail (raw_count_filt2, n=5) <span class="comment"># 查看raw_count_filt2最后5行</span></span><br></pre></td></tr></table></figure>
<blockquote>
<p>control1 control2 rep1 rep2<br>ENSMUSG00000110420        0        0    0    0<br>ENSMUSG00000110421        0        0    0    1<br>ENSMUSG00000110422        1        0    1    2<br>ENSMUSG00000110423        0        0    0    0<br>ENSMUSG00000110424       26       20   48   24</p>
</blockquote>
<p>下面，我们就可以检查一些基因的表达情况了，比如GAPDH，首先，可以在<a href="http://www.uniprot.org/" target="_blank" rel="noopener">uniprot</a>或者<a href="http://useast.ensembl.org/index.html" target="_blank" rel="noopener">Ensembl</a>数据库中找到GAPDH对应的基因编号，即ENSMUSG00000057666，并将该基因编号赋值给GAPDH：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">GAPDH &lt;- raw_count_filt2 [rownames(raw_count_filt2)==<span class="string">"ENSMUSG00000057666"</span>,] <span class="comment"># 赋值给GAPDH</span></span><br><span class="line">GAPDH <span class="comment"># 查看GAPDH表达情况</span></span><br></pre></td></tr></table></figure>
<p>输出如下：</p>
<blockquote>
<p>control1 control2 rep1 rep2<br>ENSMUSG00000057666     4136     4884 7200 4929</p>
</blockquote>
<p>如要查询其他基因的表达情况，以此类推！</p>
<p>另外，我们也可以看一看所有基因的表达概况：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">summary (raw-count_filt2)</span><br></pre></td></tr></table></figure>
<blockquote>
<p>control1           control2             rep1               rep2<br> Min.   :     0.0   Min.   :     0.0   Min.   :     0.0   Min.   :     0.0<br> 1st Qu.:     0.0   1st Qu.:     0.0   1st Qu.:     0.0   1st Qu.:     0.0<br> Median :     0.0   Median :     0.0   Median :     1.0   Median :     1.0<br> Mean   :   237.6   Mean   :   290.6   Mean   :   417.4   Mean   :   342.2<br> 3rd Qu.:    62.0   3rd Qu.:    65.0   3rd Qu.:    93.0   3rd Qu.:    71.0<br> Max.   :108056.0   Max.   :164091.0   Max.   :171319.0   Max.   :132792.0</p>
</blockquote>
<p>如果想要把分析结果保存下来，则存储为.csv格式：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">write.csv (x=raw_count_filt2, file=<span class="string">"SRR35899.csv"</span>) <span class="comment"># 保存raw_count_filt2矩阵为SRR35899.csv文件，保存位置在当前命令行目录中</span></span><br></pre></td></tr></table></figure>

      
    </div>

<div>


<ul class="post-copyright">
<li class="post-copyright-author">
<strong>本文作者：</strong>括囊无誉
</li>
<li class="post-copyright-link">
<strong>本文链接：</strong>
<a href="/RNASEQ/RNASEQ HTSeq定量基因表达水平/" title="RNASEQ分析入门笔记7- HTSeq定量基因表达水平">RNASEQ/RNASEQ HTSeq定量基因表达水平/</a>
</li>
<li class="post-copyright-license">
<strong>版权声明： </strong>
本博客所有文章均为原创作品，转载请注明出处！
</li>
</ul>

</div>

    
    
    


<div>

<div style="text-align:center;color: #ccc;font-size:14px;">------ 本文结束 ------</div>

</div>

    

    
      <div>
        <div style="padding: 10px 0; margin: 20px auto; width: 90%; text-align: center;">
  <div>坚持原创文章分享，您的支持将鼓励我继续创作！</div>
  <button id="rewardButton" disable="enable" onclick="var qr = document.getElementById('QR'); if (qr.style.display === 'none') {qr.style.display='block';} else {qr.style.display='none'}">
    <span>Donate</span>
  </button>
  <div id="QR" style="display: none;">

    
      <div id="wechat" style="display: inline-block">
        <img id="wechat_qr" src="/images/wechatpay.jpg" alt="括囊无誉 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
    

    

    

  </div>
</div>

      </div>
    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/RNASEQ/" rel="tag"># RNASEQ</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/RNASEQ/RNASEQ下载注释文件并导入IGV/" rel="next" title="RNASEQ分析入门笔记6-下载注释文件并导入IGV">
                <i class="fa fa-chevron-left"></i> RNASEQ分析入门笔记6-下载注释文件并导入IGV
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/MAC/MAC命令行操作MacVim/" rel="prev" title="MAC命令行操作MacVim">
                MAC命令行操作MacVim <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a>
        </noscript>
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image" src="/images/avatar.png" alt="括囊无誉">
          <p class="site-author-name" itemprop="name">括囊无誉</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives/">
                <span class="site-state-item-count">150</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">11</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">202</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        
          <div class="feed-link motion-element">
            <a href="/atom.xml" rel="alternate">
              <i class="fa fa-rss"></i>
              RSS
            </a>
          </div>
        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="http://www.jianshu.com/u/75d3f95b058e" target="_blank" title="简书">
                  
                    <i class="fa fa-fw fa-http://www.jianshu.com/u/75d3f95b058e"></i>
                  
                    
                      简书
                    
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#HTseq统计基因reads数"><span class="nav-number">1.</span> <span class="nav-text">HTseq统计基因reads数</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#R语言脚本合并表达矩阵"><span class="nav-number">2.</span> <span class="nav-text">R语言脚本合并表达矩阵</span></a></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">
  
  &copy;  2017 - 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">括囊无誉</span>
</div>


<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


        
<div class="busuanzi-count">
  <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="site-uv">
      <i class="fa fa-user"></i> 访问人数
      <span class="busuanzi-value" id="busuanzi_value_site_uv"></span>
      
    </span>
  

  
    <span class="site-pv">
      <i class="fa fa-eye"></i> 总访问量
      <span class="busuanzi-value" id="busuanzi_value_site_pv"></span>
      次
    </span>
  
</div>


        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>



  


  

    
      <script id="dsq-count-scr" src="https://uteric-1.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'http://uteric.github.io/RNASEQ/RNASEQ HTSeq定量基因表达水平/';
          this.page.identifier = 'RNASEQ/RNASEQ HTSeq定量基因表达水平/';
          this.page.title = 'RNASEQ分析入门笔记7- HTSeq定量基因表达水平';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://uteric-1.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script>
    

  




	





  








  

  <script type="text/javascript">
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url);
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x" /></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x" /></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  
<script>
(function(){
    var bp = document.createElement('script');
    var curProtocol = window.location.protocol.split(':')[0];
    if (curProtocol === 'https') {
        bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';        
    }
    else {
        bp.src = 'http://push.zhanzhang.baidu.com/push.js';
    }
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(bp, s);
})();
</script>


  

  

  

  

</body>
</html>
