<!DOCTYPE html>



  


<html class="theme-next mist use-motion" lang="zh-Hans">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">








<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css">




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css">


  <meta name="keywords" content="WES,">





  <link rel="alternate" href="/atom.xml" title="诸子百家" type="application/atom+xml">




  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.2">






<meta name="description" content="参考资料：Germline short variant discovery (SNPs + Indels) 下面这张图是单样本种系突变的主要分析步骤（Main steps for Germline Single-Sample Data）  一、数据质控12345678910111213141516171819202122232425262728293031323334353637383940414">
<meta name="keywords" content="WES">
<meta property="og:type" content="article">
<meta property="og:title" content="外显子测序分析二">
<meta property="og:url" content="http://uteric.github.io/WES/wes02/index.html">
<meta property="og:site_name" content="诸子百家">
<meta property="og:description" content="参考资料：Germline short variant discovery (SNPs + Indels) 下面这张图是单样本种系突变的主要分析步骤（Main steps for Germline Single-Sample Data）  一、数据质控12345678910111213141516171819202122232425262728293031323334353637383940414">
<meta property="og:locale" content="zh-Hans">
<meta property="og:image" content="http://uteric.github.io/images/LINUX/gatk01.png">
<meta property="og:updated_time" content="2020-11-06T20:47:26.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="外显子测序分析二">
<meta name="twitter:description" content="参考资料：Germline short variant discovery (SNPs + Indels) 下面这张图是单样本种系突变的主要分析步骤（Main steps for Germline Single-Sample Data）  一、数据质控12345678910111213141516171819202122232425262728293031323334353637383940414">
<meta name="twitter:image" content="http://uteric.github.io/images/LINUX/gatk01.png">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '9e233f3fa2f200ca831529de7ba10253',
      indexName: 'uteric',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://uteric.github.io/WES/wes02/">





  <title>外显子测序分析二 | 诸子百家</title>
  




<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-104081724-1', 'auto');
ga('send', 'pageview');

</script>











</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <meta name="baidu-site-verification" content="bfQpbnNmEL">

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">诸子百家</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br>
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br>
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br>
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br>
            
            关于
          </a>
        </li>
      

      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br>
            
            搜索
          </a>
        </li>
      
    </ul>
  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
<div class="local-search-header clearfix">
<span class="search-icon">
<i class="fa fa-search"></i>
</span>
<span class="popup-btn-close">
<i class="fa fa-times-circle"></i>
</span>
<div class="local-search-input-wrapper">
<input autocomplete="off" placeholder="搜索..." spellcheck="false" type="text" id="local-search-input">
</div>
</div>
<div id="local-search-result"></div>
</div>


    </div>
  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://uteric.github.io/WES/wes02/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="括囊无誉">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="诸子百家">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">外显子测序分析二</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-11-06T14:47:26-06:00">
                2020-11-06
              </time>
            

            

            
          </span>

          
            <span class="post-category">
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/编程语言/" itemprop="url" rel="index">
                    <span itemprop="name">编程语言</span>
                  </a>
                </span>

                
                
              
            </span>
          


          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/WES/wes02/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count" data-disqus-identifier="WES/wes02/" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          

          
            <span class="post-meta-divider">|</span>
            <span class="page-pv"><i class="fa fa-file-o"></i> 热度
            <span class="busuanzi-value" id="busuanzi_value_page_pv"></span>°C
            </span>
          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <p>参考资料：<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels-" target="_blank" rel="noopener">Germline short variant discovery (SNPs + Indels)</a></p>
<p>下面这张图是单样本种系突变的主要分析步骤（Main steps for Germline Single-Sample Data）</p>
<p><img src="/images/LINUX/gatk01.png" alt></p>
<h2 id="一、数据质控"><a href="#一、数据质控" class="headerlink" title="一、数据质控"></a>一、数据质控</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br></pre></td><td class="code"><pre><span class="line">fastp -i /mnt/d/wes/sample/fastq/sam01_1.fastq -I /mnt/d/wes/sample/fastq/sam01_2.fastq -o /mnt/d/wes/sample/qc/clean_sam01_1.fastq -O /mnt/d/wes/sample/qc/clean_sam01_2.fastq</span><br><span class="line"></span><br><span class="line"><span class="comment"># 耗时30秒</span></span><br><span class="line">Read1 before filtering:</span><br><span class="line">total reads: 1582559</span><br><span class="line">total bases: 238966409</span><br><span class="line">Q20 bases: 234035553(97.9366%)</span><br><span class="line">Q30 bases: 225671450(94.4365%)</span><br><span class="line"></span><br><span class="line">Read2 before filtering:</span><br><span class="line">total reads: 1582559</span><br><span class="line">total bases: 238966409</span><br><span class="line">Q20 bases: 232564639(97.3211%)</span><br><span class="line">Q30 bases: 222663993(93.1779%)</span><br><span class="line"></span><br><span class="line">Read1 after filtering:</span><br><span class="line">total reads: 1564770</span><br><span class="line">total bases: 233787024</span><br><span class="line">Q20 bases: 229439315(98.1403%)</span><br><span class="line">Q30 bases: 221439194(94.7183%)</span><br><span class="line"></span><br><span class="line">Read2 aftering filtering:</span><br><span class="line">total reads: 1564770</span><br><span class="line">total bases: 233787024</span><br><span class="line">Q20 bases: 228711758(97.8291%)</span><br><span class="line">Q30 bases: 219341614(93.8211%)</span><br><span class="line"></span><br><span class="line">Filtering result:</span><br><span class="line">reads passed filter: 3129540</span><br><span class="line">reads failed due to low quality: 35106</span><br><span class="line">reads failed due to too many N: 472</span><br><span class="line">reads failed due to too short: 0</span><br><span class="line">reads with adapter trimmed: 339368</span><br><span class="line">bases trimmed due to adapters: 4994344</span><br><span class="line"></span><br><span class="line">Duplication rate: 0.84949%</span><br><span class="line"></span><br><span class="line">Insert size peak (evaluated by paired-end reads): 190</span><br><span class="line"></span><br><span class="line">JSON report: fastp.json</span><br><span class="line">HTML report: fastp.html</span><br><span class="line"></span><br><span class="line">fastp -i /mnt/d/wes/sample/fastq/sam01_1.fastq -I /mnt/d/wes/sample/fastq/sam01_2.fastq -o /mnt/d/wes/sample/qc/clean_sam01_1.fastq -O /mnt/d/wes/sample/qc/clean_sam01_2.fastq</span><br><span class="line">fastp v0.20.1, time used: 30 seconds</span><br></pre></td></tr></table></figure>
<h2 id="二、比对至基因组"><a href="#二、比对至基因组" class="headerlink" title="二、比对至基因组"></a>二、比对至基因组</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br></pre></td><td class="code"><pre><span class="line">bwa mem -t 4 -B 4 -R <span class="string">'@RG\tID:lane1\tSM:sample1\tLB:genome\tPL:ILLUMINA'</span> -M ~/ncbi/hg38/chroms/hg38.fa /mnt/d/wes/sample/qc/clean_sam01_1.fastq /mnt/d/wes/sample/qc/clean_sam01_2.fastq | samtools view -q4 -bS -o - - | samtools sort -@4 -o /mnt/d/wes/sample/bam/clean_sam01.bam -</span><br><span class="line"><span class="comment"># 比对耗时30分钟</span></span><br><span class="line">[M::bwa_idx_load_from_disk] <span class="built_in">read</span> 0 ALT contigs</span><br><span class="line">[M::process] <span class="built_in">read</span> 267772 sequences (40000138 bp)...</span><br><span class="line">[M::process] <span class="built_in">read</span> 267742 sequences (40000164 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (0, 90228, 1, 2)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 197, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.01, 45.34)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267772 reads <span class="keyword">in</span> 272.156 CPU sec, 11919.176 real sec</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (2, 90153, 0, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 198, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.42, 45.38)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267742 reads <span class="keyword">in</span> 144.141 CPU sec, 120.016 real sec</span><br><span class="line">[M::process] <span class="built_in">read</span> 267734 sequences (40000200 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (1, 90334, 1, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 198, 232)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (43, 358)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.52, 45.61)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 421)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::process] <span class="built_in">read</span> 267748 sequences (40000292 bp)...</span><br><span class="line">[M::mem_process_seqs] Processed 267734 reads <span class="keyword">in</span> 151.547 CPU sec, 143.398 real sec</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (1, 90216, 0, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 198, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.43, 45.49)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267748 reads <span class="keyword">in</span> 146.859 CPU sec, 111.532 real sec</span><br><span class="line">[M::process] <span class="built_in">read</span> 267744 sequences (40000056 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (4, 90236, 0, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 197, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.11, 45.29)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267744 reads <span class="keyword">in</span> 123.953 CPU sec, 90.588 real sec</span><br><span class="line">[M::process] <span class="built_in">read</span> 267718 sequences (40000024 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (1, 90313, 0, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 197, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.03, 45.17)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267718 reads <span class="keyword">in</span> 141.703 CPU sec, 111.135 real sec</span><br><span class="line">[M::process] <span class="built_in">read</span> 267686 sequences (40000084 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (2, 90081, 1, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 198, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.38, 45.37)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267686 reads <span class="keyword">in</span> 134.469 CPU sec, 95.876 real sec</span><br><span class="line">[M::process] <span class="built_in">read</span> 267726 sequences (40000068 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (3, 90134, 0, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 198, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.15, 45.43)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267726 reads <span class="keyword">in</span> 134.438 CPU sec, 100.282 real sec</span><br><span class="line">[M::process] <span class="built_in">read</span> 267692 sequences (40000016 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (1, 90379, 0, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (170, 198, 232)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (46, 356)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.61, 45.49)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 418)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267692 reads <span class="keyword">in</span> 134.969 CPU sec, 100.242 real sec</span><br><span class="line">[M::process] <span class="built_in">read</span> 267726 sequences (40000064 bp)...</span><br><span class="line"> [M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (2, 90356, 0, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 197, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.19, 45.30)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::process] <span class="built_in">read</span> 267698 sequences (40000206 bp)...</span><br><span class="line">[M::mem_process_seqs] Processed 267726 reads <span class="keyword">in</span> 141.531 CPU sec, 116.261 real sec</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (2, 90064, 1, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 197, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.27, 45.37)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267698 reads <span class="keyword">in</span> 149.391 CPU sec, 125.276 real sec</span><br><span class="line">[M::process] <span class="built_in">read</span> 184554 sequences (27572736 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (0, 62039, 1, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 197, 231)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (45, 355)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (201.96, 45.23)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 417)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 184554 reads <span class="keyword">in</span> 102.375 CPU sec, 77.923 real sec</span><br><span class="line">[main] Version: 0.7.17-r1188</span><br><span class="line">[main] CMD: bwa mem -t 4 -B 4 -R @RG\tID:lane1\tSM:sample1\tLB:genome\tPL:ILLUMINA -M /home/eric/ncbi/hg38/chroms/hg38.fa /mnt/d/wes/sample/qc/clean_sam01_1.fastq /mnt/d/wes/sample/qc/clean_sam01_2.fastq</span><br><span class="line">[main] Real time: 13158.133 sec; CPU: 1804.688 sec</span><br><span class="line">[bam_sort_core] merging from 0 files and 4 <span class="keyword">in</span>-memory blocks...</span><br><span class="line"><span class="comment"># 加上索引文件</span></span><br><span class="line">samtools index /mnt/d/wes/sample/bam/clean_sam01.bam</span><br><span class="line"><span class="comment"># 查看文件大小</span></span><br><span class="line">total 159M</span><br><span class="line">-rwxrwxrwx 1 xxx xxx 155M Nov  6 18:45 clean_sam01.bam</span><br><span class="line">-rwxrwxrwx 1 xxx xxx 3.1M Nov  6 19:12 clean_sam01.bam.bai</span><br></pre></td></tr></table></figure>
<h2 id="三、PCR去重"><a href="#三、PCR去重" class="headerlink" title="三、PCR去重"></a>三、PCR去重</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br></pre></td><td class="code"><pre><span class="line">gatk MarkDuplicates -I /mnt/d/wes/sample/bam/clean_sam01.bam -O /mnt/d/wes/sample/nodup/nodup_sam01.bam -M /mnt/d/wes/sample/nodup/metrics_sam01.txt --REMOVE_DUPLICATES <span class="literal">true</span> --ASSUME_SORTED <span class="literal">true</span></span><br><span class="line"></span><br><span class="line">Using GATK jar /home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar</span><br><span class="line">Running:</span><br><span class="line">    java -Dsamjdk.use_async_io_read_samtools=<span class="literal">false</span> -Dsamjdk.use_async_io_write_samtools=<span class="literal">true</span> -Dsamjdk.use_async_io_write_tribble=<span class="literal">false</span> -Dsamjdk.compression_level=2 -jar /home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar MarkDuplicates -I /mnt/d/wes/sample/bam/clean_sam01.bam -O /mnt/d/wes/sample/nodup/nodup_sam01.bam -M /mnt/d/wes/sample/nodup/metrics_sam01.txt --REMOVE_DUPLICATES <span class="literal">true</span> --ASSUME_SORTED <span class="literal">true</span></span><br><span class="line">19:17:37.667 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_compression.so</span><br><span class="line">[Fri Nov 06 19:17:37 CST 2020] MarkDuplicates --INPUT /mnt/d/wes/sample/bam/clean_sam01.bam --OUTPUT /mnt/d/wes/sample/nodup/nodup_sam01.bam --METRICS_FILE /mnt/d/wes/sample/nodup/metrics_sam01.txt --REMOVE_DUPLICATES <span class="literal">true</span> --ASSUME_SORTED <span class="literal">true</span> --MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP 50000 --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 8000 --SORTING_COLLECTION_SIZE_RATIO 0.25 --TAG_DUPLICATE_SET_MEMBERS <span class="literal">false</span> --REMOVE_SEQUENCING_DUPLICATES <span class="literal">false</span> --TAGGING_POLICY DontTag --CLEAR_DT <span class="literal">true</span> --DUPLEX_UMI <span class="literal">false</span> --ADD_PG_TAG_TO_READS <span class="literal">true</span> --DUPLICATE_SCORING_STRATEGY SUM_OF_BASE_QUALITIES --PROGRAM_RECORD_ID MarkDuplicates --PROGRAM_GROUP_NAME MarkDuplicates --READ_NAME_REGEX &lt;optimized capture of last three <span class="string">':'</span> separated fields as numeric values&gt; --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET <span class="literal">false</span> --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 2 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX <span class="literal">false</span> --CREATE_MD5_FILE <span class="literal">false</span> --GA4GH_CLIENT_SECRETS client_secrets.json --<span class="built_in">help</span> <span class="literal">false</span> --version <span class="literal">false</span> --showHidden <span class="literal">false</span> --USE_JDK_DEFLATER <span class="literal">false</span> --USE_JDK_INFLATER <span class="literal">false</span></span><br><span class="line">Nov 06, 2020 7:17:38 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine</span><br><span class="line">INFO: Failed to detect whether we are running on Google Compute Engine.</span><br><span class="line">[Fri Nov 06 19:17:38 CST 2020] Executing as eric@LAPTOP-90TPP2NM on Linux 4.4.0-18362-Microsoft amd64; OpenJDK 64-Bit Server VM 1.8.0_152-release-1056-b12; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: Version:4.1.9.0</span><br><span class="line">INFO    2020-11-06 19:17:38     MarkDuplicates  Start of doWork freeMemory: 430727552; totalMemory: 446169088; maxMemory: 1883242496</span><br><span class="line">INFO    2020-11-06 19:17:38     MarkDuplicates  Reading input file and constructing <span class="built_in">read</span> end information.</span><br><span class="line">INFO    2020-11-06 19:17:38     MarkDuplicates  Will retain up to 6823342 data points before spilling to disk.</span><br><span class="line">WARNING 2020-11-06 19:17:38     AbstractOpticalDuplicateFinderCommandLineProgram        A field field parsed out of a <span class="built_in">read</span> name was expected to contain an <span class="built_in">integer</span> and did not. Read name: SRR12846241.10302161. Cause: String <span class="string">'SRR12846241.10302161'</span> did not start with a parsable number.</span><br><span class="line">INFO    2020-11-06 19:17:49     MarkDuplicates  Read     1,000,000 records.  Elapsed time: 00:00:10s.  Time <span class="keyword">for</span> last 1,000,000:   10s.  Last <span class="built_in">read</span> position: chr17:17,169,090</span><br><span class="line">INFO    2020-11-06 19:17:49     MarkDuplicates  Tracking 2670 as yet unmatched pairs. 78 records <span class="keyword">in</span> RAM.</span><br><span class="line">INFO    2020-11-06 19:17:58     MarkDuplicates  Read     2,000,000 records.  Elapsed time: 00:00:19s.  Time <span class="keyword">for</span> last 1,000,000:    8s.  Last <span class="built_in">read</span> position: chr5:170,878,110</span><br><span class="line">INFO    2020-11-06 19:17:58     MarkDuplicates  Tracking 3897 as yet unmatched pairs. 134 records <span class="keyword">in</span> RAM.</span><br><span class="line">INFO    2020-11-06 19:18:02     MarkDuplicates  Read 2540180 records. 4018 pairs never matched.</span><br><span class="line">INFO    2020-11-06 19:18:03     MarkDuplicates  After buildSortedReadEndLists freeMemory: 720026592; totalMemory: 1068498944; maxMemory: 1883242496</span><br><span class="line">INFO    2020-11-06 19:18:03     MarkDuplicates  Will retain up to 58851328 duplicate indices before spilling to disk.</span><br><span class="line">INFO    2020-11-06 19:18:03     MarkDuplicates  Traversing <span class="built_in">read</span> pair information and detecting duplicates.</span><br><span class="line">INFO    2020-11-06 19:18:04     MarkDuplicates  Traversing fragment information and detecting duplicates.</span><br><span class="line">INFO    2020-11-06 19:18:04     MarkDuplicates  Sorting list of duplicate records.</span><br><span class="line">INFO    2020-11-06 19:18:05     MarkDuplicates  After generateDuplicateIndexes freeMemory: 1016276296; totalMemory: 1506803712; maxMemory: 1883242496</span><br><span class="line">INFO    2020-11-06 19:18:05     MarkDuplicates  Marking 22740 records as duplicates.</span><br><span class="line">INFO    2020-11-06 19:18:05     MarkDuplicates  Found 0 optical duplicate clusters.</span><br><span class="line">INFO    2020-11-06 19:18:05     MarkDuplicates  Reads are assumed to be ordered by: coordinate</span><br><span class="line">INFO    2020-11-06 19:18:23     MarkDuplicates  Writing complete. Closing input iterator.</span><br><span class="line">INFO    2020-11-06 19:18:23     MarkDuplicates  Duplicate Index cleanup.</span><br><span class="line">INFO    2020-11-06 19:18:23     MarkDuplicates  Getting Memory Stats.</span><br><span class="line">INFO    2020-11-06 19:18:23     MarkDuplicates  Before output close freeMemory: 1489644800; totalMemory: 1512046592; maxMemory: 1883242496</span><br><span class="line">INFO    2020-11-06 19:18:23     MarkDuplicates  Closed outputs. Getting more Memory Stats.</span><br><span class="line">INFO    2020-11-06 19:18:23     MarkDuplicates  After output close freeMemory: 1519717896; totalMemory: 1537736704; maxMemory: 1883242496</span><br><span class="line">[Fri Nov 06 19:18:23 CST 2020] picard.sam.markduplicates.MarkDuplicates <span class="keyword">done</span>. Elapsed time: 0.77 minutes.</span><br><span class="line">Runtime.totalMemory()=1537736704</span><br><span class="line">Tool returned:</span><br><span class="line">0</span><br><span class="line"><span class="comment"># 加上索引文件</span></span><br><span class="line">samtools index /mnt/d/wes/sample/nodup/nodup_sam01.bam</span><br><span class="line"><span class="comment"># 查看文件大小</span></span><br><span class="line">total 216M</span><br><span class="line">-rwxrwxrwx 1 xxx xxx 3.5K Nov  6 19:18 metrics_sam01.txt</span><br><span class="line">-rwxrwxrwx 1 xxx xxx 216M Nov  6 19:18 nodup_sam01.bam</span><br></pre></td></tr></table></figure>
<h2 id="四、BQSR"><a href="#四、BQSR" class="headerlink" title="四、BQSR"></a>四、BQSR</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 第一步</span></span><br><span class="line">gatk BaseRecalibrator -I /mnt/d/wes/sample/nodup/nodup_sam01.bam -R /home/eric/ncbi/hg38/chroms/hg38.fa --known-sites /home/eric/ncbi/hg38/vcf/dbsnp_138.hg38.vcf.gz --known-sites /home/eric/ncbi/hg38/vcf/1000G_phase1.snps.high_confidence.hg38.vcf.gz --known-sites /home/eric/ncbi/hg38/vcf/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz -L /home/eric/ncbi/hg38/vcf/hg38_v0_HybSelOligos_whole_exome_illumina_coding_v1_whole_exome_illumina_coding_v1.Homo_sapiens_assembly38.targets.interval_list -O /mnt/d/wes/sample/bqsr/recal.table</span><br><span class="line"></span><br><span class="line">Using GATK jar /home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar</span><br><span class="line">Running:</span><br><span class="line">    java -Dsamjdk.use_async_io_read_samtools=<span class="literal">false</span> -Dsamjdk.use_async_io_write_samtools=<span class="literal">true</span> -Dsamjdk.use_async_io_write_tribble=<span class="literal">false</span> -Dsamjdk.compression_level=2 -jar /home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar BaseRecalibrator -I /mnt/d/wes/sample/nodup/nodup_sam01.bam -R /home/eric/ncbi/hg38/chroms/hg38.fa --known-sites /home/eric/ncbi/hg38/vcf/dbsnp_138.hg38.vcf.gz --known-sites /home/eric/ncbi/hg38/vcf/1000G_phase1.snps.high_confidence.hg38.vcf.gz --known-sites /home/eric/ncbi/hg38/vcf/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz -L /home/eric/ncbi/hg38/vcf/hg38_v0_HybSelOligos_whole_exome_illumina_coding_v1_whole_exome_illumina_coding_v1.Homo_sapiens_assembly38.targets.interval_list -O /mnt/d/wes/sample/bqsr/recal.table</span><br><span class="line">19:30:17.520 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_compression.so</span><br><span class="line">Nov 06, 2020 7:30:18 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine</span><br><span class="line">INFO: Failed to detect whether we are running on Google Compute Engine.</span><br><span class="line">19:30:18.041 INFO  BaseRecalibrator - ------------------------------------------------------------</span><br><span class="line">19:30:18.042 INFO  BaseRecalibrator - The Genome Analysis Toolkit (GATK) v4.1.9.0</span><br><span class="line">19:30:18.044 INFO  BaseRecalibrator - For support and documentation go to https://software.broadinstitute.org/gatk/</span><br><span class="line">19:30:18.046 INFO  BaseRecalibrator - Executing as eric@LAPTOP-90TPP2NM on Linux v4.4.0-18362-Microsoft amd64</span><br><span class="line">19:30:18.048 INFO  BaseRecalibrator - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_152-release-1056-b12</span><br><span class="line">19:30:18.050 INFO  BaseRecalibrator - Start Date/Time: November 6, 2020 7:30:17 PM CST</span><br><span class="line">19:30:18.052 INFO  BaseRecalibrator - ------------------------------------------------------------</span><br><span class="line">19:30:18.053 INFO  BaseRecalibrator - ------------------------------------------------------------</span><br><span class="line">19:30:18.057 INFO  BaseRecalibrator - HTSJDK Version: 2.23.0</span><br><span class="line">19:30:18.058 INFO  BaseRecalibrator - Picard Version: 2.23.3</span><br><span class="line">19:30:18.060 INFO  BaseRecalibrator - HTSJDK Defaults.COMPRESSION_LEVEL : 2</span><br><span class="line">19:30:18.061 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : <span class="literal">false</span></span><br><span class="line">19:30:18.063 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : <span class="literal">true</span></span><br><span class="line">19:30:18.064 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : <span class="literal">false</span></span><br><span class="line">19:30:18.067 INFO  BaseRecalibrator - Deflater: IntelDeflater</span><br><span class="line">19:30:18.069 INFO  BaseRecalibrator - Inflater: IntelInflater</span><br><span class="line">19:30:18.071 INFO  BaseRecalibrator - GCS max retries/reopens: 20</span><br><span class="line">19:30:18.073 INFO  BaseRecalibrator - Requester pays: disabled</span><br><span class="line">19:30:18.074 INFO  BaseRecalibrator - Initializing engine</span><br><span class="line">19:30:18.976 INFO  FeatureManager - Using codec VCFCodec to <span class="built_in">read</span> file file:///home/eric/ncbi/hg38/vcf/dbsnp_138.hg38.vcf.gz</span><br><span class="line">19:30:19.256 INFO  FeatureManager - Using codec VCFCodec to <span class="built_in">read</span> file file:///home/eric/ncbi/hg38/vcf/1000G_phase1.snps.high_confidence.hg38.vcf.gz</span><br><span class="line">19:30:19.439 INFO  FeatureManager - Using codec VCFCodec to <span class="built_in">read</span> file file:///home/eric/ncbi/hg38/vcf/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz</span><br><span class="line">19:30:19.691 INFO  FeatureManager - Using codec IntervalListCodec to <span class="built_in">read</span> file file:///home/eric/ncbi/hg38/vcf/hg38_v0_HybSelOligos_whole_exome_illumina_coding_v1_whole_exome_illumina_coding_v1.Homo_sapiens_assembly38.targets.interval_list</span><br><span class="line">19:30:24.305 INFO  IntervalArgumentCollection - Processing 38692858 bp from intervals</span><br><span class="line">19:30:24.465 INFO  BaseRecalibrator - Done initializing engine</span><br><span class="line">19:30:24.491 INFO  BaseRecalibrationEngine - The covariates being used here:</span><br><span class="line">19:30:24.493 INFO  BaseRecalibrationEngine -    ReadGroupCovariate</span><br><span class="line">19:30:24.496 INFO  BaseRecalibrationEngine -    QualityScoreCovariate</span><br><span class="line">19:30:24.497 INFO  BaseRecalibrationEngine -    ContextCovariate</span><br><span class="line">19:30:24.499 INFO  BaseRecalibrationEngine -    CycleCovariate</span><br><span class="line">19:30:24.520 INFO  ProgressMeter - Starting traversal</span><br><span class="line">19:30:24.524 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute</span><br><span class="line">19:30:34.676 INFO  ProgressMeter -        chr1:37721636              0.2                 45000         266009.9</span><br><span class="line">19:30:44.837 INFO  ProgressMeter -       chr1:153937400              0.3                108000         319023.2</span><br><span class="line">19:30:54.949 INFO  ProgressMeter -       chr1:232471505              0.5                164000         323428.9</span><br><span class="line">19:31:05.042 INFO  ProgressMeter -       chr10:96206871              0.7                224000         331712.6</span><br><span class="line">19:31:15.085 INFO  ProgressMeter -       chr11:57300855              0.8                282000         334651.9</span><br><span class="line">19:31:25.278 INFO  ProgressMeter -        chr12:2688514              1.0                343000         338748.7</span><br><span class="line">19:31:35.402 INFO  ProgressMeter -      chr12:100262225              1.2                402000         340307.9</span><br><span class="line">19:31:45.512 INFO  ProgressMeter -      chr13:111202974              1.3                456000         337832.0</span><br><span class="line">19:31:55.612 INFO  ProgressMeter -       chr15:28228255              1.5                514000         338581.1</span><br><span class="line">19:32:05.652 INFO  ProgressMeter -        chr16:1495337              1.7                572000         339375.2</span><br><span class="line">19:32:15.750 INFO  ProgressMeter -       chr16:88639021              1.9                634000         342012.5</span><br><span class="line">19:32:25.807 INFO  ProgressMeter -       chr17:58316342              2.0                700000         346300.4</span><br><span class="line">19:32:35.823 INFO  ProgressMeter -       chr18:79451785              2.2                756000         345473.7</span><br><span class="line">19:32:45.892 INFO  ProgressMeter -       chr19:40938908              2.4                822000         348881.6</span><br><span class="line">19:32:56.008 INFO  ProgressMeter -        chr2:46623835              2.5                884000         350138.3</span><br><span class="line">19:33:06.010 INFO  ProgressMeter -       chr2:169735426              2.7                946000         351489.9</span><br><span class="line">19:33:16.139 INFO  ProgressMeter -       chr20:20047689              2.9               1003000         350670.7</span><br><span class="line">19:33:26.142 INFO  ProgressMeter -       chr22:23061695              3.0               1058000         349526.8</span><br><span class="line">19:33:36.257 INFO  ProgressMeter -        chr3:48672416              3.2               1113000         348300.5</span><br><span class="line">19:33:46.412 INFO  ProgressMeter -       chr3:171093877              3.4               1168000         347126.6</span><br><span class="line">19:33:56.453 INFO  ProgressMeter -        chr4:99342797              3.5               1220000         345400.3</span><br><span class="line">19:34:06.491 INFO  ProgressMeter -        chr5:79030744              3.7               1273000         344106.8</span><br><span class="line">19:34:16.538 INFO  ProgressMeter -        chr6:10898046              3.9               1327000         343170.4</span><br><span class="line">19:34:26.588 INFO  ProgressMeter -       chr6:136558746              4.0               1379000         341811.8</span><br><span class="line">19:34:36.722 INFO  ProgressMeter -        chr7:87836098              4.2               1434000         341161.9</span><br><span class="line">19:34:46.961 INFO  ProgressMeter -        chr8:24321118              4.4               1490000         340654.5</span><br><span class="line">19:34:57.099 INFO  ProgressMeter -         chr9:5811124              4.5               1536000         338110.0</span><br><span class="line">19:35:07.197 INFO  ProgressMeter -       chr9:132014280              4.7               1593000         338130.4</span><br><span class="line">19:35:16.870 INFO  BaseRecalibrator - 0 <span class="built_in">read</span>(s) filtered by: MappingQualityNotZeroReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: MappingQualityAvailableReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: MappedReadFilter</span><br><span class="line">425 <span class="built_in">read</span>(s) filtered by: NotSecondaryAlignmentReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: NotDuplicateReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: PassesVendorQualityCheckReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: WellformedReadFilter</span><br><span class="line">425 total reads filtered</span><br><span class="line">19:35:16.886 INFO  ProgressMeter -        chrY:13260230              4.9               1646850         337976.0</span><br><span class="line">19:35:16.888 INFO  ProgressMeter - Traversal complete. Processed 1646850 total reads <span class="keyword">in</span> 4.9 minutes.</span><br><span class="line">19:35:17.006 INFO  BaseRecalibrator - Calculating quantized quality scores...</span><br><span class="line">19:35:17.053 INFO  BaseRecalibrator - Writing recalibration report...</span><br><span class="line">19:35:17.663 INFO  BaseRecalibrator - ...<span class="keyword">done</span>!</span><br><span class="line">19:35:17.712 INFO  BaseRecalibrator - BaseRecalibrator was able to recalibrate 1646850 reads</span><br><span class="line">19:35:17.753 INFO  BaseRecalibrator - Shutting down engine</span><br><span class="line">[November 6, 2020 7:35:17 PM CST] org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator <span class="keyword">done</span>. Elapsed time: 5.01 minutes.</span><br><span class="line">Runtime.totalMemory()=876609536</span><br><span class="line">Tool returned:</span><br><span class="line">SUCCESS</span><br><span class="line"></span><br><span class="line"><span class="comment"># 第二步</span></span><br><span class="line">gatk ApplyBQSR -I /mnt/d/wes/sample/nodup/nodup_sam01.bam -R /home/eric/ncbi/hg38/chroms/hg38.fa --bqsr-recal-file /mnt/d/wes/sample/bqsr/recal.table -O /mnt/d/wes/sample/bqsr/bqsr_sam01.bam</span><br><span class="line"></span><br><span class="line">Using GATK jar /home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar</span><br><span class="line">Running:</span><br><span class="line">    java -Dsamjdk.use_async_io_read_samtools=<span class="literal">false</span> -Dsamjdk.use_async_io_write_samtools=<span class="literal">true</span> -Dsamjdk.use_async_io_write_tribble=<span class="literal">false</span> -Dsamjdk.compression_level=2 -jar /home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar ApplyBQSR -I /mnt/d/wes/sample/nodup/nodup_sam01.bam -R /home/eric/ncbi/hg38/chroms/hg38.fa --bqsr-recal-file /mnt/d/wes/sample/bqsr/recal.table -O /mnt/d/wes/sample/bqsr/bqsr_sam01.bam</span><br><span class="line">19:38:44.345 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_compression.so</span><br><span class="line">Nov 06, 2020 7:38:44 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine</span><br><span class="line">INFO: Failed to detect whether we are running on Google Compute Engine.</span><br><span class="line">19:38:44.928 INFO  ApplyBQSR - ------------------------------------------------------------</span><br><span class="line">19:38:44.929 INFO  ApplyBQSR - The Genome Analysis Toolkit (GATK) v4.1.9.0</span><br><span class="line">19:38:44.931 INFO  ApplyBQSR - For support and documentation go to https://software.broadinstitute.org/gatk/</span><br><span class="line">19:38:44.934 INFO  ApplyBQSR - Executing as eric@LAPTOP-90TPP2NM on Linux v4.4.0-18362-Microsoft amd64</span><br><span class="line">19:38:44.936 INFO  ApplyBQSR - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_152-release-1056-b12</span><br><span class="line">19:38:44.938 INFO  ApplyBQSR - Start Date/Time: November 6, 2020 7:38:44 PM CST</span><br><span class="line">19:38:44.939 INFO  ApplyBQSR - ------------------------------------------------------------</span><br><span class="line">19:38:44.942 INFO  ApplyBQSR - ------------------------------------------------------------</span><br><span class="line">19:38:44.949 INFO  ApplyBQSR - HTSJDK Version: 2.23.0</span><br><span class="line">19:38:44.950 INFO  ApplyBQSR - Picard Version: 2.23.3</span><br><span class="line">19:38:44.957 INFO  ApplyBQSR - HTSJDK Defaults.COMPRESSION_LEVEL : 2</span><br><span class="line">19:38:44.959 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : <span class="literal">false</span></span><br><span class="line">19:38:44.961 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : <span class="literal">true</span></span><br><span class="line">19:38:44.962 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : <span class="literal">false</span></span><br><span class="line">19:38:44.963 INFO  ApplyBQSR - Deflater: IntelDeflater</span><br><span class="line">19:38:44.965 INFO  ApplyBQSR - Inflater: IntelInflater</span><br><span class="line">19:38:44.967 INFO  ApplyBQSR - GCS max retries/reopens: 20</span><br><span class="line">19:38:44.972 INFO  ApplyBQSR - Requester pays: disabled</span><br><span class="line">19:38:44.973 INFO  ApplyBQSR - Initializing engine</span><br><span class="line">19:38:45.850 INFO  ApplyBQSR - Done initializing engine</span><br><span class="line">19:38:45.930 INFO  ProgressMeter - Starting traversal</span><br><span class="line">19:38:45.937 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute</span><br><span class="line">19:38:55.958 INFO  ProgressMeter -       chr1:235328572              0.2                249000        1491315.6</span><br><span class="line">19:39:05.958 INFO  ProgressMeter -       chr12:96684881              0.3                597000        1789300.2</span><br><span class="line">19:39:15.963 INFO  ProgressMeter -        chr17:4901462              0.5                969000        1936450.8</span><br><span class="line">19:39:25.972 INFO  ProgressMeter -        chr2:96187143              0.7               1348000        2020383.7</span><br><span class="line">19:39:35.978 INFO  ProgressMeter -       chr3:142790913              0.8               1735000        2080377.3</span><br><span class="line">19:39:45.992 INFO  ProgressMeter -        chr7:21588589              1.0               2123000        2121126.3</span><br><span class="line">19:39:55.994 INFO  ProgressMeter -       chrX:153961533              1.2               2509000        2148883.0</span><br><span class="line">19:39:56.224 INFO  ApplyBQSR - 0 <span class="built_in">read</span>(s) filtered by: WellformedReadFilter</span><br><span class="line"></span><br><span class="line">19:39:56.227 INFO  ProgressMeter -        chrY:25168958              1.2               2517440        2148964.3</span><br><span class="line">19:39:56.228 INFO  ProgressMeter - Traversal complete. Processed 2517440 total reads <span class="keyword">in</span> 1.2 minutes.</span><br><span class="line">19:39:56.277 INFO  ApplyBQSR - Shutting down engine</span><br><span class="line">[November 6, 2020 7:39:56 PM CST] org.broadinstitute.hellbender.tools.walkers.bqsr.ApplyBQSR <span class="keyword">done</span>. Elapsed time: 1.20 minutes.</span><br><span class="line">Runtime.totalMemory()=617086976</span><br><span class="line"><span class="comment"># 查看文件</span></span><br><span class="line">total 421M</span><br><span class="line">-rwxrwxrwx 1 xxx xxx 3.8M Nov  6 19:39 bqsr_sam01.bai</span><br><span class="line">-rwxrwxrwx 1 xxx xxx 417M Nov  6 19:39 bqsr_sam01.bam</span><br><span class="line"><span class="comment"># 现在，BQSR完成后，就得到Analysis-Ready Reads</span></span><br></pre></td></tr></table></figure>
<h2 id="五、变异检测"><a href="#五、变异检测" class="headerlink" title="五、变异检测"></a>五、变异检测</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br></pre></td><td class="code"><pre><span class="line">gatk HaplotypeCaller -R /home/eric/ncbi/hg38/chroms/hg38.fa -I /mnt/d/wes/sample/bqsr/bqsr_sam01.bam -O /mnt/d/wes/sample/<span class="built_in">caller</span>/output.g.vcf.gz -ERC GVCF</span><br><span class="line"><span class="comment"># 这一步比较耗时</span></span><br><span class="line">22:06:51.680 INFO  HaplotypeCaller - 17268 <span class="built_in">read</span>(s) filtered by: MappingQualityReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: MappingQualityAvailableReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: MappedReadFilter</span><br><span class="line">1538 <span class="built_in">read</span>(s) filtered by: NotSecondaryAlignmentReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: NotDuplicateReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: PassesVendorQualityCheckReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: GoodCigarReadFilter</span><br><span class="line">0 <span class="built_in">read</span>(s) filtered by: WellformedReadFilter</span><br><span class="line">18806 total reads filtered</span><br><span class="line">22:06:51.872 INFO  ProgressMeter - chrY_KI270740v1_random:36001             86.3              10857084         125749.4</span><br><span class="line">22:06:51.881 INFO  ProgressMeter - Traversal complete. Processed 10857084 total regions <span class="keyword">in</span> 86.3 minutes.</span><br><span class="line">22:06:52.209 INFO  PairHMM - Total compute time <span class="keyword">in</span> PairHMM computeLogLikelihoods() : 298.5258293</span><br><span class="line">22:06:52.233 INFO  SmithWatermanAligner - Total compute time <span class="keyword">in</span> java Smith-Waterman : 131.45 sec</span><br><span class="line">22:06:52.234 INFO  HaplotypeCaller - Shutting down engine</span><br><span class="line">[November 6, 2020 10:06:52 PM CST] org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller <span class="keyword">done</span>. Elapsed time: 86.37 minutes.</span><br><span class="line">Runtime.totalMemory()=1270349824</span><br><span class="line"><span class="comment"># 下面转化为vcf.gz</span></span><br><span class="line">gatk GenotypeGVCFs -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/sample/<span class="built_in">caller</span>/output.g.vcf.gz -O /mnt/d/wes/sample/<span class="built_in">caller</span>/output.vcf.gz</span><br><span class="line"><span class="comment"># 这个过程很快</span></span><br><span class="line">Using GATK jar /home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar</span><br><span class="line">Running:</span><br><span class="line">    java -Dsamjdk.use_async_io_read_samtools=<span class="literal">false</span> -Dsamjdk.use_async_io_write_samtools=<span class="literal">true</span> -Dsamjdk.use_async_io_write_tribble=<span class="literal">false</span> -Dsamjdk.compression_level=2 -jar /home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar GenotypeGVCFs -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/sample/<span class="built_in">caller</span>/output.g.vcf.gz -O /mnt/d/wes/sample/<span class="built_in">caller</span>/output.vcf.gz</span><br><span class="line">22:14:25.026 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/eric/miniconda3/share/gatk4-4.1.9.0-0/gatk-package-4.1.9.0-local.jar!/com/intel/gkl/native/libgkl_compression.so</span><br><span class="line">Nov 06, 2020 10:14:25 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine</span><br><span class="line">INFO: Failed to detect whether we are running on Google Compute Engine.</span><br><span class="line">22:14:25.679 INFO  GenotypeGVCFs - ------------------------------------------------------------</span><br><span class="line">22:14:25.687 INFO  GenotypeGVCFs - The Genome Analysis Toolkit (GATK) v4.1.9.0</span><br><span class="line">22:14:25.688 INFO  GenotypeGVCFs - For support and documentation go to https://software.broadinstitute.org/gatk/</span><br><span class="line">22:14:25.691 INFO  GenotypeGVCFs - Executing as eric@LAPTOP-90TPP2NM on Linux v4.4.0-18362-Microsoft amd64</span><br><span class="line">22:14:25.693 INFO  GenotypeGVCFs - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_152-release-1056-b12</span><br><span class="line">22:14:25.697 INFO  GenotypeGVCFs - Start Date/Time: November 6, 2020 10:14:24 PM CST</span><br><span class="line">22:14:25.700 INFO  GenotypeGVCFs - ------------------------------------------------------------</span><br><span class="line">22:14:25.703 INFO  GenotypeGVCFs - ------------------------------------------------------------</span><br><span class="line">22:14:25.705 INFO  GenotypeGVCFs - HTSJDK Version: 2.23.0</span><br><span class="line">22:14:25.706 INFO  GenotypeGVCFs - Picard Version: 2.23.3</span><br><span class="line">22:14:25.709 INFO  GenotypeGVCFs - HTSJDK Defaults.COMPRESSION_LEVEL : 2</span><br><span class="line">22:14:25.711 INFO  GenotypeGVCFs - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : <span class="literal">false</span></span><br><span class="line">22:14:25.717 INFO  GenotypeGVCFs - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : <span class="literal">true</span></span><br><span class="line">22:14:25.720 INFO  GenotypeGVCFs - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : <span class="literal">false</span></span><br><span class="line">22:14:25.722 INFO  GenotypeGVCFs - Deflater: IntelDeflater</span><br><span class="line">22:14:25.724 INFO  GenotypeGVCFs - Inflater: IntelInflater</span><br><span class="line">22:14:25.728 INFO  GenotypeGVCFs - GCS max retries/reopens: 20</span><br><span class="line">22:14:25.729 INFO  GenotypeGVCFs - Requester pays: disabled</span><br><span class="line">22:14:25.734 INFO  GenotypeGVCFs - Initializing engine</span><br><span class="line">22:14:26.642 INFO  FeatureManager - Using codec VCFCodec to <span class="built_in">read</span> file file:///mnt/d/wes/sample/<span class="built_in">caller</span>/output.g.vcf.gz</span><br><span class="line">22:14:27.083 INFO  GenotypeGVCFs - Done initializing engine</span><br><span class="line">22:14:27.249 INFO  ProgressMeter - Starting traversal</span><br><span class="line">22:14:27.261 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute</span><br><span class="line">22:14:27.459 WARN  ReferenceConfidenceVariantContextMerger - Detected invalid annotations: When trying to merge variant contexts at location chr1:14653 the annotation MLEAC=[0, 0] was not a numerical value and was ignored</span><br><span class="line">22:14:27.707 WARN  InbreedingCoeff - InbreedingCoeff will not be calculated; at least 10 samples must have called genotypes</span><br><span class="line">22:14:37.332 INFO  ProgressMeter -        chr1:19155008              0.2                 82000         488725.5</span><br><span class="line">22:14:47.369 INFO  ProgressMeter -        chr1:81936862              0.3                255000         761042.6</span><br><span class="line">22:14:57.397 INFO  ProgressMeter -       chr1:201005243              0.5                520000        1035478.4</span><br><span class="line">22:15:07.551 INFO  ProgressMeter -       chr10:48156213              0.7                728000        1084247.6</span><br><span class="line">22:15:17.572 INFO  ProgressMeter -       chr11:10800804              0.8                959000        1143777.2</span><br><span class="line">22:15:27.591 INFO  ProgressMeter -      chr11:129436951              1.0               1238000        1231330.3</span><br><span class="line">22:15:37.603 INFO  ProgressMeter -      chr12:125957935              1.2               1553000        1324746.2</span><br><span class="line">22:15:47.616 INFO  ProgressMeter -       chr14:80761356              1.3               1831000        1367251.2</span><br><span class="line">22:15:57.654 INFO  ProgressMeter -       chr16:10649222              1.5               2159000        1433139.0</span><br><span class="line">22:16:07.676 INFO  ProgressMeter -       chr17:21318468              1.7               2465000        1472946.2</span><br><span class="line">22:16:17.746 INFO  ProgressMeter -       chr18:31086752              1.8               2744000        1490211.0</span><br><span class="line">22:16:27.768 INFO  ProgressMeter -       chr19:46527144              2.0               3066000        1526601.0</span><br><span class="line">22:16:37.774 INFO  ProgressMeter -       chr2:100864558              2.2               3365000        1547019.7</span><br><span class="line">22:16:47.780 INFO  ProgressMeter -       chr21:10473621              2.3               3816000        1629434.6</span><br><span class="line">22:16:57.794 INFO  ProgressMeter -       chr3:151326235              2.5               4320000        1721927.3</span><br><span class="line">22:17:07.829 INFO  ProgressMeter -       chr5:108575621              2.7               4796000        1792182.6</span><br><span class="line">22:17:17.874 INFO  ProgressMeter -        chr7:17796878              2.8               5264000        1851250.5</span><br><span class="line">22:17:27.887 INFO  ProgressMeter -        chr8:47824115              3.0               5638000        1872861.6</span><br><span class="line">22:17:37.893 INFO  ProgressMeter -       chrX:130165728              3.2               6174000        1943261.2</span><br><span class="line">22:17:38.980 INFO  ProgressMeter -        chrY:56836134              3.2               6220164        1946690.9</span><br><span class="line">22:17:38.981 INFO  ProgressMeter - Traversal complete. Processed 6220164 total variants <span class="keyword">in</span> 3.2 minutes.</span><br><span class="line">22:17:39.095 INFO  GenotypeGVCFs - Shutting down engine</span><br><span class="line">[November 6, 2020 10:17:39 PM CST] org.broadinstitute.hellbender.tools.walkers.GenotypeGVCFs <span class="keyword">done</span>. Elapsed time: 3.24 minutes.</span><br><span class="line">Runtime.totalMemory()=534773760</span><br><span class="line"></span><br><span class="line"><span class="comment"># 为理解下面的步骤，我们来看一下这个文件中的第一条记录：</span></span><br><span class="line">CHROM: chr1</span><br><span class="line">POS: 15903   </span><br><span class="line">ID: .     </span><br><span class="line">REF: G    </span><br><span class="line">ALT: GC    </span><br><span class="line">QUAL: 59.28   </span><br><span class="line">FILTER: . </span><br><span class="line">INFO: AC=2;AF=1.00;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=30.13;QD=29.64;SOR=2.303   </span><br><span class="line">FORMAT: GT:AD:DP:GQ:PL </span><br><span class="line">sample1: 1/1:0,2:2:6:71,6,0</span><br></pre></td></tr></table></figure>
<h2 id="六、位点注释"><a href="#六、位点注释" class="headerlink" title="六、位点注释"></a>六、位点注释</h2><p>位点注释需要用到三个工具：<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360050815552-CNNScoreVariants" target="_blank" rel="noopener">CNNScoreVariants</a>和<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360051308071-FilterVariantTranches" target="_blank" rel="noopener">FilterVariantTranches</a>，以及<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360051304411-Funcotator" target="_blank" rel="noopener">Funcotator</a></p>
<p>CNNScoreVariants是指使用CNN (Convolutional Neural Network)的分数对VCF文件进行注释，这个工具需要调用PYTHON程序，因此，需要预先搭建环境，最简单不易出错的方法是使用Docker，关于Docker的搭建及测试会专门写一篇文章介绍。</p>
<h3 id="6-1-启动Docker"><a href="#6-1-启动Docker" class="headerlink" title="6.1 启动Docker"></a>6.1 启动Docker</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 由于在Docker内部无法调用外部文件，因此，需要将外部文件挂载在Docker中，如果把Docker看作是一个独立的操作系统，挂载后的外部文件即相当于Docker内部的一个盘符，方便调用，在此，使用-V参数将D盘挂载到Docker下，并命名为my_data目录，路径为"/gatk/my_data/"；同时，使用-it参数启动gatk。</span></span><br><span class="line">$ docker run -v /mnt/d:/gatk/my_data -it broadinstitute/gatk:4.1.3.0</span><br><span class="line"></span><br><span class="line">(gatk) root@9f0ac1b25dec:/gatk<span class="comment">#</span></span><br></pre></td></tr></table></figure>
<h3 id="6-2-使用CNNScoreVariants进行评分"><a href="#6-2-使用CNNScoreVariants进行评分" class="headerlink" title="6.2 使用CNNScoreVariants进行评分"></a>6.2 使用CNNScoreVariants进行评分</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br></pre></td><td class="code"><pre><span class="line">gatk CNNScoreVariants -V /gatk/my_data/wes/sample/<span class="built_in">caller</span>/output.vcf.gz -R /gatk/my_data/ncbi/hg38/chroms/hg38.fa -O /gatk/my_data/wes/sample/annotate/annotated.vcf --<span class="built_in">disable</span>-avx-check <span class="literal">true</span></span><br><span class="line"><span class="comment"># --disable-avx-check true这个参数根据需要使用，如果系统中AVX的版本低于1.6，可以使用此参数跳过检查，否则会报错。</span></span><br><span class="line"></span><br><span class="line">Running:</span><br><span class="line">    java -Dsamjdk.use_async_io_read_samtools=<span class="literal">false</span> -Dsamjdk.use_async_io_write_samtools=<span class="literal">true</span> -Dsamjdk.use_async_io_write_tribble=<span class="literal">false</span> -Dsamjdk.compression_level=2 -jar /gatk/gatk-package-4.1.3.0-local.jar CNNScoreVariants -V /gatk/my_data/wes/sample/<span class="built_in">caller</span>/output.vcf.gz -R /gatk/my_data/ncbi/hg38/chroms/hg38.fa -O /gatk/my_data/wes/sample/annotate/annotated.vcf --<span class="built_in">disable</span>-avx-check <span class="literal">true</span></span><br><span class="line">22:13:19.307 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.1.3.0-local.jar!/com/intel/gkl/native/libgkl_compression.so</span><br><span class="line">Nov 11, 2020 10:13:21 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine</span><br><span class="line">INFO: Failed to detect whether we are running on Google Compute Engine.</span><br><span class="line">22:13:21.208 INFO  CNNScoreVariants - ------------------------------------------------------------</span><br><span class="line">22:13:21.209 INFO  CNNScoreVariants - The Genome Analysis Toolkit (GATK) v4.1.3.0</span><br><span class="line">22:13:21.209 INFO  CNNScoreVariants - For support and documentation go to https://software.broadinstitute.org/gatk/</span><br><span class="line">22:13:21.212 INFO  CNNScoreVariants - Executing as root@9f0ac1b25dec on Linux v4.19.128-microsoft-standard amd64</span><br><span class="line">22:13:21.213 INFO  CNNScoreVariants - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_191-8u191-b12-0ubuntu0.16.04.1-b12</span><br><span class="line">22:13:21.213 INFO  CNNScoreVariants - Start Date/Time: November 11, 2020 10:13:19 PM UTC</span><br><span class="line">22:13:21.214 INFO  CNNScoreVariants - ------------------------------------------------------------</span><br><span class="line">22:13:21.214 INFO  CNNScoreVariants - ------------------------------------------------------------</span><br><span class="line">22:13:21.217 INFO  CNNScoreVariants - HTSJDK Version: 2.20.1</span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - Picard Version: 2.20.5</span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2</span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : <span class="literal">false</span></span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : <span class="literal">true</span></span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : <span class="literal">false</span></span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - Deflater: IntelDeflater</span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - Inflater: IntelInflater</span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - GCS max retries/reopens: 20</span><br><span class="line">22:13:21.220 INFO  CNNScoreVariants - Requester pays: disabled</span><br><span class="line">22:13:21.221 INFO  CNNScoreVariants - Initializing engine</span><br><span class="line">22:13:22.011 INFO  FeatureManager - Using codec VCFCodec to <span class="built_in">read</span> file file:///gatk/my_data/wes/sample/<span class="built_in">caller</span>/output.vcf.gz</span><br><span class="line">22:13:22.334 INFO  CNNScoreVariants - Done initializing engine</span><br><span class="line">22:13:37.467 INFO  CNNScoreVariants - Using key:CNN_1D <span class="keyword">for</span> CNN architecture:/tmp/1d_cnn_mix_train_full_bn.4399861944292417098.json and weights:/tmp/1d_cnn_mix_train_full_bn.6480348361258943885.hd5</span><br><span class="line">22:13:42.239 INFO  ProgressMeter - Starting traversal</span><br><span class="line">22:13:42.239 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute</span><br><span class="line">22:13:42.247 INFO  CNNScoreVariants - Starting pass 0 through the variants</span><br><span class="line">22:14:02.248 INFO  ProgressMeter -        chr1:87045715              0.3                  3000           9000.0</span><br><span class="line">22:14:13.936 INFO  ProgressMeter -       chr1:147029254              0.5                  4000           7571.9</span><br><span class="line">22:14:25.604 INFO  ProgressMeter -       chr1:169397278              0.7                  5000           6918.2</span><br><span class="line">22:14:38.391 INFO  ProgressMeter -       chr1:214663767              0.9                  6000           6411.3</span><br><span class="line">22:14:49.750 INFO  ProgressMeter -        chr10:3082324              1.1                  7000           6221.3</span><br><span class="line">22:15:02.307 INFO  ProgressMeter -       chr10:45759533              1.3                  8000           5995.1</span><br><span class="line">22:15:15.155 INFO  ProgressMeter -       chr10:93404283              1.5                  9000           5811.8</span><br><span class="line">22:15:26.166 INFO  ProgressMeter -      chr10:133168333              1.7                 10000           5773.3</span><br><span class="line">22:15:37.367 INFO  ProgressMeter -       chr11:18255670              1.9                 11000           5732.8</span><br><span class="line">22:15:49.865 INFO  ProgressMeter -       chr11:63490944              2.1                 12000           5641.5</span><br><span class="line">22:16:02.650 INFO  ProgressMeter -      chr11:101593990              2.3                 13000           5555.2</span><br><span class="line">22:16:13.848 INFO  ProgressMeter -        chr12:2786919              2.5                 14000           5540.6</span><br><span class="line">22:16:25.342 INFO  ProgressMeter -       chr12:31102025              2.7                 15000           5518.0</span><br><span class="line">22:16:41.195 INFO  ProgressMeter -       chr12:71769943              3.0                 16000           5364.5</span><br><span class="line">22:16:53.911 INFO  ProgressMeter -      chr12:122986425              3.2                 17000           5321.6</span><br><span class="line">22:17:05.693 INFO  ProgressMeter -       chr13:35044829              3.4                 18000           5308.4</span><br><span class="line">22:17:17.926 INFO  ProgressMeter -       chr14:18656315              3.6                 19000           5285.5</span><br><span class="line">22:17:29.413 INFO  ProgressMeter -       chr14:61865740              3.8                 20000           5282.3</span><br><span class="line">22:17:40.900 INFO  ProgressMeter - chr14_GL000225v1_random:23601              4.0                 21000           5279.5</span><br><span class="line">22:17:52.338 INFO  ProgressMeter -       chr15:51215311              4.2                 22000           5277.9</span><br><span class="line">22:18:03.518 INFO  ProgressMeter -       chr15:78796665              4.4                 23000           5281.7</span><br><span class="line">22:18:14.921 INFO  ProgressMeter -        chr16:3483577              4.5                 24000           5280.9</span><br><span class="line">22:18:26.399 INFO  ProgressMeter -       chr16:34594921              4.7                 25000           5278.7</span><br><span class="line">22:18:40.819 INFO  ProgressMeter -       chr16:84976364              5.0                 26000           5224.7</span><br><span class="line">22:18:57.583 INFO  ProgressMeter -       chr17:10640146              5.3                 27000           5137.3</span><br><span class="line">22:19:09.159 INFO  ProgressMeter -       chr17:40023414              5.4                 28000           5138.9</span><br><span class="line">22:19:20.628 INFO  ProgressMeter -       chr17:70095791              5.6                 29000           5142.0</span><br><span class="line">22:19:35.491 INFO  ProgressMeter -       chr18:11639357              5.9                 30000           5095.5</span><br><span class="line">01:21:13.590 INFO  ProgressMeter -         chr19:966458            187.5                 31000            165.3</span><br><span class="line">01:21:28.856 INFO  ProgressMeter -       chr19:12440901            187.8                 32000            170.4</span><br><span class="line">01:21:41.329 INFO  ProgressMeter -       chr19:38505335            188.0                 33000            175.5</span><br><span class="line">01:21:52.685 INFO  ProgressMeter -       chr19:51594208            188.2                 34000            180.7</span><br><span class="line">01:22:04.454 INFO  ProgressMeter -        chr2:23887741            188.4                 35000            185.8</span><br><span class="line">01:22:16.139 INFO  ProgressMeter -        chr2:72465143            188.6                 36000            190.9</span><br><span class="line">01:22:27.642 INFO  ProgressMeter -       chr2:112531455            188.8                 37000            196.0</span><br><span class="line">01:22:39.186 INFO  ProgressMeter -       chr2:165155693            188.9                 38000            201.1</span><br><span class="line">01:22:50.654 INFO  ProgressMeter -       chr2:208338389            189.1                 39000            206.2</span><br><span class="line">01:23:02.084 INFO  ProgressMeter -        chr20:1578455            189.3                 40000            211.3</span><br><span class="line">01:23:13.819 INFO  ProgressMeter -       chr20:38000451            189.5                 41000            216.3</span><br><span class="line">01:23:25.157 INFO  ProgressMeter -       chr21:14304122            189.7                 42000            221.4</span><br><span class="line">01:23:37.058 INFO  ProgressMeter -       chr22:36261582            189.9                 44000            231.7</span><br><span class="line">01:23:48.698 INFO  ProgressMeter -        chr3:15475403            190.1                 45000            236.7</span><br><span class="line">01:24:00.525 INFO  ProgressMeter -        chr3:57577393            190.3                 46000            241.7</span><br><span class="line">01:24:12.212 INFO  ProgressMeter -       chr3:117027527            190.5                 47000            246.7</span><br><span class="line">01:24:23.981 INFO  ProgressMeter -       chr3:171792652            190.7                 48000            251.7</span><br><span class="line">01:24:35.067 INFO  ProgressMeter -         chr4:7025314            190.9                 49000            256.7</span><br><span class="line">01:24:47.070 INFO  ProgressMeter -        chr4:68063022            191.1                 50000            261.7</span><br><span class="line">01:24:58.504 INFO  ProgressMeter -       chr4:125476979            191.3                 51000            266.6</span><br><span class="line">01:25:13.077 INFO  ProgressMeter -         chr5:6611870            191.5                 52000            271.5</span><br><span class="line">01:25:25.667 INFO  ProgressMeter -        chr5:79730716            191.7                 53000            276.4</span><br><span class="line">01:25:50.363 INFO  ProgressMeter -       chr5:149028538            192.1                 54000            281.1</span><br><span class="line">01:26:03.247 INFO  ProgressMeter -        chr6:10047168            192.4                 55000            285.9</span><br><span class="line">01:26:16.475 INFO  ProgressMeter -        chr6:52813539            192.6                 56000            290.8</span><br><span class="line">01:26:28.586 INFO  ProgressMeter -       chr6:131950089            192.8                 57000            295.7</span><br><span class="line">01:26:40.160 INFO  ProgressMeter -         chr7:2278738            193.0                 58000            300.6</span><br><span class="line">01:26:51.860 INFO  ProgressMeter -        chr7:47885733            193.2                 59000            305.4</span><br><span class="line">01:27:03.179 INFO  ProgressMeter -        chr7:88163268            193.3                 60000            310.3</span><br><span class="line">01:27:14.805 INFO  ProgressMeter -       chr7:129734366            193.5                 61000            315.2</span><br><span class="line">01:27:26.079 INFO  ProgressMeter -         chr8:8217007            193.7                 62000            320.0</span><br><span class="line">01:27:37.491 INFO  ProgressMeter -        chr8:43510803            193.9                 63000            324.9</span><br><span class="line">01:27:49.555 INFO  ProgressMeter -       chr8:136245034            194.1                 64000            329.7</span><br><span class="line">01:28:01.077 INFO  ProgressMeter -        chr9:39809861            194.3                 65000            334.5</span><br><span class="line">01:28:12.564 INFO  ProgressMeter -        chr9:93613446            194.5                 66000            339.3</span><br><span class="line">01:28:23.919 INFO  ProgressMeter -       chr9:131070323            194.7                 67000            344.1</span><br><span class="line">01:28:35.122 INFO  ProgressMeter - chrUn_KI270754v1:33254            194.9                 68000            348.9</span><br><span class="line">01:28:47.983 INFO  ProgressMeter -       chrX:145716091            195.1                 69000            353.7</span><br><span class="line">01:28:48.562 INFO  CNNScoreVariants - Finished pass 0 through the variants</span><br><span class="line">01:29:06.929 INFO  CNNScoreVariants - Starting pass 1 through the variants</span><br><span class="line">01:29:07.160 INFO  ProgressMeter -        chr1:13198782            195.4                 70000            358.2</span><br><span class="line">01:29:11.852 INFO  CNNScoreVariants - Finished pass 1 through the variants</span><br><span class="line">01:29:11.853 INFO  CNNScoreVariants - No variants filtered by: AllowAllVariantsVariantFilter</span><br><span class="line">01:29:11.856 INFO  CNNScoreVariants - No reads filtered by: AllowAllReadsReadFilter</span><br><span class="line">01:29:11.861 INFO  ProgressMeter -       chrX:101883466            195.5                138600            709.0</span><br><span class="line">01:29:11.862 INFO  ProgressMeter - Traversal complete. Processed 138600 total variants <span class="keyword">in</span> 195.5 minutes.</span><br><span class="line">01:29:11.863 INFO  CNNScoreVariants - Done scoring variants with CNN.</span><br><span class="line">01:29:12.315 INFO  CNNScoreVariants - Shutting down engine</span><br><span class="line">[November 12, 2020 1:29:12 AM UTC] org.broadinstitute.hellbender.tools.walkers.vqsr.CNNScoreVariants <span class="keyword">done</span>. Elapsed time: 195.88 minutes.</span><br><span class="line">Runtime.totalMemory()=241696768</span><br><span class="line"></span><br><span class="line"><span class="comment"># 此步生成的annotated.vcf作为下一步的输入文件，这个注释过的文件是在INFO字段中加入CNN_1D的评分，如"CNN_1D=-0.800"</span></span><br><span class="line"><span class="comment"># 跟输入的文件作个对比看看：</span></span><br><span class="line">CHROM: chr1</span><br><span class="line">POS: 15903   </span><br><span class="line">ID: .     </span><br><span class="line">REF: G    </span><br><span class="line">ALT: GC    </span><br><span class="line">QUAL: 59.28   </span><br><span class="line">FILTER: . </span><br><span class="line">INFO(输入): AC=2;AF=1.00;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=30.13;QD=29.64;SOR=2.303</span><br><span class="line">INFO(输出): AC=2;AF=1.00;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=30.13;QD=29.64;SOR=2.303;*CNN_1D=-0.800;*   </span><br><span class="line">FORMAT: GT:AD:DP:GQ:PL </span><br><span class="line">sample1: 1/1:0,2:2:6:71,6,0</span><br><span class="line"><span class="comment"># 两个文件的区别在于，输出文件在INFO字段加上CNN_1D评分，下一步将以此评分进行过滤</span></span><br></pre></td></tr></table></figure>
<h3 id="6-3-使用FilterVariantTranches过滤"><a href="#6-3-使用FilterVariantTranches过滤" class="headerlink" title="6.3 使用FilterVariantTranches过滤"></a>6.3 使用FilterVariantTranches过滤</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 这个工具根据注释后的VCF中的INFO字段的分数进行过滤，注释可以是CNNScoreVariants(CNNLOD)或VQSR (VQSLOD)</span></span><br><span class="line"></span><br><span class="line">gatk FilterVariantTranches \</span><br><span class="line">   -V input.vcf.gz \</span><br><span class="line">   --resource hapmap.vcf \</span><br><span class="line">   --resource mills.vcf \</span><br><span class="line">   --info-key CNN_1D \</span><br><span class="line">   --snp-tranche 99.95 \</span><br><span class="line">   --indel-tranche 99.4 \</span><br><span class="line">   -O filtered.vcf</span><br><span class="line"></span><br><span class="line"><span class="comment"># 在Docker中运行</span></span><br><span class="line"></span><br><span class="line">gatk FilterVariantTranches -V /gatk/my_data/wes/sample/annotate/annotated.vcf --resource /gatk/my_data/ncbi/hg38/vcf/hapmap_3.3.hg38.vcf.gz --resource /gatk/my_data/ncbi/hg38/vcf/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz --info-key CNN_1D --snp-tranche 99.95 --indel-tranche 99.4 -O /gatk/my_data/wes/sample/annotate/annotated.filtered.vcf</span><br><span class="line"></span><br><span class="line">Running:</span><br><span class="line">    java -Dsamjdk.use_async_io_read_samtools=<span class="literal">false</span> -Dsamjdk.use_async_io_write_samtools=<span class="literal">true</span> -Dsamjdk.use_async_io_write_tribble=<span class="literal">false</span> -Dsamjdk.compression_level=2 -jar /gatk/gatk-package-4.1.3.0-local.jar FilterVariantTranches -V /gatk/my_data/wes/sample/annotate/annotated.vcf --resource /gatk/my_data/ncbi/hg38/vcf/hapmap_3.3.hg38.vcf.gz --resource /gatk/my_data/ncbi/hg38/vcf/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz --info-key CNN_1D --snp-tranche 99.95 --indel-tranche 99.4 -O /gatk/my_data/wes/sample/annotate/annotated.filtered.vcf</span><br><span class="line">01:40:26.170 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.1.3.0-local.jar!/com/intel/gkl/native/libgkl_compression.so</span><br><span class="line">Nov 12, 2020 1:40:28 AM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine</span><br><span class="line">INFO: Failed to detect whether we are running on Google Compute Engine.</span><br><span class="line">01:40:28.454 INFO  FilterVariantTranches - ------------------------------------------------------------</span><br><span class="line">01:40:28.455 INFO  FilterVariantTranches - The Genome Analysis Toolkit (GATK) v4.1.3.0</span><br><span class="line">01:40:28.455 INFO  FilterVariantTranches - For support and documentation go to https://software.broadinstitute.org/gatk/</span><br><span class="line">01:40:28.456 INFO  FilterVariantTranches - Executing as root@9f0ac1b25dec on Linux v4.19.128-microsoft-standard amd64</span><br><span class="line">01:40:28.456 INFO  FilterVariantTranches - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_191-8u191-b12-0ubuntu0.16.04.1-b12</span><br><span class="line">01:40:28.456 INFO  FilterVariantTranches - Start Date/Time: November 12, 2020 1:40:26 AM UTC</span><br><span class="line">01:40:28.457 INFO  FilterVariantTranches - ------------------------------------------------------------</span><br><span class="line">01:40:28.457 INFO  FilterVariantTranches - ------------------------------------------------------------</span><br><span class="line">01:40:28.458 INFO  FilterVariantTranches - HTSJDK Version: 2.20.1</span><br><span class="line">01:40:28.458 INFO  FilterVariantTranches - Picard Version: 2.20.5</span><br><span class="line">01:40:28.458 INFO  FilterVariantTranches - HTSJDK Defaults.COMPRESSION_LEVEL : 2</span><br><span class="line">01:40:28.458 INFO  FilterVariantTranches - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : <span class="literal">false</span></span><br><span class="line">01:40:28.458 INFO  FilterVariantTranches - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : <span class="literal">true</span></span><br><span class="line">01:40:28.458 INFO  FilterVariantTranches - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : <span class="literal">false</span></span><br><span class="line">01:40:28.459 INFO  FilterVariantTranches - Deflater: IntelDeflater</span><br><span class="line">01:40:28.459 INFO  FilterVariantTranches - Inflater: IntelInflater</span><br><span class="line">01:40:28.459 INFO  FilterVariantTranches - GCS max retries/reopens: 20</span><br><span class="line">01:40:28.459 INFO  FilterVariantTranches - Requester pays: disabled</span><br><span class="line">01:40:28.459 INFO  FilterVariantTranches - Initializing engine</span><br><span class="line">01:40:29.423 INFO  FeatureManager - Using codec VCFCodec to <span class="built_in">read</span> file file:///gatk/my_data/ncbi/hg38/vcf/hapmap_3.3.hg38.vcf.gz</span><br><span class="line">01:40:30.099 INFO  FeatureManager - Using codec VCFCodec to <span class="built_in">read</span> file file:///gatk/my_data/ncbi/hg38/vcf/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz</span><br><span class="line">01:40:30.362 INFO  FeatureManager - Using codec VCFCodec to <span class="built_in">read</span> file file:///gatk/my_data/wes/sample/annotate/annotated.vcf</span><br><span class="line">01:40:30.630 INFO  FilterVariantTranches - Done initializing engine</span><br><span class="line">01:40:30.796 INFO  ProgressMeter - Starting traversal</span><br><span class="line">01:40:30.797 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute</span><br><span class="line">01:40:30.801 INFO  FilterVariantTranches - Starting pass 0 through the variants</span><br><span class="line">01:40:40.840 INFO  ProgressMeter -       chr17:40023414              0.2                 28000         167330.7</span><br><span class="line">01:40:50.889 INFO  ProgressMeter -       chr9:131070323              0.3                 67000         200099.6</span><br><span class="line">01:40:51.287 INFO  FilterVariantTranches - Finished pass 0 through the variants</span><br><span class="line">01:40:51.287 INFO  FilterVariantTranches - Found 57803 SNPs and 11479 indels with INFO score key:CNN_1D.</span><br><span class="line">01:40:51.287 INFO  FilterVariantTranches - Found 26323 SNPs and 3646 indels <span class="keyword">in</span> the resources.</span><br><span class="line">01:40:51.324 INFO  FilterVariantTranches - Starting pass 1 through the variants</span><br><span class="line">01:40:54.831 INFO  FilterVariantTranches - Finished pass 1 through the variants</span><br><span class="line">01:40:54.834 INFO  FilterVariantTranches - No variants filtered by: AllowAllVariantsVariantFilter</span><br><span class="line">01:40:54.836 INFO  FilterVariantTranches - No reads filtered by: AllowAllReadsReadFilter</span><br><span class="line">01:40:54.837 INFO  ProgressMeter -       chrX:101883466              0.4                138600         345952.2</span><br><span class="line">01:40:54.838 INFO  ProgressMeter - Traversal complete. Processed 138600 total variants <span class="keyword">in</span> 0.4 minutes.</span><br><span class="line">01:40:54.839 INFO  FilterVariantTranches - Filtered 1760 SNPs out of 57803 and filtered 188 indels out of 11479 with INFO score: CNN_1D.</span><br><span class="line">01:40:55.436 INFO  FilterVariantTranches - Shutting down engine</span><br><span class="line">[November 12, 2020 1:40:55 AM UTC] org.broadinstitute.hellbender.tools.walkers.vqsr.FilterVariantTranches <span class="keyword">done</span>. Elapsed time: 0.49 minutes.</span><br><span class="line">Runtime.totalMemory()=663748608</span><br><span class="line"></span><br><span class="line"><span class="comment"># 再看看输出文件的第一条记录，与前面的两条记录作对比：</span></span><br><span class="line">CHROM: chr1</span><br><span class="line">POS: 15903   </span><br><span class="line">ID: .     </span><br><span class="line">REF: G    </span><br><span class="line">ALT: GC    </span><br><span class="line">QUAL: 59.28   </span><br><span class="line">FILTER: PASS</span><br><span class="line">INFO(Step1): AC=2;AF=1.00;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=30.13;QD=29.64;SOR=2.303</span><br><span class="line">INFO(输入): AC=2;AF=1.00;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=30.13;QD=29.64;SOR=2.303;CNN_1D=-0.800; </span><br><span class="line">INFO(输出): AC=2;AF=1.00;AN=2;DP=3;ExcessHet=3.0103;FS=0.000;MLEAC=1;MLEAF=0.500;MQ=30.13;QD=29.64;SOR=2.303;CNN_1D=-0.800;</span><br><span class="line">FORMAT: GT:AD:DP:GQ:PL </span><br><span class="line">sample1: 1/1:0,2:2:6:71,6,0</span><br><span class="line"><span class="comment"># INFO字段没有变化，但FILTER字段由"."变为"PASS"</span></span><br></pre></td></tr></table></figure>
<h3 id="6-4-使用Funcotator进行功能注释"><a href="#6-4-使用Funcotator进行功能注释" class="headerlink" title="6.4 使用Funcotator进行功能注释"></a>6.4 使用Funcotator进行功能注释</h3><p>在使用Funcotator进行功能注释前，需要下载Data Sources。GATK有两套预封装数据源，分别对应germline和somatic，可以使用gatk FuncotatorDataSourceDownloader工具进行下载</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># For somatic data sources: (29GB)</span></span><br><span class="line">gatk FuncotatorDataSourceDownloader --somatic --validate-integrity --extract-after-download</span><br><span class="line"><span class="comment"># For germline data sources:</span></span><br><span class="line">gatk FuncotatorDataSourceDownloader --germline --validate-integrity --extract-after-download</span><br></pre></td></tr></table></figure>
<p>数据源下载完成后即可进行分析，语法示例：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">./gatk Funcotator \</span><br><span class="line">   -R reference.fasta \</span><br><span class="line">   -V input.vcf \</span><br><span class="line">   -O outputFile \</span><br><span class="line">   --output-file-format MAF \ <span class="comment">#可以选择VCF或MAF</span></span><br><span class="line">   --data-sources-path dataSourcesFolder/ \</span><br><span class="line">   --ref-version hg19</span><br></pre></td></tr></table></figure>
<p>在本例中，使用如下命令：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Germline Mutation</span></span><br><span class="line">gatk Funcotator -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/sample/annotate/annotated.filtered.vcf -O /mnt/d/wes/sample/annotate/annotated.filtered.funcotator.germline.vcf --output-file-format VCF --data-sources-path /mnt/d/ncbi/funcotator/funcotator_dataSources.v1.7.20200521g/ --ref-version hg38</span><br><span class="line"></span><br><span class="line"><span class="comment"># Somatic Mutation</span></span><br><span class="line">gatk Funcotator -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/sample/annotate/annotated.filtered.vcf -O /mnt/d/wes/sample/annotate/annotated.filtered.funcotator.somatic.vcf --output-file-format VCF --data-sources-path /mnt/d/ncbi/funcotator/funcotator_dataSources.v1.7.20200521s/ --ref-version hg38</span><br></pre></td></tr></table></figure>
<p>至此，我们得到了注释好的VCF文件，下面，就可以进行后续的功能分析了。</p>

      
    </div>

<div>


<ul class="post-copyright">
<li class="post-copyright-author">
<strong>本文作者：</strong>括囊无誉
</li>
<li class="post-copyright-link">
<strong>本文链接：</strong>
<a href="/WES/wes02/" title="外显子测序分析二">WES/wes02/</a>
</li>
<li class="post-copyright-license">
<strong>版权声明： </strong>
本博客所有文章均为原创作品，转载请注明出处！
</li>
</ul>

</div>

    
    
    


<div>

<div style="text-align:center;color: #ccc;font-size:14px;">------ 本文结束 ------</div>

</div>

    

    
      <div>
        <div style="padding: 10px 0; margin: 20px auto; width: 90%; text-align: center;">
  <div>坚持原创文章分享，您的支持将鼓励我继续创作！</div>
  <button id="rewardButton" disable="enable" onclick="var qr = document.getElementById('QR'); if (qr.style.display === 'none') {qr.style.display='block';} else {qr.style.display='none'}">
    <span>Donate</span>
  </button>
  <div id="QR" style="display: none;">

    
      <div id="wechat" style="display: inline-block">
        <img id="wechat_qr" src="/images/wechatpay.jpg" alt="括囊无誉 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
    

    

    

  </div>
</div>

      </div>
    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/WES/" rel="tag"># WES</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/Linux/seqkit/" rel="next" title="使用SEQKIT拆分FASTQ文件">
                <i class="fa fa-chevron-left"></i> 使用SEQKIT拆分FASTQ文件
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/Linux/condaEnvs/" rel="prev" title="使用CONDA构建环境">
                使用CONDA构建环境 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a>
        </noscript>
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image" src="/images/avatar.png" alt="括囊无誉">
          <p class="site-author-name" itemprop="name">括囊无誉</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives/">
                <span class="site-state-item-count">207</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">11</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">218</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        
          <div class="feed-link motion-element">
            <a href="/atom.xml" rel="alternate">
              <i class="fa fa-rss"></i>
              RSS
            </a>
          </div>
        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="http://www.jianshu.com/u/75d3f95b058e" target="_blank" title="简书">
                  
                    <i class="fa fa-fw fa-http://www.jianshu.com/u/75d3f95b058e"></i>
                  
                    
                      简书
                    
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#一、数据质控"><span class="nav-number">1.</span> <span class="nav-text">一、数据质控</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#二、比对至基因组"><span class="nav-number">2.</span> <span class="nav-text">二、比对至基因组</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#三、PCR去重"><span class="nav-number">3.</span> <span class="nav-text">三、PCR去重</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#四、BQSR"><span class="nav-number">4.</span> <span class="nav-text">四、BQSR</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#五、变异检测"><span class="nav-number">5.</span> <span class="nav-text">五、变异检测</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#六、位点注释"><span class="nav-number">6.</span> <span class="nav-text">六、位点注释</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#6-1-启动Docker"><span class="nav-number">6.1.</span> <span class="nav-text">6.1 启动Docker</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-2-使用CNNScoreVariants进行评分"><span class="nav-number">6.2.</span> <span class="nav-text">6.2 使用CNNScoreVariants进行评分</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-3-使用FilterVariantTranches过滤"><span class="nav-number">6.3.</span> <span class="nav-text">6.3 使用FilterVariantTranches过滤</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6-4-使用Funcotator进行功能注释"><span class="nav-number">6.4.</span> <span class="nav-text">6.4 使用Funcotator进行功能注释</span></a></li></ol></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">
  
  &copy;  2017 - 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">括囊无誉</span>
</div>


<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


        
<div class="busuanzi-count">
  <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="site-uv">
      <i class="fa fa-user"></i> 访问人数
      <span class="busuanzi-value" id="busuanzi_value_site_uv"></span>
      
    </span>
  

  
    <span class="site-pv">
      <i class="fa fa-eye"></i> 总访问量
      <span class="busuanzi-value" id="busuanzi_value_site_pv"></span>
      次
    </span>
  
</div>


        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>



  


  

    
      <script id="dsq-count-scr" src="https://uteric-1.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'http://uteric.github.io/WES/wes02/';
          this.page.identifier = 'WES/wes02/';
          this.page.title = '外显子测序分析二';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://uteric-1.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script>
    

  




	





  








  

  <script type="text/javascript">
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url);
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x" /></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x" /></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  
<script>
(function(){
    var bp = document.createElement('script');
    var curProtocol = window.location.protocol.split(':')[0];
    if (curProtocol === 'https') {
        bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';        
    }
    else {
        bp.src = 'http://push.zhanzhang.baidu.com/push.js';
    }
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(bp, s);
})();
</script>


  

  

  

  

</body>
</html>
