<!DOCTYPE html>



  


<html class="theme-next mist use-motion" lang="zh-Hans">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">








<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css">




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css">


  <meta name="keywords" content="WES,">





  <link rel="alternate" href="/atom.xml" title="诸子百家" type="application/atom+xml">




  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.2">






<meta name="description" content="一、原始数据下载 如果是从测序公司拿到数据，一般会得到处理过的fastq格式，而我们在此以SRA为数据源获得原始数据。以PRJNA669198这个项目为例，我们下载SRR12846241，原始数据2.76Gb，样本来自31岁的男性colon，患有林奇综合征（Lynch syndrome，或称HNPCC,遗传性非息肉性大肠癌），测序平台是Illumina，双端测序，以上是基本信息。 1prefetc">
<meta name="keywords" content="WES">
<meta property="og:type" content="article">
<meta property="og:title" content="外显子测序分析教程">
<meta property="og:url" content="http://uteric.github.io/WES/wes01/index.html">
<meta property="og:site_name" content="诸子百家">
<meta property="og:description" content="一、原始数据下载 如果是从测序公司拿到数据，一般会得到处理过的fastq格式，而我们在此以SRA为数据源获得原始数据。以PRJNA669198这个项目为例，我们下载SRR12846241，原始数据2.76Gb，样本来自31岁的男性colon，患有林奇综合征（Lynch syndrome，或称HNPCC,遗传性非息肉性大肠癌），测序平台是Illumina，双端测序，以上是基本信息。 1prefetc">
<meta property="og:locale" content="zh-Hans">
<meta property="og:updated_time" content="2020-10-21T18:08:00.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="外显子测序分析教程">
<meta name="twitter:description" content="一、原始数据下载 如果是从测序公司拿到数据，一般会得到处理过的fastq格式，而我们在此以SRA为数据源获得原始数据。以PRJNA669198这个项目为例，我们下载SRR12846241，原始数据2.76Gb，样本来自31岁的男性colon，患有林奇综合征（Lynch syndrome，或称HNPCC,遗传性非息肉性大肠癌），测序平台是Illumina，双端测序，以上是基本信息。 1prefetc">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Mist',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '9e233f3fa2f200ca831529de7ba10253',
      indexName: 'uteric',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://uteric.github.io/WES/wes01/">





  <title>外显子测序分析教程 | 诸子百家</title>
  




<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-104081724-1', 'auto');
ga('send', 'pageview');

</script>











</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <meta name="baidu-site-verification" content="bfQpbnNmEL">

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">诸子百家</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br>
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br>
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br>
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br>
            
            关于
          </a>
        </li>
      

      
        <li class="menu-item menu-item-search">
          
            <a href="javascript:;" class="popup-trigger">
          
            
              <i class="menu-item-icon fa fa-search fa-fw"></i> <br>
            
            搜索
          </a>
        </li>
      
    </ul>
  

  
    <div class="site-search">
      
  <div class="popup search-popup local-search-popup">
<div class="local-search-header clearfix">
<span class="search-icon">
<i class="fa fa-search"></i>
</span>
<span class="popup-btn-close">
<i class="fa fa-times-circle"></i>
</span>
<div class="local-search-input-wrapper">
<input autocomplete="off" placeholder="搜索..." spellcheck="false" type="text" id="local-search-input">
</div>
</div>
<div id="local-search-result"></div>
</div>


    </div>
  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://uteric.github.io/WES/wes01/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="括囊无誉">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="诸子百家">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">外显子测序分析教程</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-10-21T13:08:00-05:00">
                2020-10-21
              </time>
            

            

            
          </span>

          
            <span class="post-category">
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/编程语言/" itemprop="url" rel="index">
                    <span itemprop="name">编程语言</span>
                  </a>
                </span>

                
                
              
            </span>
          


          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/WES/wes01/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count" data-disqus-identifier="WES/wes01/" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          

          
            <span class="post-meta-divider">|</span>
            <span class="page-pv"><i class="fa fa-file-o"></i> 热度
            <span class="busuanzi-value" id="busuanzi_value_page_pv"></span>°C
            </span>
          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <p><strong>一、原始数据下载</strong></p>
<p>如果是从测序公司拿到数据，一般会得到处理过的fastq格式，而我们在此以SRA为数据源获得原始数据。以<a href="https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP287563&amp;o=bytes_l%3Aa" target="_blank" rel="noopener">PRJNA669198</a>这个项目为例，我们下载<em>SRR12846241</em>，原始数据2.76Gb，样本来自31岁的男性colon，患有林奇综合征（Lynch syndrome，或称HNPCC,遗传性非息肉性大肠癌），测序平台是Illumina，双端测序，以上是基本信息。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">prefetch SRR12846241</span><br></pre></td></tr></table></figure>
<p>注：prefetch是sratoolkit中的一个程序，用于从SRA数据库下载测序数据，以上是下载单个文件，若要下载多个文件，可以在SRA数据库的原始数据页面获取Accession List (SRR_Acc_List.txt)，批量下载命令如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">prefetch --option-file SRR_Acc_List.txt</span><br></pre></td></tr></table></figure>
<p>使用prefetch命令下载，数据将默认储存在~/ncbi/public/sra/目录下。</p>
<p><strong>二、原始数据转换</strong></p>
<p>下载完成后，我们获得的原始数据是SRA格式（SRR12846241.sra），需要首先使用sratoolkit转换为fastq格式。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">fastq-dump --split-3 ./SRR12846241.sra</span><br><span class="line"><span class="comment">#Read 31651170 spots for ./SRR12846241.sra</span></span><br><span class="line"><span class="comment">#Written 31651170 spots for ./SRR12846241.sra</span></span><br><span class="line"><span class="comment">#生成两个文件</span></span><br><span class="line"><span class="comment">#SRR12846241_1.fastq  SRR12846241_2.fastq</span></span><br></pre></td></tr></table></figure>
<p>解释一下，对于双端测序，使用–split-3参数，将生成1.fastq和2.fastq两个文件；对于单端测序，需要去掉该参数，只生成一个.fastq文件。</p>
<p><strong>三、测序数据质控</strong></p>
<p>在进行下一步之前，我们先建立一些必要的目录，以便对后续的数据进行分类</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mkdir &#123;fastq,qc,bam,nodup&#125;</span><br><span class="line"><span class="comment">#将生成的.fastq文件移至目录fastq下</span></span><br></pre></td></tr></table></figure>
<p>如要查看数据质量，我们可以使用经典的程序fastqc，语法如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">fastqc -o ./qc -t 4 ./fastq/*.fastq</span><br><span class="line"><span class="comment"># -o，指定输出目录；-t 线程数；后面是输入文件</span></span><br><span class="line"><span class="comment"># 这个分析过程很快，而且显示进程</span></span><br><span class="line">Started analysis of SRR12846241_1.fastq</span><br><span class="line">Started analysis of SRR12846241_2.fastq</span><br><span class="line">Approx 5% complete <span class="keyword">for</span> SRR12846241_1.fastq</span><br><span class="line">Approx 5% complete <span class="keyword">for</span> SRR12846241_2.fastq</span><br><span class="line">Approx 10% complete <span class="keyword">for</span> SRR12846241_1.fastq</span><br><span class="line">Approx 10% complete <span class="keyword">for</span> SRR12846241_2.fastq</span><br><span class="line">...</span><br><span class="line">Analysis complete <span class="keyword">for</span> SRR12846241_1.fastq</span><br><span class="line">Analysis complete <span class="keyword">for</span> SRR12846241_2.fastq</span><br><span class="line"><span class="comment"># 运行完成后会生成如下文件</span></span><br><span class="line">SRR12846241_1_fastqc.html  SRR12846241_1_fastqc.zip  SRR12846241_2_fastqc.html  SRR12846241_2_fastqc.zip</span><br></pre></td></tr></table></figure>
<p>fastqc是用于查看数据质量的，打开生成的.html文件可以看到，这个数据的质量是相当好的，这得益于近年来测序技术的巨大进步。</p>
<p>数据质控的部分我们可以使用fastp，这个程序可以进行过滤（去除低质量序列，较多N的序列），去掉接头，进行碱基校正，输出质控报告等。</p>
<p>对于单端测序，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fastp -i input.fastq -o output.fastq</span><br></pre></td></tr></table></figure>
<p>对于双端测序，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fastp -i input.1.fastq -I input.2.fastq -o output.1.fastq -O output.2.fastq</span><br></pre></td></tr></table></figure>
<p>这个程序的多数功能是默认开启的，不需额外设置</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br></pre></td><td class="code"><pre><span class="line">fastp -i ./fastq/SRR12846241_1.fastq -I ./fastq/SRR12846241_2.fastq -o ./qc/cleaned_1.fastq -O ./qc/cleaned_2.fastq -w 4</span><br><span class="line"><span class="comment"># --adapter_sequence read1接头序列，如不指定则自动检测;</span></span><br><span class="line"><span class="comment"># --adapter_sequence_r2 read1接头序列，如不指定则自动检测;</span></span><br><span class="line"><span class="comment"># -c 适用于双端测序，对碱基进行校正，通过查找每一对read的重合区，然后对mismatch的碱基进行校正，方法是使用高质量的read取代低质量的read;</span></span><br><span class="line"><span class="comment"># -q 设置低质量的标准，默认值15；</span></span><br><span class="line"><span class="comment"># -u 允许不合格碱基的百分比，默认值40%；</span></span><br><span class="line"><span class="comment"># -n 若read中的N碱基数据大于此值，则该对reads将被丢弃，默认值5；</span></span><br><span class="line"><span class="comment"># -w 线程数，默认值2，根据自己电脑的CPU来确定；</span></span><br><span class="line"><span class="comment"># -5/-3 滑窗裁剪，需要指定threshold,-5 滑窗从5'向3'移动，-3滑窗从3'向5'移动；</span></span><br><span class="line"><span class="comment"># 结果如下</span></span><br><span class="line">Read1 before filtering:</span><br><span class="line">total reads: 31651170</span><br><span class="line">total bases: 4779326670</span><br><span class="line">Q20 bases: 4680761815(97.9377%)</span><br><span class="line">Q30 bases: 4513495612(94.4379%)</span><br><span class="line"></span><br><span class="line">Read2 before filtering:</span><br><span class="line">total reads: 31651170</span><br><span class="line">total bases: 4779326670</span><br><span class="line">Q20 bases: 4651248152(97.3202%)</span><br><span class="line">Q30 bases: 4453204808(93.1764%)</span><br><span class="line"></span><br><span class="line">Read1 after filtering:</span><br><span class="line">total reads: 31297805</span><br><span class="line">total bases: 4676179403</span><br><span class="line">Q20 bases: 4589169104(98.1393%)</span><br><span class="line">Q30 bases: 4429126929(94.7168%)</span><br><span class="line"></span><br><span class="line">Read2 aftering filtering:</span><br><span class="line">total reads: 31297805</span><br><span class="line">total bases: 4676179403</span><br><span class="line">Q20 bases: 4574607381(97.8279%)</span><br><span class="line">Q30 bases: 4387146408(93.819%)</span><br><span class="line"></span><br><span class="line">Filtering result:</span><br><span class="line">reads passed filter: 62595610</span><br><span class="line">reads failed due to low quality: 696646</span><br><span class="line">reads failed due to too many N: 10084</span><br><span class="line">reads failed due to too short: 0</span><br><span class="line">reads with adapter trimmed: 6762458</span><br><span class="line">bases trimmed due to adapters: 99756614</span><br><span class="line"></span><br><span class="line">Duplication rate: 14.5952%</span><br><span class="line"></span><br><span class="line">Insert size peak (evaluated by paired-end reads): 189</span><br><span class="line"></span><br><span class="line">JSON report: fastp.json</span><br><span class="line">HTML report: fastp.html</span><br><span class="line"></span><br><span class="line">fastp -i ./fastq/SRR12846241_1.fastq -I ./fastq/SRR12846241_2.fastq -o ./qc/cleaned_1.fastq -O ./qc/cleaned_2.fastq -w 4</span><br><span class="line">fastp v0.20.1, time used: 642 seconds</span><br></pre></td></tr></table></figure>
<p><strong>四、比对至基因组</strong></p>
<p>对于DNA的比对，使用BWA这个程序，使用之前，需要为BWA生成索引文件，命令如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br></pre></td><td class="code"><pre><span class="line">bwa index hg38.fa</span><br><span class="line"><span class="comment"># 过程如下，耗时约100分钟</span></span><br><span class="line">[bwa_index] Pack FASTA... 45.94 sec</span><br><span class="line">[bwa_index] Construct BWT <span class="keyword">for</span> the packed sequence...</span><br><span class="line">[BWTIncCreate] textLength=6418572210, availableWord=463634060</span><br><span class="line">[BWTIncConstructFromPacked] 10 iterations <span class="keyword">done</span>. 99999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 20 iterations <span class="keyword">done</span>. 199999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 30 iterations <span class="keyword">done</span>. 299999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 40 iterations <span class="keyword">done</span>. 399999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 50 iterations <span class="keyword">done</span>. 499999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 60 iterations <span class="keyword">done</span>. 599999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 70 iterations <span class="keyword">done</span>. 699999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 80 iterations <span class="keyword">done</span>. 799999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 90 iterations <span class="keyword">done</span>. 899999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 100 iterations <span class="keyword">done</span>. 999999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 110 iterations <span class="keyword">done</span>. 1099999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 120 iterations <span class="keyword">done</span>. 1199999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 130 iterations <span class="keyword">done</span>. 1299999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 140 iterations <span class="keyword">done</span>. 1399999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 150 iterations <span class="keyword">done</span>. 1499999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 160 iterations <span class="keyword">done</span>. 1599999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 170 iterations <span class="keyword">done</span>. 1699999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 180 iterations <span class="keyword">done</span>. 1799999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 190 iterations <span class="keyword">done</span>. 1899999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 200 iterations <span class="keyword">done</span>. 1999999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 210 iterations <span class="keyword">done</span>. 2099999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 220 iterations <span class="keyword">done</span>. 2199999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 230 iterations <span class="keyword">done</span>. 2299999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 240 iterations <span class="keyword">done</span>. 2399999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 250 iterations <span class="keyword">done</span>. 2499999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 260 iterations <span class="keyword">done</span>. 2599999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 270 iterations <span class="keyword">done</span>. 2699999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 280 iterations <span class="keyword">done</span>. 2799999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 290 iterations <span class="keyword">done</span>. 2899999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 300 iterations <span class="keyword">done</span>. 2999999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 310 iterations <span class="keyword">done</span>. 3099999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 320 iterations <span class="keyword">done</span>. 3199999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 330 iterations <span class="keyword">done</span>. 3299999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 340 iterations <span class="keyword">done</span>. 3399999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 350 iterations <span class="keyword">done</span>. 3499999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 360 iterations <span class="keyword">done</span>. 3599999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 370 iterations <span class="keyword">done</span>. 3699999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 380 iterations <span class="keyword">done</span>. 3799999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 390 iterations <span class="keyword">done</span>. 3899999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 400 iterations <span class="keyword">done</span>. 3999999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 410 iterations <span class="keyword">done</span>. 4099999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 420 iterations <span class="keyword">done</span>. 4199999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 430 iterations <span class="keyword">done</span>. 4299999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 440 iterations <span class="keyword">done</span>. 4399999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 450 iterations <span class="keyword">done</span>. 4499999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 460 iterations <span class="keyword">done</span>. 4599999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 470 iterations <span class="keyword">done</span>. 4699999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 480 iterations <span class="keyword">done</span>. 4799999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 490 iterations <span class="keyword">done</span>. 4899999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 500 iterations <span class="keyword">done</span>. 4999999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 510 iterations <span class="keyword">done</span>. 5099999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 520 iterations <span class="keyword">done</span>. 5199999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 530 iterations <span class="keyword">done</span>. 5299999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 540 iterations <span class="keyword">done</span>. 5399999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 550 iterations <span class="keyword">done</span>. 5499999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 560 iterations <span class="keyword">done</span>. 5599999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 570 iterations <span class="keyword">done</span>. 5699999986 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 580 iterations <span class="keyword">done</span>. 5798165394 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 590 iterations <span class="keyword">done</span>. 5886412978 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 600 iterations <span class="keyword">done</span>. 5964843650 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 610 iterations <span class="keyword">done</span>. 6034549010 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 620 iterations <span class="keyword">done</span>. 6096499330 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 630 iterations <span class="keyword">done</span>. 6151556914 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 640 iterations <span class="keyword">done</span>. 6200488210 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 650 iterations <span class="keyword">done</span>. 6243974434 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 660 iterations <span class="keyword">done</span>. 6282621122 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 670 iterations <span class="keyword">done</span>. 6316966322 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 680 iterations <span class="keyword">done</span>. 6347488418 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 690 iterations <span class="keyword">done</span>. 6374612482 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 700 iterations <span class="keyword">done</span>. 6398716386 characters processed.</span><br><span class="line">[BWTIncConstructFromPacked] 710 iterations <span class="keyword">done</span>. 6418572210 characters processed.</span><br><span class="line">[bwt_gen] Finished constructing BWT <span class="keyword">in</span> 710 iterations.</span><br><span class="line">[bwa_index] 4475.92 seconds elapse.</span><br><span class="line">[bwa_index] Update BWT... 34.89 sec</span><br><span class="line">[bwa_index] Pack forward-only FASTA... 30.80 sec</span><br><span class="line">[bwa_index] Construct SA from BWT and Occ...  1760.91 sec</span><br><span class="line">[main] Version: 0.7.17-r1188</span><br><span class="line">[main] CMD: bwa index hg38.fa</span><br><span class="line">[main] Real time: 6467.646 sec; CPU: 6348.453 sec</span><br><span class="line"><span class="comment"># 运行完成后，生成如下文件</span></span><br><span class="line">hg38.fa.amb  hg38.fa.ann  hg38.fa.bwt  hg38.fa.fai  hg38.fa.pac  hg38.fa.sa</span><br><span class="line"><span class="comment"># 若没有这些索引文件，则会在运行时报错，提示找不到参考基因组</span></span><br></pre></td></tr></table></figure>
<p>下面，使用BWA进行MAPPING，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#    bwa mem\</span></span><br><span class="line"><span class="comment">#       -t $NUM_THREADS\ 线程数</span></span><br><span class="line"><span class="comment">#       -B $MISMATCH_PENALTY\ 空位罚分</span></span><br><span class="line"><span class="comment">#       -R '@RG\tID:lane1\tSM:sample1\tLB:genome\tPL:ILLUMINA'\ 设定头文件,必填，ID：通道名或样本名,通过这个信息分组，必须唯一；SM：样本名；LB：⽂库名；PL：测序平台信息[ILLUMINA, SOLID]</span></span><br><span class="line"><span class="comment">#       -M\ 将shorter split hits标记为次优，以兼容 Picard’s markDuplicates软件</span></span><br><span class="line"><span class="comment">#       $INDEX\ 指定参考基因组</span></span><br><span class="line"><span class="comment">#       ./$sample\_1.fastq\ 输入fastq</span></span><br><span class="line"><span class="comment">#       ./$sample\_2.fastq\ 输入fastq</span></span><br><span class="line"><span class="comment">#       | samtools view -q4 -bS -o - -\ 将上一步生成的SAM转为BAM</span></span><br><span class="line"><span class="comment">#       | samtools sort -@4 -o ./bam_files/$sample.bam - 排序并输出BAM</span></span><br><span class="line">bwa mem -t 4 -B 4 -M ~/ncbi/hg38/chroms/hg38.fa /mnt/d/wes/qc/cleaned_1.fastq /mnt/d/wes/qc/cleaned_2.fastq | samtools view -q4 -bS -o - - | samtools sort -@4 -o /mnt/d/wes/bam/clean.bam -</span><br><span class="line"><span class="comment"># 这个过程是非常漫长的，对于普通的电脑可能耗时10个小时，做生信分析配备一台超高性能的电脑是很必要的，否则时间都花在等待当中</span></span><br><span class="line"><span class="comment"># 下面这是一个典型的比对单元，整个过程是由数百个这样的单元组成</span></span><br><span class="line">[M::process] <span class="built_in">read</span> 267710 sequences (40000156 bp)...</span><br><span class="line">[M::mem_pestat] <span class="comment"># candidate unique pairs for (FF, FR, RF, RR): (0, 90279, 1, 0)</span></span><br><span class="line">[M::mem_pestat] skip orientation FF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] analyzing insert size distribution <span class="keyword">for</span> orientation FR...</span><br><span class="line">[M::mem_pestat] (25, 50, 75) percentile: (169, 198, 232)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> computing mean and std.dev: (43, 358)</span><br><span class="line">[M::mem_pestat] mean and std.dev: (202.73, 45.61)</span><br><span class="line">[M::mem_pestat] low and high boundaries <span class="keyword">for</span> proper pairs: (1, 421)</span><br><span class="line">[M::mem_pestat] skip orientation RF as there are not enough pairs</span><br><span class="line">[M::mem_pestat] skip orientation RR as there are not enough pairs</span><br><span class="line">[M::mem_process_seqs] Processed 267710 reads <span class="keyword">in</span> 105.719 CPU sec, 61.174 real sec</span><br><span class="line"><span class="comment"># 运行结束后，会生成一个clean.bam文件</span></span><br></pre></td></tr></table></figure>
<p><strong>五、加头文件</strong></p>
<p>这步是外加的，因为在使用BWA进行比对时没有加-R参数，因此缺少Read Group信息，可以通过picard tool中的AddOrReplaceReadGroups.jar加入RG信息，因为picard被整合在GATK中，因此，我们可以使用GATK来加RG</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">gatk AddOrReplaceReadGroups -I /mnt/d/wes/bam/clean.bam -O /mnt/d/wes/bam/clean.RG.bam -LB genome -PL ILLUMINA -PU unit1 -SM sample1</span><br><span class="line"><span class="comment"># 参数解释</span></span><br><span class="line">--RGLB,-LB &lt;String&gt;           Read-Group library  Required.</span><br><span class="line">--RGPL,-PL &lt;String&gt;           Read-Group platform (e.g. ILLUMINA, SOLID)  Required.</span><br><span class="line">--RGPU,-PU &lt;String&gt;           Read-Group platform unit (eg. run barcode)  Required.</span><br><span class="line">--RGSM,-SM &lt;String&gt;           Read-Group sample name  Required.</span><br></pre></td></tr></table></figure>
<p><strong>六、PCR去重</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">gatk MarkDuplicates -I /mnt/d/wes/bam/clean.RG.bam -O /mnt/d/wes/nodup/nodup.bam -M /mnt/d/wes/nodup/metrics.txt --REMOVE_DUPLICATES <span class="literal">true</span> --ASSUME_SORTED <span class="literal">true</span> </span><br><span class="line"><span class="comment"># 参数解释</span></span><br><span class="line">--METRICS_FILE,-M &lt;File&gt;  生成metrics的文件</span><br><span class="line">--REMOVE_DUPLICATES 默认为<span class="literal">false</span>，若为<span class="literal">true</span>，则在生成的文件中去掉duplicates</span><br><span class="line">--ASSUME_SORTED 默认为<span class="literal">false</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 生成索引文件</span></span><br><span class="line">samtools index /mnt/d/wes/nodup/nodup.bam</span><br></pre></td></tr></table></figure>
<p><strong>七、BQSR (Recalibration Base Quality Score)</strong></p>
<p>下面进行变异检测，首先，我们需要先为hg38.fa生成dict文件，这可以通过调用picard中的CreateSequenceDictonary模块来实现，也可以直接在GATK中调用。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">java -jar /home/eric/picard/build/libs/picard.jar CreateSequenceDictionary R=hg38.fa O=hg38.dict</span><br><span class="line">gatk CreateSequenceDictionary -R hg38.fa -O hg38.dict &amp;&amp; <span class="built_in">echo</span> <span class="string">"** dict done **"</span> <span class="comment"># 这两个命令等价，&amp;&amp; echo "** dict done **"是提示dict完成，可以去掉</span></span><br></pre></td></tr></table></figure>
<p>另外，补充hg38.fa.fai文件的生成过程，可以调用samtool faidx来生成</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools faidx hg38.fa</span><br></pre></td></tr></table></figure>
<p>上面这两个文件的生成过程都很快，几秒钟就可以完成。</p>
<p>BQSR分两步进行，分别使用BaseRecalibrator与ApplyBQSR两个工具。</p>
<p>第一步：<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036898312-BaseRecalibrator" target="_blank" rel="noopener">BaseRecalibrator</a>，此工具根据物种的known sites，生成一个校正质量值所需要的表格文件，基本用法如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">gatk BaseRecalibrator \</span><br><span class="line">   -I my_reads.bam \</span><br><span class="line">   -R reference.fasta \</span><br><span class="line">   --known-sites sites_of_variation.vcf \</span><br><span class="line">   --known-sites another/optional/setOfSitesToMask.vcf \</span><br><span class="line">   -O recal_data.table</span><br></pre></td></tr></table></figure>
<p>在本例中，执行如下命令：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk BaseRecalibrator -I /mnt/d/wes/nodup/nodup.bam -R /home/eric/ncbi/hg38/chroms/hg38.fa --known-sites /home/eric/ncbi/hg38/vcf/dbsnp_138.hg38.vcf.gz --known-sites /home/eric/ncbi/hg38/vcf/1000G_phase1.snps.high_confidence.hg38.vcf.gz --known-sites /home/eric/ncbi/hg38/vcf/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz -L /home/eric/ncbi/hg38/vcf/hg38_v0_HybSelOligos_whole_exome_illumina_coding_v1_whole_exome_illumina_coding_v1.Homo_sapiens_assembly38.targets.interval_list -O /mnt/d/wes/bqsr/recal_data2.table</span><br></pre></td></tr></table></figure>
<p>第二步：<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037055712-ApplyBQSR" target="_blank" rel="noopener">ApplyBQSR</a>，此工具执行BQSR的第二步，具体来说，它会根据第一步中BaseRecalibrator工具生成的重新校准表来重新校准输入的BAM的质量，并输出重新校准的BAM文件，基本用法如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">gatk ApplyBQSR \</span><br><span class="line">   -R reference.fasta \</span><br><span class="line">   -I input.bam \</span><br><span class="line">   --bqsr-recal-file recalibration.table \</span><br><span class="line">   -O output.bam</span><br></pre></td></tr></table></figure>
<p>在本例中，执行如下命令：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk ApplyBQSR -I /mnt/d/wes/nodup/nodup.bam -R /home/eric/ncbi/hg38/chroms/hg38.fa --bqsr-recal-file /mnt/d/wes/bqsr/recal_data2.table -O /mnt/d/wes/bqsr/bqsr.bam</span><br></pre></td></tr></table></figure>
<p>下面，可以对生成的BAM文件进行验证，使用<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036854731-ValidateSamFile-Picard-" target="_blank" rel="noopener">ValidateSamFile</a>工具，若显示no errors found，就可以进行变异检测了。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk ValidateSamFile -I /mnt/d/wes/bqsr/bqsr.bam</span><br></pre></td></tr></table></figure>
<p>如发现错误，则可以使用<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036713471-FixMateInformation-Picard-" target="_blank" rel="noopener">FixMateInformation</a>工具进行修复</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk FixMateInformation -I /mnt/d/wes/bqsr/bqsr.bam -O /mnt/d/wes/bqsr/fixed_bqsr.bam</span><br></pre></td></tr></table></figure>
<p>变异检测，使用<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037225632-HaplotypeCaller" target="_blank" rel="noopener">HaplotypeCaller</a>工具，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">gatk HaplotypeCaller\</span><br><span class="line">   -R Homo_sapiens_assembly38.fasta \</span><br><span class="line">   -I input.bam \</span><br><span class="line">   -O output.g.vcf.gz \</span><br><span class="line">   -ERC GVCF</span><br></pre></td></tr></table></figure>
<p>在本例中，使用如下参数，此步极费时间，我这台低配电脑耗时10个小时</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk HaplotypeCaller -R /home/eric/ncbi/hg38/chroms/hg38.fa -I /mnt/d/wes/bqsr/bqsr.bam -O /mnt/d/wes/<span class="built_in">caller</span>/output.g.vcf.gz -ERC GVCF</span><br></pre></td></tr></table></figure>
<p>生成g.vcf.gz文件之后，就可以使用<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036855131-GenotypeGVCFs" target="_blank" rel="noopener">GenotypeGVCFs</a>工具对多个样本执行joint genotyping，目的是将多个样本合成一个文件 </p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">gatk GenotypeGVCFs</span><br><span class="line">    -R Homo_sapiens_assembly38.fasta \</span><br><span class="line">    -V input.g.vcf.gz \</span><br><span class="line">    -O output.vcf.gz </span><br><span class="line"></span><br><span class="line">gatk GenotypeGVCFs -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/<span class="built_in">caller</span>/output.g.vcf.gz -O /mnt/d/wes/<span class="built_in">caller</span>/output.vcf.gz</span><br></pre></td></tr></table></figure>
<p>提取SNV，使用<a href="https://gatk.broadinstitute.org/hc/en-us/articles/360040508071-SelectVariants" target="_blank" rel="noopener">SelectVariants</a>工具，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">gatk SelectVariants \</span><br><span class="line">     -R Homo_sapiens_assembly38.fasta \</span><br><span class="line">     -V input.vcf \</span><br><span class="line">     --select-type-to-include SNP \ <span class="comment"># 可选以下参数：NO_VARIATION, SNP, MNP, INDEL, SYMBOLIC, MIXED</span></span><br><span class="line">     -O output.vcf</span><br></pre></td></tr></table></figure>
<p>在本例中，使用如下命令：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk SelectVariants -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/<span class="built_in">caller</span>/output.vcf.gz --select-type-to-include SNP -O /mnt/d/wes/<span class="built_in">caller</span>/snp.vcf</span><br></pre></td></tr></table></figure>
<p>这个过程极快，只需要几秒即可完成。下面，提取INDEL，仍然使用SelectVariants工具，只是参数选择INDEL，</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk SelectVariants -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/<span class="built_in">caller</span>/output.vcf.gz --select-type-to-include INDEL -O /mnt/d/wes/<span class="built_in">caller</span>/indel.vcf</span><br></pre></td></tr></table></figure>
<p>突变位点质量值重矫正(VQSR)</p>
<p>这一步是对突变位点进行质控，原理是通过比对与已知变异位点的交集，评估各个突变位点的可信度。已知变异集包括：hapmap，omni，1000G，dbsnp。</p>
<p>第一步：SNP的VQSR的过滤</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">gatk VariantRecalibrator \</span><br><span class="line">   -R Homo_sapiens_assembly38.fasta \</span><br><span class="line">   -V input.vcf.gz \</span><br><span class="line">   --resource hapmap,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">true</span>,prior=15.0:hapmap_3.3.hg38.sites.vcf.gz \</span><br><span class="line">   --resource omni,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">false</span>,prior=12.0:1000G_omni2.5.hg38.sites.vcf.gz \</span><br><span class="line">   --resource 1000G,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">false</span>,prior=10.0:1000G_phase1.snps.high_confidence.hg38.vcf.gz \</span><br><span class="line">   --resource dbsnp,known=<span class="literal">true</span>,training=<span class="literal">false</span>,truth=<span class="literal">false</span>,prior=2.0:Homo_sapiens_assembly38.dbsnp138.vcf.gz \</span><br><span class="line">   -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \</span><br><span class="line">   -mode SNP \ <span class="comment"># 可选参数：SNP，INDEL，BOTH（仅用于测试目的）</span></span><br><span class="line">   -O output.recal \</span><br><span class="line">   --tranches-file output.tranches \</span><br><span class="line">   --rscript-file output.plots.R</span><br><span class="line"></span><br><span class="line"><span class="comment"># 在本例中，命令如下：</span></span><br><span class="line">gatk VariantRecalibrator -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/<span class="built_in">caller</span>/output.vcf.gz --resource:hapmap,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">true</span>,prior=15.0 /home/eric/ncbi/hg38/vcf/hapmap_3.3.hg38.vcf.gz --resource:omni,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">false</span>,prior=12.0 /home/eric/ncbi/hg38/vcf/1000G_omni2.5.hg38.vcf.gz --resource:1000G,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">false</span>,prior=10.0 /home/eric/ncbi/hg38/vcf/1000G_phase1.snps.high_confidence.hg38.vcf.gz --resource:dbsnp,known=<span class="literal">true</span>,training=<span class="literal">false</span>,truth=<span class="literal">false</span>,prior=2.0 /home/eric/ncbi/hg38/vcf/dbsnp_138.hg38.vcf.gz -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -mode SNP -O /mnt/d/wes/<span class="built_in">caller</span>/output.snp.recal --tranches-file /mnt/d/wes/<span class="built_in">caller</span>/output.snp.tranches --rscript-file /mnt/d/wes/<span class="built_in">caller</span>/output.snp.plots.R</span><br></pre></td></tr></table></figure>
<p>INDEL的VQSR的过滤，参数有点变动，–resource也不同</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk VariantRecalibrator -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/<span class="built_in">caller</span>/output.vcf.gz --max-gaussians 4 --resource:mills,known=<span class="literal">false</span>,training=<span class="literal">true</span>,truth=<span class="literal">true</span>,prior=12.0 /home/eric/ncbi/hg38/vcf/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz --resource:dbsnp,known=<span class="literal">true</span>,training=<span class="literal">false</span>,truth=<span class="literal">false</span>,prior=2.0 /home/eric/ncbi/hg38/vcf/dbsnp_146.hg38.vcf.gz -an QD -an DP -an MQRankSum -an ReadPosRankSum -an FS -an SOR -mode INDEL -O /mnt/d/wes/<span class="built_in">caller</span>/output.indel.recal --tranches-file /mnt/d/wes/<span class="built_in">caller</span>/output.indel.tranches --rscript-file /mnt/d/wes/<span class="built_in">caller</span>/output.indel.plots.R</span><br></pre></td></tr></table></figure>
<p>第二步：ApplyVQSR to SNP（Apply a score cutoff to filter variants based on a recalibration table），即将第一步生成的阀值文件应用于vcf.gz，并生成新的vcf.gz文件，这个过程将阀值以下的突变去除</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">gatk ApplyVQSR \</span><br><span class="line">   -R Homo_sapiens_assembly38.fasta \</span><br><span class="line">   -V input.vcf.gz \ <span class="comment"># 输入文件是由GenotypeGVCFs生成的vcf.gz</span></span><br><span class="line">   -O output.vcf.gz \</span><br><span class="line">   --truth-sensitivity-filter-level 99.0 \</span><br><span class="line">   --tranches-file output.tranches \ <span class="comment"># VariantRecalibrator生成的snp.tranches文件</span></span><br><span class="line">   --recal-file output.recal \ <span class="comment"># VariantRecalibrator生成的snp.recal文件</span></span><br><span class="line">   -mode SNP <span class="comment"># 参数为SNP,INDEL，或BOTH</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 在本例中，命令如下：</span></span><br><span class="line">gatk ApplyVQSR -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/<span class="built_in">caller</span>/output.vcf.gz -O /mnt/d/wes/<span class="built_in">caller</span>/output.vqsr.snp.vcf.gz --truth-sensitivity-filter-level 99.0 --tranches-file /mnt/d/wes/<span class="built_in">caller</span>/output.snp.tranches --recal-file /mnt/d/wes/<span class="built_in">caller</span>/output.snp.recal -mode SNP</span><br></pre></td></tr></table></figure>
<p>INDEL也是如此，Applying recalibration to INDEL，命令如下：</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk ApplyVQSR -R /home/eric/ncbi/hg38/chroms/hg38.fa -V /mnt/d/wes/<span class="built_in">caller</span>/output.vcf.gz -O /mnt/d/wes/<span class="built_in">caller</span>/output.vqsr.indel.vcf.gz --truth-sensitivity-filter-level 99.0 --tranches-file /mnt/d/wes/<span class="built_in">caller</span>/output.indel.tranches --recal-file /mnt/d/wes/<span class="built_in">caller</span>/output.indel.recal -mode INDEL</span><br></pre></td></tr></table></figure>
<p>位点注释</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"> gatk CNNScoreVariants \</span><br><span class="line">   -V vcf_to_annotate.vcf.gz \</span><br><span class="line">   -R reference.fasta \</span><br><span class="line">   -O annotated.vcf</span><br><span class="line"><span class="comment"># SNP文件注释</span></span><br><span class="line">gatk CNNScoreVariants -V /mnt/d/wes/<span class="built_in">caller</span>/output.vqsr.snp.vcf.gz -R /home/eric/ncbi/hg38/chroms/hg38.fa -O /mnt/d/wes/annotate/annotate.snp.vcf --<span class="built_in">disable</span>-avx-check <span class="literal">true</span></span><br><span class="line"><span class="comment"># INDEL文件注释</span></span><br><span class="line">gatk CNNScoreVariants -V /mnt/d/wes/<span class="built_in">caller</span>/output.vqsr.indel.vcf.gz -R /home/eric/ncbi/hg38/chroms/hg38.fa -O /mnt/d/wes/annotate/annotate.snp.vcf</span><br></pre></td></tr></table></figure>
<p>附：GATK参考文件下载地址：<a href="ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg38/" target="_blank" rel="noopener">gatk/bundle/hg38</a>，这些参考文件可以在多个地址下载，但是有些因为压缩格式的问题会报错，这个网站上的参考文件已经验证，可以使用。</p>

      
    </div>

<div>


<ul class="post-copyright">
<li class="post-copyright-author">
<strong>本文作者：</strong>括囊无誉
</li>
<li class="post-copyright-link">
<strong>本文链接：</strong>
<a href="/WES/wes01/" title="外显子测序分析教程">WES/wes01/</a>
</li>
<li class="post-copyright-license">
<strong>版权声明： </strong>
本博客所有文章均为原创作品，转载请注明出处！
</li>
</ul>

</div>

    
    
    


<div>

<div style="text-align:center;color: #ccc;font-size:14px;">------ 本文结束 ------</div>

</div>

    

    
      <div>
        <div style="padding: 10px 0; margin: 20px auto; width: 90%; text-align: center;">
  <div>坚持原创文章分享，您的支持将鼓励我继续创作！</div>
  <button id="rewardButton" disable="enable" onclick="var qr = document.getElementById('QR'); if (qr.style.display === 'none') {qr.style.display='block';} else {qr.style.display='none'}">
    <span>Donate</span>
  </button>
  <div id="QR" style="display: none;">

    
      <div id="wechat" style="display: inline-block">
        <img id="wechat_qr" src="/images/wechatpay.jpg" alt="括囊无誉 WeChat Pay">
        <p>WeChat Pay</p>
      </div>
    

    

    

  </div>
</div>

      </div>
    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/WES/" rel="tag"># WES</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/R/reshape/" rel="next" title="在R中拆分与合并数据">
                <i class="fa fa-chevron-left"></i> 在R中拆分与合并数据
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/R/maltools/" rel="prev" title="使用MAFTOOLS分析肿瘤变异数据">
                使用MAFTOOLS分析肿瘤变异数据 <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          
  <div class="comments" id="comments">
    
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a>
        </noscript>
      </div>
    
  </div>


        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      

      <section class="site-overview sidebar-panel sidebar-panel-active">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image" src="/images/avatar.png" alt="括囊无誉">
          <p class="site-author-name" itemprop="name">括囊无誉</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
              <a href="/archives/">
                <span class="site-state-item-count">215</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-categories">
              <a href="/categories/index.html">
                <span class="site-state-item-count">11</span>
                <span class="site-state-item-name">分类</span>
              </a>
            </div>
          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">220</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        
          <div class="feed-link motion-element">
            <a href="/atom.xml" rel="alternate">
              <i class="fa fa-rss"></i>
              RSS
            </a>
          </div>
        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="http://www.jianshu.com/u/75d3f95b058e" target="_blank" title="简书">
                  
                    <i class="fa fa-fw fa-http://www.jianshu.com/u/75d3f95b058e"></i>
                  
                    
                      简书
                    
                </a>
              </span>
            
          
        </div>

        
        

        
        

        


      </section>

      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">
  
  &copy;  2017 - 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">括囊无誉</span>
</div>


<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Mist
  </a>
</div>


        
<div class="busuanzi-count">
  <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js"></script>

  
    <span class="site-uv">
      <i class="fa fa-user"></i> 访问人数
      <span class="busuanzi-value" id="busuanzi_value_site_uv"></span>
      
    </span>
  

  
    <span class="site-pv">
      <i class="fa fa-eye"></i> 总访问量
      <span class="busuanzi-value" id="busuanzi_value_site_pv"></span>
      次
    </span>
  
</div>


        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  

  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>



  


  

    
      <script id="dsq-count-scr" src="https://uteric-1.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'http://uteric.github.io/WES/wes01/';
          this.page.identifier = 'WES/wes01/';
          this.page.title = '外显子测序分析教程';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://uteric-1.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script>
    

  




	





  








  

  <script type="text/javascript">
    // Popup Window;
    var isfetched = false;
    var isXml = true;
    // Search DB path;
    var search_path = "search.xml";
    if (search_path.length === 0) {
      search_path = "search.xml";
    } else if (/json$/i.test(search_path)) {
      isXml = false;
    }
    var path = "/" + search_path;
    // monitor main search box;

    var onPopupClose = function (e) {
      $('.popup').hide();
      $('#local-search-input').val('');
      $('.search-result-list').remove();
      $('#no-result').remove();
      $(".local-search-pop-overlay").remove();
      $('body').css('overflow', '');
    }

    function proceedsearch() {
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay"></div>')
        .css('overflow', 'hidden');
      $('.search-popup-overlay').click(onPopupClose);
      $('.popup').toggle();
      var $localSearchInput = $('#local-search-input');
      $localSearchInput.attr("autocapitalize", "none");
      $localSearchInput.attr("autocorrect", "off");
      $localSearchInput.focus();
    }

    // search function;
    var searchFunc = function(path, search_id, content_id) {
      'use strict';

      // start loading animation
      $("body")
        .append('<div class="search-popup-overlay local-search-pop-overlay">' +
          '<div id="search-loading-icon">' +
          '<i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>' +
          '</div>' +
          '</div>')
        .css('overflow', 'hidden');
      $("#search-loading-icon").css('margin', '20% auto 0 auto').css('text-align', 'center');

      $.ajax({
        url: path,
        dataType: isXml ? "xml" : "json",
        async: true,
        success: function(res) {
          // get the contents from search data
          isfetched = true;
          $('.popup').detach().appendTo('.header-inner');
          var datas = isXml ? $("entry", res).map(function() {
            return {
              title: $("title", this).text(),
              content: $("content",this).text(),
              url: $("url" , this).text()
            };
          }).get() : res;
          var input = document.getElementById(search_id);
          var resultContent = document.getElementById(content_id);
          var inputEventFunction = function() {
            var searchText = input.value.trim().toLowerCase();
            var keywords = searchText.split(/[\s\-]+/);
            if (keywords.length > 1) {
              keywords.push(searchText);
            }
            var resultItems = [];
            if (searchText.length > 0) {
              // perform local searching
              datas.forEach(function(data) {
                var isMatch = false;
                var hitCount = 0;
                var searchTextCount = 0;
                var title = data.title.trim();
                var titleInLowerCase = title.toLowerCase();
                var content = data.content.trim().replace(/<[^>]+>/g,"");
                var contentInLowerCase = content.toLowerCase();
                var articleUrl = decodeURIComponent(data.url);
                var indexOfTitle = [];
                var indexOfContent = [];
                // only match articles with not empty titles
                if(title != '') {
                  keywords.forEach(function(keyword) {
                    function getIndexByWord(word, text, caseSensitive) {
                      var wordLen = word.length;
                      if (wordLen === 0) {
                        return [];
                      }
                      var startPosition = 0, position = [], index = [];
                      if (!caseSensitive) {
                        text = text.toLowerCase();
                        word = word.toLowerCase();
                      }
                      while ((position = text.indexOf(word, startPosition)) > -1) {
                        index.push({position: position, word: word});
                        startPosition = position + wordLen;
                      }
                      return index;
                    }

                    indexOfTitle = indexOfTitle.concat(getIndexByWord(keyword, titleInLowerCase, false));
                    indexOfContent = indexOfContent.concat(getIndexByWord(keyword, contentInLowerCase, false));
                  });
                  if (indexOfTitle.length > 0 || indexOfContent.length > 0) {
                    isMatch = true;
                    hitCount = indexOfTitle.length + indexOfContent.length;
                  }
                }

                // show search results

                if (isMatch) {
                  // sort index by position of keyword

                  [indexOfTitle, indexOfContent].forEach(function (index) {
                    index.sort(function (itemLeft, itemRight) {
                      if (itemRight.position !== itemLeft.position) {
                        return itemRight.position - itemLeft.position;
                      } else {
                        return itemLeft.word.length - itemRight.word.length;
                      }
                    });
                  });

                  // merge hits into slices

                  function mergeIntoSlice(text, start, end, index) {
                    var item = index[index.length - 1];
                    var position = item.position;
                    var word = item.word;
                    var hits = [];
                    var searchTextCountInSlice = 0;
                    while (position + word.length <= end && index.length != 0) {
                      if (word === searchText) {
                        searchTextCountInSlice++;
                      }
                      hits.push({position: position, length: word.length});
                      var wordEnd = position + word.length;

                      // move to next position of hit

                      index.pop();
                      while (index.length != 0) {
                        item = index[index.length - 1];
                        position = item.position;
                        word = item.word;
                        if (wordEnd > position) {
                          index.pop();
                        } else {
                          break;
                        }
                      }
                    }
                    searchTextCount += searchTextCountInSlice;
                    return {
                      hits: hits,
                      start: start,
                      end: end,
                      searchTextCount: searchTextCountInSlice
                    };
                  }

                  var slicesOfTitle = [];
                  if (indexOfTitle.length != 0) {
                    slicesOfTitle.push(mergeIntoSlice(title, 0, title.length, indexOfTitle));
                  }

                  var slicesOfContent = [];
                  while (indexOfContent.length != 0) {
                    var item = indexOfContent[indexOfContent.length - 1];
                    var position = item.position;
                    var word = item.word;
                    // cut out 100 characters
                    var start = position - 20;
                    var end = position + 80;
                    if(start < 0){
                      start = 0;
                    }
                    if (end < position + word.length) {
                      end = position + word.length;
                    }
                    if(end > content.length){
                      end = content.length;
                    }
                    slicesOfContent.push(mergeIntoSlice(content, start, end, indexOfContent));
                  }

                  // sort slices in content by search text's count and hits' count

                  slicesOfContent.sort(function (sliceLeft, sliceRight) {
                    if (sliceLeft.searchTextCount !== sliceRight.searchTextCount) {
                      return sliceRight.searchTextCount - sliceLeft.searchTextCount;
                    } else if (sliceLeft.hits.length !== sliceRight.hits.length) {
                      return sliceRight.hits.length - sliceLeft.hits.length;
                    } else {
                      return sliceLeft.start - sliceRight.start;
                    }
                  });

                  // select top N slices in content

                  var upperBound = parseInt('1');
                  if (upperBound >= 0) {
                    slicesOfContent = slicesOfContent.slice(0, upperBound);
                  }

                  // highlight title and content

                  function highlightKeyword(text, slice) {
                    var result = '';
                    var prevEnd = slice.start;
                    slice.hits.forEach(function (hit) {
                      result += text.substring(prevEnd, hit.position);
                      var end = hit.position + hit.length;
                      result += '<b class="search-keyword">' + text.substring(hit.position, end) + '</b>';
                      prevEnd = end;
                    });
                    result += text.substring(prevEnd, slice.end);
                    return result;
                  }

                  var resultItem = '';

                  if (slicesOfTitle.length != 0) {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + highlightKeyword(title, slicesOfTitle[0]) + "</a>";
                  } else {
                    resultItem += "<li><a href='" + articleUrl + "' class='search-result-title'>" + title + "</a>";
                  }

                  slicesOfContent.forEach(function (slice) {
                    resultItem += "<a href='" + articleUrl + "'>" +
                      "<p class=\"search-result\">" + highlightKeyword(content, slice) +
                      "...</p>" + "</a>";
                  });

                  resultItem += "</li>";
                  resultItems.push({
                    item: resultItem,
                    searchTextCount: searchTextCount,
                    hitCount: hitCount,
                    id: resultItems.length
                  });
                }
              })
            };
            if (keywords.length === 1 && keywords[0] === "") {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-search fa-5x" /></div>'
            } else if (resultItems.length === 0) {
              resultContent.innerHTML = '<div id="no-result"><i class="fa fa-frown-o fa-5x" /></div>'
            } else {
              resultItems.sort(function (resultLeft, resultRight) {
                if (resultLeft.searchTextCount !== resultRight.searchTextCount) {
                  return resultRight.searchTextCount - resultLeft.searchTextCount;
                } else if (resultLeft.hitCount !== resultRight.hitCount) {
                  return resultRight.hitCount - resultLeft.hitCount;
                } else {
                  return resultRight.id - resultLeft.id;
                }
              });
              var searchResultList = '<ul class=\"search-result-list\">';
              resultItems.forEach(function (result) {
                searchResultList += result.item;
              })
              searchResultList += "</ul>";
              resultContent.innerHTML = searchResultList;
            }
          }

          if ('auto' === 'auto') {
            input.addEventListener('input', inputEventFunction);
          } else {
            $('.search-icon').click(inputEventFunction);
            input.addEventListener('keypress', function (event) {
              if (event.keyCode === 13) {
                inputEventFunction();
              }
            });
          }

          // remove loading animation
          $(".local-search-pop-overlay").remove();
          $('body').css('overflow', '');

          proceedsearch();
        }
      });
    }

    // handle and trigger popup window;
    $('.popup-trigger').click(function(e) {
      e.stopPropagation();
      if (isfetched === false) {
        searchFunc(path, 'local-search-input', 'local-search-result');
      } else {
        proceedsearch();
      };
    });

    $('.popup-btn-close').click(onPopupClose);
    $('.popup').click(function(e){
      e.stopPropagation();
    });
    $(document).on('keyup', function (event) {
      var shouldDismissSearchPopup = event.which === 27 &&
        $('.search-popup').is(':visible');
      if (shouldDismissSearchPopup) {
        onPopupClose();
      }
    });
  </script>





  

  
<script>
(function(){
    var bp = document.createElement('script');
    var curProtocol = window.location.protocol.split(':')[0];
    if (curProtocol === 'https') {
        bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';        
    }
    else {
        bp.src = 'http://push.zhanzhang.baidu.com/push.js';
    }
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(bp, s);
})();
</script>


  

  

  

  

</body>
</html>
